// @vectorize-io/vectorize-client/dist/apis/AIPlatformConnectorsApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g = Object.create((typeof Iterator === "function" ? Iterator : Object).prototype);
    return g.next = verb(0), g["throw"] = verb(1), g["return"] = verb(2), typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIPlatformConnectorsApi = void 0;
var runtime = require("../runtime");
var index_1 = require("../models/index");
/**
 *
 */
var AIPlatformConnectorsApi = /** @class */ (function (_super) {
    __extends(AIPlatformConnectorsApi, _super);
    function AIPlatformConnectorsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Creates a new AI platform connector for embeddings and processing. The specific configuration fields required depend on the platform type selected.
     * Create a new AI platform connector
     */
    AIPlatformConnectorsApi.prototype.createAIPlatformConnectorRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling createAIPlatformConnector().');
                        }
                        if (requestParameters['createAIPlatformConnectorRequest'] == null) {
                            throw new runtime.RequiredError('createAIPlatformConnectorRequest', 'Required parameter "createAIPlatformConnectorRequest" was null or undefined when calling createAIPlatformConnector().');
                        }
                        queryParameters = {};
                        if (requestParameters['workspaceId'] != null) {
                            queryParameters['workspaceId'] = requestParameters['workspaceId'];
                        }
                        headerParameters = {};
                        headerParameters['Content-Type'] = 'application/json';
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/connectors/aiplatforms".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                            body: (0, index_1.CreateAIPlatformConnectorRequestToJSON)(requestParameters['createAIPlatformConnectorRequest']),
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.CreateAIPlatformConnectorResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Creates a new AI platform connector for embeddings and processing. The specific configuration fields required depend on the platform type selected.
     * Create a new AI platform connector
     */
    AIPlatformConnectorsApi.prototype.createAIPlatformConnector = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.createAIPlatformConnectorRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Delete an AI platform connector
     * Delete an AI platform connector
     */
    AIPlatformConnectorsApi.prototype.deleteAIPlatformConnectorRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling deleteAIPlatformConnector().');
                        }
                        if (requestParameters['aiPlatformConnectorId'] == null) {
                            throw new runtime.RequiredError('aiPlatformConnectorId', 'Required parameter "aiPlatformConnectorId" was null or undefined when calling deleteAIPlatformConnector().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/connectors/aiplatforms/{aiPlatformConnectorId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("aiPlatformConnectorId", "}"), encodeURIComponent(String(requestParameters['aiPlatformConnectorId']))),
                            method: 'DELETE',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.DeleteAIPlatformConnectorResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Delete an AI platform connector
     * Delete an AI platform connector
     */
    AIPlatformConnectorsApi.prototype.deleteAIPlatformConnector = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.deleteAIPlatformConnectorRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get an AI platform connector
     * Get an AI platform connector
     */
    AIPlatformConnectorsApi.prototype.getAIPlatformConnectorRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getAIPlatformConnector().');
                        }
                        if (requestParameters['aiPlatformConnectorId'] == null) {
                            throw new runtime.RequiredError('aiPlatformConnectorId', 'Required parameter "aiPlatformConnectorId" was null or undefined when calling getAIPlatformConnector().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/connectors/aiplatforms/{aiPlatformConnectorId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("aiPlatformConnectorId", "}"), encodeURIComponent(String(requestParameters['aiPlatformConnectorId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.AIPlatformConnectorFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get an AI platform connector
     * Get an AI platform connector
     */
    AIPlatformConnectorsApi.prototype.getAIPlatformConnector = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getAIPlatformConnectorRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get all existing AI Platform connectors
     * Get all existing AI Platform connectors
     */
    AIPlatformConnectorsApi.prototype.getAIPlatformConnectorsRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getAIPlatformConnectors().');
                        }
                        queryParameters = {};
                        if (requestParameters['workspaceId'] != null) {
                            queryParameters['workspaceId'] = requestParameters['workspaceId'];
                        }
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/connectors/aiplatforms".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetAIPlatformConnectors200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get all existing AI Platform connectors
     * Get all existing AI Platform connectors
     */
    AIPlatformConnectorsApi.prototype.getAIPlatformConnectors = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getAIPlatformConnectorsRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Update an AI Platform connector
     * Update an AI Platform connector
     */
    AIPlatformConnectorsApi.prototype.updateAIPlatformConnectorRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling updateAIPlatformConnector().');
                        }
                        if (requestParameters['aiPlatformConnectorId'] == null) {
                            throw new runtime.RequiredError('aiPlatformConnectorId', 'Required parameter "aiPlatformConnectorId" was null or undefined when calling updateAIPlatformConnector().');
                        }
                        if (requestParameters['updateAIPlatformConnectorRequest'] == null) {
                            throw new runtime.RequiredError('updateAIPlatformConnectorRequest', 'Required parameter "updateAIPlatformConnectorRequest" was null or undefined when calling updateAIPlatformConnector().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters['Content-Type'] = 'application/json';
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/connectors/aiplatforms/{aiPlatformConnectorId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("aiPlatformConnectorId", "}"), encodeURIComponent(String(requestParameters['aiPlatformConnectorId']))),
                            method: 'PATCH',
                            headers: headerParameters,
                            query: queryParameters,
                            body: (0, index_1.UpdateAIPlatformConnectorRequestToJSON)(requestParameters['updateAIPlatformConnectorRequest']),
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.UpdateAIPlatformConnectorResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Update an AI Platform connector
     * Update an AI Platform connector
     */
    AIPlatformConnectorsApi.prototype.updateAIPlatformConnector = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.updateAIPlatformConnectorRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return AIPlatformConnectorsApi;
}(runtime.BaseAPI));
exports.AIPlatformConnectorsApi = AIPlatformConnectorsApi;
