// @vectorize-io/vectorize-client/dist/apis/PipelinesApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g = Object.create((typeof Iterator === "function" ? Iterator : Object).prototype);
    return g.next = verb(0), g["throw"] = verb(1), g["return"] = verb(2), typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelinesApi = void 0;
var runtime = require("../runtime");
var index_1 = require("../models/index");
/**
 *
 */
var PipelinesApi = /** @class */ (function (_super) {
    __extends(PipelinesApi, _super);
    function PipelinesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Creates a new pipeline with source connectors, destination connector, and AI platform configuration. The specific configuration fields required depend on the connector types selected.
     * Create a new pipeline
     */
    PipelinesApi.prototype.createPipelineRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling createPipeline().');
                        }
                        if (requestParameters['pipelineConfigurationSchema'] == null) {
                            throw new runtime.RequiredError('pipelineConfigurationSchema', 'Required parameter "pipelineConfigurationSchema" was null or undefined when calling createPipeline().');
                        }
                        queryParameters = {};
                        if (requestParameters['workspaceId'] != null) {
                            queryParameters['workspaceId'] = requestParameters['workspaceId'];
                        }
                        headerParameters = {};
                        headerParameters['Content-Type'] = 'application/json';
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                            body: (0, index_1.PipelineConfigurationSchemaToJSON)(requestParameters['pipelineConfigurationSchema']),
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.CreatePipelineResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Creates a new pipeline with source connectors, destination connector, and AI platform configuration. The specific configuration fields required depend on the connector types selected.
     * Create a new pipeline
     */
    PipelinesApi.prototype.createPipeline = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.createPipelineRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Delete a pipeline
     * Delete a pipeline
     */
    PipelinesApi.prototype.deletePipelineRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling deletePipeline().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling deletePipeline().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'DELETE',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.DeletePipelineResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Delete a pipeline
     * Delete a pipeline
     */
    PipelinesApi.prototype.deletePipeline = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.deletePipelineRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get deep research result
     * Get deep research result
     */
    PipelinesApi.prototype.getDeepResearchResultRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getDeepResearchResult().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling getDeepResearchResult().');
                        }
                        if (requestParameters['researchId'] == null) {
                            throw new runtime.RequiredError('researchId', 'Required parameter "researchId" was null or undefined when calling getDeepResearchResult().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/deep-research/{researchId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))).replace("{".concat("researchId", "}"), encodeURIComponent(String(requestParameters['researchId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetDeepResearchResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get deep research result
     * Get deep research result
     */
    PipelinesApi.prototype.getDeepResearchResult = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getDeepResearchResultRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get a pipeline
     * Get a pipeline
     */
    PipelinesApi.prototype.getPipelineRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getPipeline().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling getPipeline().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetPipelineResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get a pipeline
     * Get a pipeline
     */
    PipelinesApi.prototype.getPipeline = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPipelineRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get pipeline events
     * Get pipeline events
     */
    PipelinesApi.prototype.getPipelineEventsRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getPipelineEvents().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling getPipelineEvents().');
                        }
                        queryParameters = {};
                        if (requestParameters['nextToken'] != null) {
                            queryParameters['nextToken'] = requestParameters['nextToken'];
                        }
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/events".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetPipelineEventsResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get pipeline events
     * Get pipeline events
     */
    PipelinesApi.prototype.getPipelineEvents = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPipelineEventsRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Get pipeline metrics
     * Get pipeline metrics
     */
    PipelinesApi.prototype.getPipelineMetricsRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getPipelineMetrics().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling getPipelineMetrics().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/metrics".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetPipelineMetricsResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Get pipeline metrics
     * Get pipeline metrics
     */
    PipelinesApi.prototype.getPipelineMetrics = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPipelineMetricsRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Returns a list of all pipelines in the organization
     * Get all pipelines
     */
    PipelinesApi.prototype.getPipelinesRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling getPipelines().');
                        }
                        queryParameters = {};
                        if (requestParameters['workspaceId'] != null) {
                            queryParameters['workspaceId'] = requestParameters['workspaceId'];
                        }
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))),
                            method: 'GET',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetPipelinesResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Returns a list of all pipelines in the organization
     * Get all pipelines
     */
    PipelinesApi.prototype.getPipelines = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPipelinesRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Retrieve documents from a pipeline
     * Retrieve documents from a pipeline
     */
    PipelinesApi.prototype.retrieveDocumentsRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling retrieveDocuments().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling retrieveDocuments().');
                        }
                        if (requestParameters['retrieveDocumentsRequest'] == null) {
                            throw new runtime.RequiredError('retrieveDocumentsRequest', 'Required parameter "retrieveDocumentsRequest" was null or undefined when calling retrieveDocuments().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters['Content-Type'] = 'application/json';
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/retrieval".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                            body: (0, index_1.RetrieveDocumentsRequestToJSON)(requestParameters['retrieveDocumentsRequest']),
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.RetrieveDocumentsResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Retrieve documents from a pipeline
     * Retrieve documents from a pipeline
     */
    PipelinesApi.prototype.retrieveDocuments = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.retrieveDocumentsRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Start a deep research
     * Start a deep research
     */
    PipelinesApi.prototype.startDeepResearchRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling startDeepResearch().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling startDeepResearch().');
                        }
                        if (requestParameters['startDeepResearchRequest'] == null) {
                            throw new runtime.RequiredError('startDeepResearchRequest', 'Required parameter "startDeepResearchRequest" was null or undefined when calling startDeepResearch().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters['Content-Type'] = 'application/json';
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/deep-research".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                            body: (0, index_1.StartDeepResearchRequestToJSON)(requestParameters['startDeepResearchRequest']),
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.StartDeepResearchResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Start a deep research
     * Start a deep research
     */
    PipelinesApi.prototype.startDeepResearch = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.startDeepResearchRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Start a pipeline
     * Start a pipeline
     */
    PipelinesApi.prototype.startPipelineRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling startPipeline().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling startPipeline().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/start".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.StartPipelineResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Start a pipeline
     * Start a pipeline
     */
    PipelinesApi.prototype.startPipeline = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.startPipelineRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * Stop a pipeline
     * Stop a pipeline
     */
    PipelinesApi.prototype.stopPipelineRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters['organizationId'] == null) {
                            throw new runtime.RequiredError('organizationId', 'Required parameter "organizationId" was null or undefined when calling stopPipeline().');
                        }
                        if (requestParameters['pipelineId'] == null) {
                            throw new runtime.RequiredError('pipelineId', 'Required parameter "pipelineId" was null or undefined when calling stopPipeline().');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("bearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2: return [4 /*yield*/, this.request({
                            path: "/org/{organizationId}/pipelines/{pipelineId}/stop".replace("{".concat("organizationId", "}"), encodeURIComponent(String(requestParameters['organizationId']))).replace("{".concat("pipelineId", "}"), encodeURIComponent(String(requestParameters['pipelineId']))),
                            method: 'POST',
                            headers: headerParameters,
                            query: queryParameters,
                        }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.StopPipelineResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * Stop a pipeline
     * Stop a pipeline
     */
    PipelinesApi.prototype.stopPipeline = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.stopPipelineRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return PipelinesApi;
}(runtime.BaseAPI));
exports.PipelinesApi = PipelinesApi;
