// @vectorize-io/vectorize-client/dist/models/AddUserToSourceConnectorRequestSelectedFilesAnyOfValue.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOfValue = instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOfValue;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSON = AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSONTyped = AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSONTyped;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSON = AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSONTyped = AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSONTyped;
/**
 * Check if a given object implements the AddUserToSourceConnectorRequestSelectedFilesAnyOfValue interface.
 */
function instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOfValue(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('mimeType' in value) || value['mimeType'] === undefined)
        return false;
    return true;
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfValueFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'mimeType': json['mimeType'],
    };
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfValueToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'mimeType': value['mimeType'],
    };
}
