// @vectorize-io/vectorize-client/dist/models/AddUserToSourceConnectorRequestSelectedFiles.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddUserToSourceConnectorRequestSelectedFiles = instanceOfAddUserToSourceConnectorRequestSelectedFiles;
exports.AddUserToSourceConnectorRequestSelectedFilesFromJSON = AddUserToSourceConnectorRequestSelectedFilesFromJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesFromJSONTyped = AddUserToSourceConnectorRequestSelectedFilesFromJSONTyped;
exports.AddUserToSourceConnectorRequestSelectedFilesToJSON = AddUserToSourceConnectorRequestSelectedFilesToJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesToJSONTyped = AddUserToSourceConnectorRequestSelectedFilesToJSONTyped;
/**
 * Check if a given object implements the AddUserToSourceConnectorRequestSelectedFiles interface.
 */
function instanceOfAddUserToSourceConnectorRequestSelectedFiles(value) {
    return true;
}
function AddUserToSourceConnectorRequestSelectedFilesFromJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesFromJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'pageIds': json['pageIds'] == null ? undefined : json['pageIds'],
        'databaseIds': json['databaseIds'] == null ? undefined : json['databaseIds'],
    };
}
function AddUserToSourceConnectorRequestSelectedFilesToJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesToJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'pageIds': value['pageIds'],
        'databaseIds': value['databaseIds'],
    };
}
