// @vectorize-io/vectorize-client/dist/models/AddUserFromSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddUserFromSourceConnectorResponse = instanceOfAddUserFromSourceConnectorResponse;
exports.AddUserFromSourceConnectorResponseFromJSON = AddUserFromSourceConnectorResponseFromJSON;
exports.AddUserFromSourceConnectorResponseFromJSONTyped = AddUserFromSourceConnectorResponseFromJSONTyped;
exports.AddUserFromSourceConnectorResponseToJSON = AddUserFromSourceConnectorResponseToJSON;
exports.AddUserFromSourceConnectorResponseToJSONTyped = AddUserFromSourceConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the AddUserFromSourceConnectorResponse interface.
 */
function instanceOfAddUserFromSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function AddUserFromSourceConnectorResponseFromJSON(json) {
    return AddUserFromSourceConnectorResponseFromJSONTyped(json, false);
}
function AddUserFromSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function AddUserFromSourceConnectorResponseToJSON(json) {
    return AddUserFromSourceConnectorResponseToJSONTyped(json, false);
}
function AddUserFromSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
