// @vectorize-io/vectorize-client/dist/models/AddUserToSourceConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddUserToSourceConnectorRequest = instanceOfAddUserToSourceConnectorRequest;
exports.AddUserToSourceConnectorRequestFromJSON = AddUserToSourceConnectorRequestFromJSON;
exports.AddUserToSourceConnectorRequestFromJSONTyped = AddUserToSourceConnectorRequestFromJSONTyped;
exports.AddUserToSourceConnectorRequestToJSON = AddUserToSourceConnectorRequestToJSON;
exports.AddUserToSourceConnectorRequestToJSONTyped = AddUserToSourceConnectorRequestToJSONTyped;
var AddUserToSourceConnectorRequestSelectedFiles_1 = require("./AddUserToSourceConnectorRequestSelectedFiles");
/**
 * Check if a given object implements the AddUserToSourceConnectorRequest interface.
 */
function instanceOfAddUserToSourceConnectorRequest(value) {
    if (!('userId' in value) || value['userId'] === undefined)
        return false;
    if (!('selectedFiles' in value) || value['selectedFiles'] === undefined)
        return false;
    return true;
}
function AddUserToSourceConnectorRequestFromJSON(json) {
    return AddUserToSourceConnectorRequestFromJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'userId': json['userId'],
        'selectedFiles': (0, AddUserToSourceConnectorRequestSelectedFiles_1.AddUserToSourceConnectorRequestSelectedFilesFromJSON)(json['selectedFiles']),
        'refreshToken': json['refreshToken'] == null ? undefined : json['refreshToken'],
        'accessToken': json['accessToken'] == null ? undefined : json['accessToken'],
    };
}
function AddUserToSourceConnectorRequestToJSON(json) {
    return AddUserToSourceConnectorRequestToJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'userId': value['userId'],
        'selectedFiles': (0, AddUserToSourceConnectorRequestSelectedFiles_1.AddUserToSourceConnectorRequestSelectedFilesToJSON)(value['selectedFiles']),
        'refreshToken': value['refreshToken'],
        'accessToken': value['accessToken'],
    };
}
