// @vectorize-io/vectorize-client/dist/models/AddUserToSourceConnectorRequestSelectedFilesAnyOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOf = instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOf;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSON = AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSONTyped = AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSONTyped;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSON = AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSON;
exports.AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSONTyped = AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSONTyped;
/**
 * Check if a given object implements the AddUserToSourceConnectorRequestSelectedFilesAnyOf interface.
 */
function instanceOfAddUserToSourceConnectorRequestSelectedFilesAnyOf(value) {
    return true;
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'pageIds': json['pageIds'] == null ? undefined : json['pageIds'],
        'databaseIds': json['databaseIds'] == null ? undefined : json['databaseIds'],
    };
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSON(json) {
    return AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSONTyped(json, false);
}
function AddUserToSourceConnectorRequestSelectedFilesAnyOfToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'pageIds': value['pageIds'],
        'databaseIds': value['databaseIds'],
    };
}
