// @vectorize-io/vectorize-client/dist/models/AIPlatformTypeForPipeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIPlatformTypeForPipeline = void 0;
exports.instanceOfAIPlatformTypeForPipeline = instanceOfAIPlatformTypeForPipeline;
exports.AIPlatformTypeForPipelineFromJSON = AIPlatformTypeForPipelineFromJSON;
exports.AIPlatformTypeForPipelineFromJSONTyped = AIPlatformTypeForPipelineFromJSONTyped;
exports.AIPlatformTypeForPipelineToJSON = AIPlatformTypeForPipelineToJSON;
exports.AIPlatformTypeForPipelineToJSONTyped = AIPlatformTypeForPipelineToJSONTyped;
/**
 *
 * @export
 */
exports.AIPlatformTypeForPipeline = {
    Bedrock: 'BEDROCK',
    Vertex: 'VERTEX',
    Openai: 'OPENAI',
    Voyage: 'VOYAGE',
    Vectorize: 'VECTORIZE',
    Anthropic: 'ANTHROPIC',
    Groq: 'GROQ'
};
function instanceOfAIPlatformTypeForPipeline(value) {
    for (var key in exports.AIPlatformTypeForPipeline) {
        if (Object.prototype.hasOwnProperty.call(exports.AIPlatformTypeForPipeline, key)) {
            if (exports.AIPlatformTypeForPipeline[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function AIPlatformTypeForPipelineFromJSON(json) {
    return AIPlatformTypeForPipelineFromJSONTyped(json, false);
}
function AIPlatformTypeForPipelineFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function AIPlatformTypeForPipelineToJSON(value) {
    return value;
}
function AIPlatformTypeForPipelineToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
