// @vectorize-io/vectorize-client/dist/models/AIPlatformConfigSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIPlatformConfigSchemaExtractionStrategyEnum = exports.AIPlatformConfigSchemaChunkingStrategyEnum = exports.AIPlatformConfigSchemaEmbeddingModelEnum = void 0;
exports.instanceOfAIPlatformConfigSchema = instanceOfAIPlatformConfigSchema;
exports.AIPlatformConfigSchemaFromJSON = AIPlatformConfigSchemaFromJSON;
exports.AIPlatformConfigSchemaFromJSONTyped = AIPlatformConfigSchemaFromJSONTyped;
exports.AIPlatformConfigSchemaToJSON = AIPlatformConfigSchemaToJSON;
exports.AIPlatformConfigSchemaToJSONTyped = AIPlatformConfigSchemaToJSONTyped;
/**
 * @export
 */
exports.AIPlatformConfigSchemaEmbeddingModelEnum = {
    VectorizeOpenAiTextEmbedding2: 'VECTORIZE_OPEN_AI_TEXT_EMBEDDING_2',
    VectorizeOpenAiTextEmbedding3Large: 'VECTORIZE_OPEN_AI_TEXT_EMBEDDING_3_LARGE',
    VectorizeOpenAiTextEmbedding3Small: 'VECTORIZE_OPEN_AI_TEXT_EMBEDDING_3_SMALL',
    VectorizeVoyageAi2: 'VECTORIZE_VOYAGE_AI_2',
    VectorizeVoyageAi3: 'VECTORIZE_VOYAGE_AI_3',
    VectorizeVoyageAi3Lite: 'VECTORIZE_VOYAGE_AI_3_LITE',
    VectorizeVoyageAi3Large: 'VECTORIZE_VOYAGE_AI_3_LARGE',
    VectorizeVoyageAiFinance2: 'VECTORIZE_VOYAGE_AI_FINANCE_2',
    VectorizeVoyageAiMultilingual2: 'VECTORIZE_VOYAGE_AI_MULTILINGUAL_2',
    VectorizeVoyageAiLaw2: 'VECTORIZE_VOYAGE_AI_LAW_2',
    VectorizeVoyageAiCode2: 'VECTORIZE_VOYAGE_AI_CODE_2',
    VectorizeVoyageAi35: 'VECTORIZE_VOYAGE_AI_35',
    VectorizeVoyageAi35Lite: 'VECTORIZE_VOYAGE_AI_35_LITE',
    VectorizeVoyageAiCode3: 'VECTORIZE_VOYAGE_AI_CODE_3',
    VectorizeTitanTextEmbedding2: 'VECTORIZE_TITAN_TEXT_EMBEDDING_2',
    VectorizeTitanTextEmbedding1: 'VECTORIZE_TITAN_TEXT_EMBEDDING_1',
    OpenAiTextEmbedding2: 'OPEN_AI_TEXT_EMBEDDING_2',
    OpenAiTextEmbedding3Small: 'OPEN_AI_TEXT_EMBEDDING_3_SMALL',
    OpenAiTextEmbedding3Large: 'OPEN_AI_TEXT_EMBEDDING_3_LARGE',
    VoyageAi2: 'VOYAGE_AI_2',
    VoyageAi3: 'VOYAGE_AI_3',
    VoyageAi3Lite: 'VOYAGE_AI_3_LITE',
    VoyageAi3Large: 'VOYAGE_AI_3_LARGE',
    VoyageAiFinance2: 'VOYAGE_AI_FINANCE_2',
    VoyageAiMultilingual2: 'VOYAGE_AI_MULTILINGUAL_2',
    VoyageAiLaw2: 'VOYAGE_AI_LAW_2',
    VoyageAiCode2: 'VOYAGE_AI_CODE_2',
    VoyageAi35: 'VOYAGE_AI_35',
    VoyageAi35Lite: 'VOYAGE_AI_35_LITE',
    VoyageAiCode3: 'VOYAGE_AI_CODE_3',
    TitanTextEmbedding1: 'TITAN_TEXT_EMBEDDING_1',
    TitanTextEmbedding2: 'TITAN_TEXT_EMBEDDING_2',
    VertexTextEmbedding4: 'VERTEX_TEXT_EMBEDDING_4',
    VertexTextEmbeddingGecko3: 'VERTEX_TEXT_EMBEDDING_GECKO_3',
    VertexGeckoMultilingual1: 'VERTEX_GECKO_MULTILINGUAL_1',
    VertexMultilingualEmbedding2: 'VERTEX_MULTILINGUAL_EMBEDDING_2'
};
/**
 * @export
 */
exports.AIPlatformConfigSchemaChunkingStrategyEnum = {
    Fixed: 'FIXED',
    Sentence: 'SENTENCE',
    Paragraph: 'PARAGRAPH',
    Markdown: 'MARKDOWN'
};
/**
 * @export
 */
exports.AIPlatformConfigSchemaExtractionStrategyEnum = {
    Fast: 'FAST',
    Iris: 'IRIS',
    Mixed: 'MIXED'
};
/**
 * Check if a given object implements the AIPlatformConfigSchema interface.
 */
function instanceOfAIPlatformConfigSchema(value) {
    return true;
}
function AIPlatformConfigSchemaFromJSON(json) {
    return AIPlatformConfigSchemaFromJSONTyped(json, false);
}
function AIPlatformConfigSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'embeddingModel': json['embeddingModel'] == null ? undefined : json['embeddingModel'],
        'chunkingStrategy': json['chunkingStrategy'] == null ? undefined : json['chunkingStrategy'],
        'chunkSize': json['chunkSize'] == null ? undefined : json['chunkSize'],
        'chunkOverlap': json['chunkOverlap'] == null ? undefined : json['chunkOverlap'],
        'dimensions': json['dimensions'] == null ? undefined : json['dimensions'],
        'extractionStrategy': json['extractionStrategy'] == null ? undefined : json['extractionStrategy'],
    };
}
function AIPlatformConfigSchemaToJSON(json) {
    return AIPlatformConfigSchemaToJSONTyped(json, false);
}
function AIPlatformConfigSchemaToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'embeddingModel': value['embeddingModel'],
        'chunkingStrategy': value['chunkingStrategy'],
        'chunkSize': value['chunkSize'],
        'chunkOverlap': value['chunkOverlap'],
        'dimensions': value['dimensions'],
        'extractionStrategy': value['extractionStrategy'],
    };
}
