// @vectorize-io/vectorize-client/dist/models/AIPlatformConnectorType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIPlatformConnectorType = void 0;
exports.instanceOfAIPlatformConnectorType = instanceOfAIPlatformConnectorType;
exports.AIPlatformConnectorTypeFromJSON = AIPlatformConnectorTypeFromJSON;
exports.AIPlatformConnectorTypeFromJSONTyped = AIPlatformConnectorTypeFromJSONTyped;
exports.AIPlatformConnectorTypeToJSON = AIPlatformConnectorTypeToJSON;
exports.AIPlatformConnectorTypeToJSONTyped = AIPlatformConnectorTypeToJSONTyped;
/**
 *
 * @export
 */
exports.AIPlatformConnectorType = {
    Bedrock: 'BEDROCK',
    Vertex: 'VERTEX',
    Openai: 'OPENAI',
    Voyage: 'VOYAGE',
    Anthropic: 'ANTHROPIC',
    Groq: 'GROQ'
};
function instanceOfAIPlatformConnectorType(value) {
    for (var key in exports.AIPlatformConnectorType) {
        if (Object.prototype.hasOwnProperty.call(exports.AIPlatformConnectorType, key)) {
            if (exports.AIPlatformConnectorType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function AIPlatformConnectorTypeFromJSON(json) {
    return AIPlatformConnectorTypeFromJSONTyped(json, false);
}
function AIPlatformConnectorTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function AIPlatformConnectorTypeToJSON(value) {
    return value;
}
function AIPlatformConnectorTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
