// @vectorize-io/vectorize-client/dist/models/Anthropic.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnthropicTypeEnum = void 0;
exports.instanceOfAnthropic = instanceOfAnthropic;
exports.AnthropicFromJSON = AnthropicFromJSON;
exports.AnthropicFromJSONTyped = AnthropicFromJSONTyped;
exports.AnthropicToJSON = AnthropicToJSON;
exports.AnthropicToJSONTyped = AnthropicToJSONTyped;
var ANTHROPICAuthConfig_1 = require("./ANTHROPICAuthConfig");
/**
 * @export
 */
exports.AnthropicTypeEnum = {
    Anthropic: 'ANTHROPIC'
};
/**
 * Check if a given object implements the Anthropic interface.
 */
function instanceOfAnthropic(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function AnthropicFromJSON(json) {
    return AnthropicFromJSONTyped(json, false);
}
function AnthropicFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, ANTHROPICAuthConfig_1.ANTHROPICAuthConfigFromJSON)(json['config']),
    };
}
function AnthropicToJSON(json) {
    return AnthropicToJSONTyped(json, false);
}
function AnthropicToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, ANTHROPICAuthConfig_1.ANTHROPICAuthConfigToJSON)(value['config']),
    };
}
