// @vectorize-io/vectorize-client/dist/models/Anthropic1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAnthropic1 = instanceOfAnthropic1;
exports.Anthropic1FromJSON = Anthropic1FromJSON;
exports.Anthropic1FromJSONTyped = Anthropic1FromJSONTyped;
exports.Anthropic1ToJSON = Anthropic1ToJSON;
exports.Anthropic1ToJSONTyped = Anthropic1ToJSONTyped;
/**
 * Check if a given object implements the Anthropic1 interface.
 */
function instanceOfAnthropic1(value) {
    return true;
}
function Anthropic1FromJSON(json) {
    return Anthropic1FromJSONTyped(json, false);
}
function Anthropic1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Anthropic1ToJSON(json) {
    return Anthropic1ToJSONTyped(json, false);
}
function Anthropic1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
