// @vectorize-io/vectorize-client/dist/models/ANTHROPICAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfANTHROPICAuthConfig = instanceOfANTHROPICAuthConfig;
exports.ANTHROPICAuthConfigFromJSON = ANTHROPICAuthConfigFromJSON;
exports.ANTHROPICAuthConfigFromJSONTyped = ANTHROPICAuthConfigFromJSONTyped;
exports.ANTHROPICAuthConfigToJSON = ANTHROPICAuthConfigToJSON;
exports.ANTHROPICAuthConfigToJSONTyped = ANTHROPICAuthConfigToJSONTyped;
/**
 * Check if a given object implements the ANTHROPICAuthConfig interface.
 */
function instanceOfANTHROPICAuthConfig(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    return true;
}
function ANTHROPICAuthConfigFromJSON(json) {
    return ANTHROPICAuthConfigFromJSONTyped(json, false);
}
function ANTHROPICAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
function ANTHROPICAuthConfigToJSON(json) {
    return ANTHROPICAuthConfigToJSONTyped(json, false);
}
function ANTHROPICAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
    };
}
