// @vectorize-io/vectorize-client/dist/models/AwsS31.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAwsS31 = instanceOfAwsS31;
exports.AwsS31FromJSON = AwsS31FromJSON;
exports.AwsS31FromJSONTyped = AwsS31FromJSONTyped;
exports.AwsS31ToJSON = AwsS31ToJSON;
exports.AwsS31ToJSONTyped = AwsS31ToJSONTyped;
var AWSS3AuthConfig_1 = require("./AWSS3AuthConfig");
/**
 * Check if a given object implements the AwsS31 interface.
 */
function instanceOfAwsS31(value) {
    return true;
}
function AwsS31FromJSON(json) {
    return AwsS31FromJSONTyped(json, false);
}
function AwsS31FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, AWSS3AuthConfig_1.AWSS3AuthConfigFromJSON)(json['config']),
    };
}
function AwsS31ToJSON(json) {
    return AwsS31ToJSONTyped(json, false);
}
function AwsS31ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, AWSS3AuthConfig_1.AWSS3AuthConfigToJSON)(value['config']),
    };
}
