// @vectorize-io/vectorize-client/dist/models/AWSS3Config.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AWSS3ConfigFileExtensionsEnum = void 0;
exports.instanceOfAWSS3Config = instanceOfAWSS3Config;
exports.AWSS3ConfigFromJSON = AWSS3ConfigFromJSON;
exports.AWSS3ConfigFromJSONTyped = AWSS3ConfigFromJSONTyped;
exports.AWSS3ConfigToJSON = AWSS3ConfigToJSON;
exports.AWSS3ConfigToJSONTyped = AWSS3ConfigToJSONTyped;
/**
 * @export
 */
exports.AWSS3ConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Md: 'md',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the AWSS3Config interface.
 */
function instanceOfAWSS3Config(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    if (!('idleTime' in value) || value['idleTime'] === undefined)
        return false;
    return true;
}
function AWSS3ConfigFromJSON(json) {
    return AWSS3ConfigFromJSONTyped(json, false);
}
function AWSS3ConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'idleTime': json['idle-time'],
        'recursive': json['recursive'] == null ? undefined : json['recursive'],
        'pathPrefix': json['path-prefix'] == null ? undefined : json['path-prefix'],
        'pathMetadataRegex': json['path-metadata-regex'] == null ? undefined : json['path-metadata-regex'],
        'pathRegexGroupNames': json['path-regex-group-names'] == null ? undefined : json['path-regex-group-names'],
    };
}
function AWSS3ConfigToJSON(json) {
    return AWSS3ConfigToJSONTyped(json, false);
}
function AWSS3ConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'idle-time': value['idleTime'],
        'recursive': value['recursive'],
        'path-prefix': value['pathPrefix'],
        'path-metadata-regex': value['pathMetadataRegex'],
        'path-regex-group-names': value['pathRegexGroupNames'],
    };
}
