// @vectorize-io/vectorize-client/dist/models/AWSS3AuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAWSS3AuthConfig = instanceOfAWSS3AuthConfig;
exports.AWSS3AuthConfigFromJSON = AWSS3AuthConfigFromJSON;
exports.AWSS3AuthConfigFromJSONTyped = AWSS3AuthConfigFromJSONTyped;
exports.AWSS3AuthConfigToJSON = AWSS3AuthConfigToJSON;
exports.AWSS3AuthConfigToJSONTyped = AWSS3AuthConfigToJSONTyped;
/**
 * Check if a given object implements the AWSS3AuthConfig interface.
 */
function instanceOfAWSS3AuthConfig(value) {
    if (!('accessKey' in value) || value['accessKey'] === undefined)
        return false;
    if (!('secretKey' in value) || value['secretKey'] === undefined)
        return false;
    if (!('bucketName' in value) || value['bucketName'] === undefined)
        return false;
    if (!('archiver' in value) || value['archiver'] === undefined)
        return false;
    return true;
}
function AWSS3AuthConfigFromJSON(json) {
    return AWSS3AuthConfigFromJSONTyped(json, false);
}
function AWSS3AuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'accessKey': json['access-key'],
        'secretKey': json['secret-key'],
        'bucketName': json['bucket-name'],
        'endpoint': json['endpoint'] == null ? undefined : json['endpoint'],
        'region': json['region'] == null ? undefined : json['region'],
        'archiver': json['archiver'],
    };
}
function AWSS3AuthConfigToJSON(json) {
    return AWSS3AuthConfigToJSONTyped(json, false);
}
function AWSS3AuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'access-key': value['accessKey'],
        'secret-key': value['secretKey'],
        'bucket-name': value['bucketName'],
        'endpoint': value['endpoint'],
        'region': value['region'],
        'archiver': value['archiver'],
    };
}
