// @vectorize-io/vectorize-client/dist/models/AwsS3.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsS3TypeEnum = void 0;
exports.instanceOfAwsS3 = instanceOfAwsS3;
exports.AwsS3FromJSON = AwsS3FromJSON;
exports.AwsS3FromJSONTyped = AwsS3FromJSONTyped;
exports.AwsS3ToJSON = AwsS3ToJSON;
exports.AwsS3ToJSONTyped = AwsS3ToJSONTyped;
var AWSS3AuthConfig_1 = require("./AWSS3AuthConfig");
/**
 * @export
 */
exports.AwsS3TypeEnum = {
    AwsS3: 'AWS_S3'
};
/**
 * Check if a given object implements the AwsS3 interface.
 */
function instanceOfAwsS3(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function AwsS3FromJSON(json) {
    return AwsS3FromJSONTyped(json, false);
}
function AwsS3FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, AWSS3AuthConfig_1.AWSS3AuthConfigFromJSON)(json['config']),
    };
}
function AwsS3ToJSON(json) {
    return AwsS3ToJSONTyped(json, false);
}
function AwsS3ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, AWSS3AuthConfig_1.AWSS3AuthConfigToJSON)(value['config']),
    };
}
