// @vectorize-io/vectorize-client/dist/models/AZUREBLOBAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAZUREBLOBAuthConfig = instanceOfAZUREBLOBAuthConfig;
exports.AZUREBLOBAuthConfigFromJSON = AZUREBLOBAuthConfigFromJSON;
exports.AZUREBLOBAuthConfigFromJSONTyped = AZUREBLOBAuthConfigFromJSONTyped;
exports.AZUREBLOBAuthConfigToJSON = AZUREBLOBAuthConfigToJSON;
exports.AZUREBLOBAuthConfigToJSONTyped = AZUREBLOBAuthConfigToJSONTyped;
/**
 * Check if a given object implements the AZUREBLOBAuthConfig interface.
 */
function instanceOfAZUREBLOBAuthConfig(value) {
    if (!('storageAccountName' in value) || value['storageAccountName'] === undefined)
        return false;
    if (!('storageAccountKey' in value) || value['storageAccountKey'] === undefined)
        return false;
    if (!('container' in value) || value['container'] === undefined)
        return false;
    return true;
}
function AZUREBLOBAuthConfigFromJSON(json) {
    return AZUREBLOBAuthConfigFromJSONTyped(json, false);
}
function AZUREBLOBAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'storageAccountName': json['storage-account-name'],
        'storageAccountKey': json['storage-account-key'],
        'container': json['container'],
        'endpoint': json['endpoint'] == null ? undefined : json['endpoint'],
    };
}
function AZUREBLOBAuthConfigToJSON(json) {
    return AZUREBLOBAuthConfigToJSONTyped(json, false);
}
function AZUREBLOBAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'storage-account-name': value['storageAccountName'],
        'storage-account-key': value['storageAccountKey'],
        'container': value['container'],
        'endpoint': value['endpoint'],
    };
}
