// @vectorize-io/vectorize-client/dist/models/Azureaisearch.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureaisearchTypeEnum = void 0;
exports.instanceOfAzureaisearch = instanceOfAzureaisearch;
exports.AzureaisearchFromJSON = AzureaisearchFromJSON;
exports.AzureaisearchFromJSONTyped = AzureaisearchFromJSONTyped;
exports.AzureaisearchToJSON = AzureaisearchToJSON;
exports.AzureaisearchToJSONTyped = AzureaisearchToJSONTyped;
var AZUREAISEARCHAuthConfig_1 = require("./AZUREAISEARCHAuthConfig");
/**
 * @export
 */
exports.AzureaisearchTypeEnum = {
    Azureaisearch: 'AZUREAISEARCH'
};
/**
 * Check if a given object implements the Azureaisearch interface.
 */
function instanceOfAzureaisearch(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function AzureaisearchFromJSON(json) {
    return AzureaisearchFromJSONTyped(json, false);
}
function AzureaisearchFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, AZUREAISEARCHAuthConfig_1.AZUREAISEARCHAuthConfigFromJSON)(json['config']),
    };
}
function AzureaisearchToJSON(json) {
    return AzureaisearchToJSONTyped(json, false);
}
function AzureaisearchToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, AZUREAISEARCHAuthConfig_1.AZUREAISEARCHAuthConfigToJSON)(value['config']),
    };
}
