// @vectorize-io/vectorize-client/dist/models/AzureBlob.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureBlobTypeEnum = void 0;
exports.instanceOfAzureBlob = instanceOfAzureBlob;
exports.AzureBlobFromJSON = AzureBlobFromJSON;
exports.AzureBlobFromJSONTyped = AzureBlobFromJSONTyped;
exports.AzureBlobToJSON = AzureBlobToJSON;
exports.AzureBlobToJSONTyped = AzureBlobToJSONTyped;
var AZUREBLOBAuthConfig_1 = require("./AZUREBLOBAuthConfig");
/**
 * @export
 */
exports.AzureBlobTypeEnum = {
    AzureBlob: 'AZURE_BLOB'
};
/**
 * Check if a given object implements the AzureBlob interface.
 */
function instanceOfAzureBlob(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function AzureBlobFromJSON(json) {
    return AzureBlobFromJSONTyped(json, false);
}
function AzureBlobFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, AZUREBLOBAuthConfig_1.AZUREBLOBAuthConfigFromJSON)(json['config']),
    };
}
function AzureBlobToJSON(json) {
    return AzureBlobToJSONTyped(json, false);
}
function AzureBlobToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, AZUREBLOBAuthConfig_1.AZUREBLOBAuthConfigToJSON)(value['config']),
    };
}
