// @vectorize-io/vectorize-client/dist/models/Azureaisearch1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAzureaisearch1 = instanceOfAzureaisearch1;
exports.Azureaisearch1FromJSON = Azureaisearch1FromJSON;
exports.Azureaisearch1FromJSONTyped = Azureaisearch1FromJSONTyped;
exports.Azureaisearch1ToJSON = Azureaisearch1ToJSON;
exports.Azureaisearch1ToJSONTyped = Azureaisearch1ToJSONTyped;
var AZUREAISEARCHAuthConfig_1 = require("./AZUREAISEARCHAuthConfig");
/**
 * Check if a given object implements the Azureaisearch1 interface.
 */
function instanceOfAzureaisearch1(value) {
    return true;
}
function Azureaisearch1FromJSON(json) {
    return Azureaisearch1FromJSONTyped(json, false);
}
function Azureaisearch1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, AZUREAISEARCHAuthConfig_1.AZUREAISEARCHAuthConfigFromJSON)(json['config']),
    };
}
function Azureaisearch1ToJSON(json) {
    return Azureaisearch1ToJSONTyped(json, false);
}
function Azureaisearch1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, AZUREAISEARCHAuthConfig_1.AZUREAISEARCHAuthConfigToJSON)(value['config']),
    };
}
