// @vectorize-io/vectorize-client/dist/models/AzureBlob1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAzureBlob1 = instanceOfAzureBlob1;
exports.AzureBlob1FromJSON = AzureBlob1FromJSON;
exports.AzureBlob1FromJSONTyped = AzureBlob1FromJSONTyped;
exports.AzureBlob1ToJSON = AzureBlob1ToJSON;
exports.AzureBlob1ToJSONTyped = AzureBlob1ToJSONTyped;
var AZUREBLOBAuthConfig_1 = require("./AZUREBLOBAuthConfig");
/**
 * Check if a given object implements the AzureBlob1 interface.
 */
function instanceOfAzureBlob1(value) {
    return true;
}
function AzureBlob1FromJSON(json) {
    return AzureBlob1FromJSONTyped(json, false);
}
function AzureBlob1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, AZUREBLOBAuthConfig_1.AZUREBLOBAuthConfigFromJSON)(json['config']),
    };
}
function AzureBlob1ToJSON(json) {
    return AzureBlob1ToJSONTyped(json, false);
}
function AzureBlob1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, AZUREBLOBAuthConfig_1.AZUREBLOBAuthConfigToJSON)(value['config']),
    };
}
