// @vectorize-io/vectorize-client/dist/models/AZUREAISEARCHConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAZUREAISEARCHConfig = instanceOfAZUREAISEARCHConfig;
exports.AZUREAISEARCHConfigFromJSON = AZUREAISEARCHConfigFromJSON;
exports.AZUREAISEARCHConfigFromJSONTyped = AZUREAISEARCHConfigFromJSONTyped;
exports.AZUREAISEARCHConfigToJSON = AZUREAISEARCHConfigToJSON;
exports.AZUREAISEARCHConfigToJSONTyped = AZUREAISEARCHConfigToJSONTyped;
/**
 * Check if a given object implements the AZUREAISEARCHConfig interface.
 */
function instanceOfAZUREAISEARCHConfig(value) {
    if (!('index' in value) || value['index'] === undefined)
        return false;
    return true;
}
function AZUREAISEARCHConfigFromJSON(json) {
    return AZUREAISEARCHConfigFromJSONTyped(json, false);
}
function AZUREAISEARCHConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'index': json['index'],
    };
}
function AZUREAISEARCHConfigToJSON(json) {
    return AZUREAISEARCHConfigToJSONTyped(json, false);
}
function AZUREAISEARCHConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'index': value['index'],
    };
}
