// @vectorize-io/vectorize-client/dist/models/AZUREAISEARCHAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAZUREAISEARCHAuthConfig = instanceOfAZUREAISEARCHAuthConfig;
exports.AZUREAISEARCHAuthConfigFromJSON = AZUREAISEARCHAuthConfigFromJSON;
exports.AZUREAISEARCHAuthConfigFromJSONTyped = AZUREAISEARCHAuthConfigFromJSONTyped;
exports.AZUREAISEARCHAuthConfigToJSON = AZUREAISEARCHAuthConfigToJSON;
exports.AZUREAISEARCHAuthConfigToJSONTyped = AZUREAISEARCHAuthConfigToJSONTyped;
/**
 * Check if a given object implements the AZUREAISEARCHAuthConfig interface.
 */
function instanceOfAZUREAISEARCHAuthConfig(value) {
    if (!('serviceName' in value) || value['serviceName'] === undefined)
        return false;
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function AZUREAISEARCHAuthConfigFromJSON(json) {
    return AZUREAISEARCHAuthConfigFromJSONTyped(json, false);
}
function AZUREAISEARCHAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'serviceName': json['service-name'],
        'apiKey': json['api-key'],
    };
}
function AZUREAISEARCHAuthConfigToJSON(json) {
    return AZUREAISEARCHAuthConfigToJSONTyped(json, false);
}
function AZUREAISEARCHAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'service-name': value['serviceName'],
        'api-key': value['apiKey'],
    };
}
