// @vectorize-io/vectorize-client/dist/models/BEDROCKAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBEDROCKAuthConfig = instanceOfBEDROCKAuthConfig;
exports.BEDROCKAuthConfigFromJSON = BEDROCKAuthConfigFromJSON;
exports.BEDROCKAuthConfigFromJSONTyped = BEDROCKAuthConfigFromJSONTyped;
exports.BEDROCKAuthConfigToJSON = BEDROCKAuthConfigToJSON;
exports.BEDROCKAuthConfigToJSONTyped = BEDROCKAuthConfigToJSONTyped;
/**
 * Check if a given object implements the BEDROCKAuthConfig interface.
 */
function instanceOfBEDROCKAuthConfig(value) {
    if (!('accessKey' in value) || value['accessKey'] === undefined)
        return false;
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('region' in value) || value['region'] === undefined)
        return false;
    return true;
}
function BEDROCKAuthConfigFromJSON(json) {
    return BEDROCKAuthConfigFromJSONTyped(json, false);
}
function BEDROCKAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'accessKey': json['access-key'],
        'key': json['key'],
        'region': json['region'],
    };
}
function BEDROCKAuthConfigToJSON(json) {
    return BEDROCKAuthConfigToJSONTyped(json, false);
}
function BEDROCKAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'access-key': value['accessKey'],
        'key': value['key'],
        'region': value['region'],
    };
}
