// @vectorize-io/vectorize-client/dist/models/Bedrock1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBedrock1 = instanceOfBedrock1;
exports.Bedrock1FromJSON = Bedrock1FromJSON;
exports.Bedrock1FromJSONTyped = Bedrock1FromJSONTyped;
exports.Bedrock1ToJSON = Bedrock1ToJSON;
exports.Bedrock1ToJSONTyped = Bedrock1ToJSONTyped;
/**
 * Check if a given object implements the Bedrock1 interface.
 */
function instanceOfBedrock1(value) {
    return true;
}
function Bedrock1FromJSON(json) {
    return Bedrock1FromJSONTyped(json, false);
}
function Bedrock1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Bedrock1ToJSON(json) {
    return Bedrock1ToJSONTyped(json, false);
}
function Bedrock1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
