// @vectorize-io/vectorize-client/dist/models/Bedrock.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BedrockTypeEnum = void 0;
exports.instanceOfBedrock = instanceOfBedrock;
exports.BedrockFromJSON = BedrockFromJSON;
exports.BedrockFromJSONTyped = BedrockFromJSONTyped;
exports.BedrockToJSON = BedrockToJSON;
exports.BedrockToJSONTyped = BedrockToJSONTyped;
var BEDROCKAuthConfig_1 = require("./BEDROCKAuthConfig");
/**
 * @export
 */
exports.BedrockTypeEnum = {
    Bedrock: 'BEDROCK'
};
/**
 * Check if a given object implements the Bedrock interface.
 */
function instanceOfBedrock(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function BedrockFromJSON(json) {
    return BedrockFromJSONTyped(json, false);
}
function BedrockFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, BEDROCKAuthConfig_1.BEDROCKAuthConfigFromJSON)(json['config']),
    };
}
function BedrockToJSON(json) {
    return BedrockToJSONTyped(json, false);
}
function BedrockToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, BEDROCKAuthConfig_1.BEDROCKAuthConfigToJSON)(value['config']),
    };
}
