// @vectorize-io/vectorize-client/dist/models/CAPELLAConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCAPELLAConfig = instanceOfCAPELLAConfig;
exports.CAPELLAConfigFromJSON = CAPELLAConfigFromJSON;
exports.CAPELLAConfigFromJSONTyped = CAPELLAConfigFromJSONTyped;
exports.CAPELLAConfigToJSON = CAPELLAConfigToJSON;
exports.CAPELLAConfigToJSONTyped = CAPELLAConfigToJSONTyped;
/**
 * Check if a given object implements the CAPELLAConfig interface.
 */
function instanceOfCAPELLAConfig(value) {
    if (!('bucket' in value) || value['bucket'] === undefined)
        return false;
    if (!('scope' in value) || value['scope'] === undefined)
        return false;
    if (!('collection' in value) || value['collection'] === undefined)
        return false;
    if (!('index' in value) || value['index'] === undefined)
        return false;
    return true;
}
function CAPELLAConfigFromJSON(json) {
    return CAPELLAConfigFromJSONTyped(json, false);
}
function CAPELLAConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'bucket': json['bucket'],
        'scope': json['scope'],
        'collection': json['collection'],
        'index': json['index'],
    };
}
function CAPELLAConfigToJSON(json) {
    return CAPELLAConfigToJSONTyped(json, false);
}
function CAPELLAConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'bucket': value['bucket'],
        'scope': value['scope'],
        'collection': value['collection'],
        'index': value['index'],
    };
}
