// @vectorize-io/vectorize-client/dist/models/Capella1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCapella1 = instanceOfCapella1;
exports.Capella1FromJSON = Capella1FromJSON;
exports.Capella1FromJSONTyped = Capella1FromJSONTyped;
exports.Capella1ToJSON = Capella1ToJSON;
exports.Capella1ToJSONTyped = Capella1ToJSONTyped;
var CAPELLAAuthConfig_1 = require("./CAPELLAAuthConfig");
/**
 * Check if a given object implements the Capella1 interface.
 */
function instanceOfCapella1(value) {
    return true;
}
function Capella1FromJSON(json) {
    return Capella1FromJSONTyped(json, false);
}
function Capella1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, CAPELLAAuthConfig_1.CAPELLAAuthConfigFromJSON)(json['config']),
    };
}
function Capella1ToJSON(json) {
    return Capella1ToJSONTyped(json, false);
}
function Capella1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, CAPELLAAuthConfig_1.CAPELLAAuthConfigToJSON)(value['config']),
    };
}
