// @vectorize-io/vectorize-client/dist/models/Capella.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapellaTypeEnum = void 0;
exports.instanceOfCapella = instanceOfCapella;
exports.CapellaFromJSON = CapellaFromJSON;
exports.CapellaFromJSONTyped = CapellaFromJSONTyped;
exports.CapellaToJSON = CapellaToJSON;
exports.CapellaToJSONTyped = CapellaToJSONTyped;
var CAPELLAAuthConfig_1 = require("./CAPELLAAuthConfig");
/**
 * @export
 */
exports.CapellaTypeEnum = {
    Capella: 'CAPELLA'
};
/**
 * Check if a given object implements the Capella interface.
 */
function instanceOfCapella(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function CapellaFromJSON(json) {
    return CapellaFromJSONTyped(json, false);
}
function CapellaFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, CAPELLAAuthConfig_1.CAPELLAAuthConfigFromJSON)(json['config']),
    };
}
function CapellaToJSON(json) {
    return CapellaToJSONTyped(json, false);
}
function CapellaToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, CAPELLAAuthConfig_1.CAPELLAAuthConfigToJSON)(value['config']),
    };
}
