// @vectorize-io/vectorize-client/dist/models/CAPELLAAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCAPELLAAuthConfig = instanceOfCAPELLAAuthConfig;
exports.CAPELLAAuthConfigFromJSON = CAPELLAAuthConfigFromJSON;
exports.CAPELLAAuthConfigFromJSONTyped = CAPELLAAuthConfigFromJSONTyped;
exports.CAPELLAAuthConfigToJSON = CAPELLAAuthConfigToJSON;
exports.CAPELLAAuthConfigToJSONTyped = CAPELLAAuthConfigToJSONTyped;
/**
 * Check if a given object implements the CAPELLAAuthConfig interface.
 */
function instanceOfCAPELLAAuthConfig(value) {
    if (!('username' in value) || value['username'] === undefined)
        return false;
    if (!('password' in value) || value['password'] === undefined)
        return false;
    if (!('connectionString' in value) || value['connectionString'] === undefined)
        return false;
    return true;
}
function CAPELLAAuthConfigFromJSON(json) {
    return CAPELLAAuthConfigFromJSONTyped(json, false);
}
function CAPELLAAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'username': json['username'],
        'password': json['password'],
        'connectionString': json['connection-string'],
    };
}
function CAPELLAAuthConfigToJSON(json) {
    return CAPELLAAuthConfigToJSONTyped(json, false);
}
function CAPELLAAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'username': value['username'],
        'password': value['password'],
        'connection-string': value['connectionString'],
    };
}
