// @vectorize-io/vectorize-client/dist/models/Confluence1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfConfluence1 = instanceOfConfluence1;
exports.Confluence1FromJSON = Confluence1FromJSON;
exports.Confluence1FromJSONTyped = Confluence1FromJSONTyped;
exports.Confluence1ToJSON = Confluence1ToJSON;
exports.Confluence1ToJSONTyped = Confluence1ToJSONTyped;
var CONFLUENCEAuthConfig_1 = require("./CONFLUENCEAuthConfig");
/**
 * Check if a given object implements the Confluence1 interface.
 */
function instanceOfConfluence1(value) {
    return true;
}
function Confluence1FromJSON(json) {
    return Confluence1FromJSONTyped(json, false);
}
function Confluence1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, CONFLUENCEAuthConfig_1.CONFLUENCEAuthConfigFromJSON)(json['config']),
    };
}
function Confluence1ToJSON(json) {
    return Confluence1ToJSONTyped(json, false);
}
function Confluence1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, CONFLUENCEAuthConfig_1.CONFLUENCEAuthConfigToJSON)(value['config']),
    };
}
