// @vectorize-io/vectorize-client/dist/models/CONFLUENCEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCONFLUENCEConfig = instanceOfCONFLUENCEConfig;
exports.CONFLUENCEConfigFromJSON = CONFLUENCEConfigFromJSON;
exports.CONFLUENCEConfigFromJSONTyped = CONFLUENCEConfigFromJSONTyped;
exports.CONFLUENCEConfigToJSON = CONFLUENCEConfigToJSON;
exports.CONFLUENCEConfigToJSONTyped = CONFLUENCEConfigToJSONTyped;
/**
 * Check if a given object implements the CONFLUENCEConfig interface.
 */
function instanceOfCONFLUENCEConfig(value) {
    if (!('spaces' in value) || value['spaces'] === undefined)
        return false;
    return true;
}
function CONFLUENCEConfigFromJSON(json) {
    return CONFLUENCEConfigFromJSONTyped(json, false);
}
function CONFLUENCEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'spaces': json['spaces'],
        'rootParents': json['root-parents'] == null ? undefined : json['root-parents'],
    };
}
function CONFLUENCEConfigToJSON(json) {
    return CONFLUENCEConfigToJSONTyped(json, false);
}
function CONFLUENCEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'spaces': value['spaces'],
        'root-parents': value['rootParents'],
    };
}
