// @vectorize-io/vectorize-client/dist/models/CONFLUENCEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCONFLUENCEAuthConfig = instanceOfCONFLUENCEAuthConfig;
exports.CONFLUENCEAuthConfigFromJSON = CONFLUENCEAuthConfigFromJSON;
exports.CONFLUENCEAuthConfigFromJSONTyped = CONFLUENCEAuthConfigFromJSONTyped;
exports.CONFLUENCEAuthConfigToJSON = CONFLUENCEAuthConfigToJSON;
exports.CONFLUENCEAuthConfigToJSONTyped = CONFLUENCEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the CONFLUENCEAuthConfig interface.
 */
function instanceOfCONFLUENCEAuthConfig(value) {
    if (!('username' in value) || value['username'] === undefined)
        return false;
    if (!('apiToken' in value) || value['apiToken'] === undefined)
        return false;
    if (!('domain' in value) || value['domain'] === undefined)
        return false;
    return true;
}
function CONFLUENCEAuthConfigFromJSON(json) {
    return CONFLUENCEAuthConfigFromJSONTyped(json, false);
}
function CONFLUENCEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'username': json['username'],
        'apiToken': json['api-token'],
        'domain': json['domain'],
    };
}
function CONFLUENCEAuthConfigToJSON(json) {
    return CONFLUENCEAuthConfigToJSONTyped(json, false);
}
function CONFLUENCEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'username': value['username'],
        'api-token': value['apiToken'],
        'domain': value['domain'],
    };
}
