// @vectorize-io/vectorize-client/dist/models/Confluence.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfluenceTypeEnum = void 0;
exports.instanceOfConfluence = instanceOfConfluence;
exports.ConfluenceFromJSON = ConfluenceFromJSON;
exports.ConfluenceFromJSONTyped = ConfluenceFromJSONTyped;
exports.ConfluenceToJSON = ConfluenceToJSON;
exports.ConfluenceToJSONTyped = ConfluenceToJSONTyped;
var CONFLUENCEAuthConfig_1 = require("./CONFLUENCEAuthConfig");
/**
 * @export
 */
exports.ConfluenceTypeEnum = {
    Confluence: 'CONFLUENCE'
};
/**
 * Check if a given object implements the Confluence interface.
 */
function instanceOfConfluence(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function ConfluenceFromJSON(json) {
    return ConfluenceFromJSONTyped(json, false);
}
function ConfluenceFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, CONFLUENCEAuthConfig_1.CONFLUENCEAuthConfigFromJSON)(json['config']),
    };
}
function ConfluenceToJSON(json) {
    return ConfluenceToJSONTyped(json, false);
}
function ConfluenceToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, CONFLUENCEAuthConfig_1.CONFLUENCEAuthConfigToJSON)(value['config']),
    };
}
