// @vectorize-io/vectorize-client/dist/models/CreateAIPlatformConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateAIPlatformConnectorResponse = instanceOfCreateAIPlatformConnectorResponse;
exports.CreateAIPlatformConnectorResponseFromJSON = CreateAIPlatformConnectorResponseFromJSON;
exports.CreateAIPlatformConnectorResponseFromJSONTyped = CreateAIPlatformConnectorResponseFromJSONTyped;
exports.CreateAIPlatformConnectorResponseToJSON = CreateAIPlatformConnectorResponseToJSON;
exports.CreateAIPlatformConnectorResponseToJSONTyped = CreateAIPlatformConnectorResponseToJSONTyped;
var CreatedAIPlatformConnector_1 = require("./CreatedAIPlatformConnector");
/**
 * Check if a given object implements the CreateAIPlatformConnectorResponse interface.
 */
function instanceOfCreateAIPlatformConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('connector' in value) || value['connector'] === undefined)
        return false;
    return true;
}
function CreateAIPlatformConnectorResponseFromJSON(json) {
    return CreateAIPlatformConnectorResponseFromJSONTyped(json, false);
}
function CreateAIPlatformConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'connector': (0, CreatedAIPlatformConnector_1.CreatedAIPlatformConnectorFromJSON)(json['connector']),
    };
}
function CreateAIPlatformConnectorResponseToJSON(json) {
    return CreateAIPlatformConnectorResponseToJSONTyped(json, false);
}
function CreateAIPlatformConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'connector': (0, CreatedAIPlatformConnector_1.CreatedAIPlatformConnectorToJSON)(value['connector']),
    };
}
