// @vectorize-io/vectorize-client/dist/models/CreatedAIPlatformConnector.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatedAIPlatformConnector = instanceOfCreatedAIPlatformConnector;
exports.CreatedAIPlatformConnectorFromJSON = CreatedAIPlatformConnectorFromJSON;
exports.CreatedAIPlatformConnectorFromJSONTyped = CreatedAIPlatformConnectorFromJSONTyped;
exports.CreatedAIPlatformConnectorToJSON = CreatedAIPlatformConnectorToJSON;
exports.CreatedAIPlatformConnectorToJSONTyped = CreatedAIPlatformConnectorToJSONTyped;
/**
 * Check if a given object implements the CreatedAIPlatformConnector interface.
 */
function instanceOfCreatedAIPlatformConnector(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    return true;
}
function CreatedAIPlatformConnectorFromJSON(json) {
    return CreatedAIPlatformConnectorFromJSONTyped(json, false);
}
function CreatedAIPlatformConnectorFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'id': json['id'],
    };
}
function CreatedAIPlatformConnectorToJSON(json) {
    return CreatedAIPlatformConnectorToJSONTyped(json, false);
}
function CreatedAIPlatformConnectorToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'id': value['id'],
    };
}
