// @vectorize-io/vectorize-client/dist/models/CreatePipelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatePipelineResponse = instanceOfCreatePipelineResponse;
exports.CreatePipelineResponseFromJSON = CreatePipelineResponseFromJSON;
exports.CreatePipelineResponseFromJSONTyped = CreatePipelineResponseFromJSONTyped;
exports.CreatePipelineResponseToJSON = CreatePipelineResponseToJSON;
exports.CreatePipelineResponseToJSONTyped = CreatePipelineResponseToJSONTyped;
var CreatePipelineResponseData_1 = require("./CreatePipelineResponseData");
/**
 * Check if a given object implements the CreatePipelineResponse interface.
 */
function instanceOfCreatePipelineResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function CreatePipelineResponseFromJSON(json) {
    return CreatePipelineResponseFromJSONTyped(json, false);
}
function CreatePipelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (0, CreatePipelineResponseData_1.CreatePipelineResponseDataFromJSON)(json['data']),
    };
}
function CreatePipelineResponseToJSON(json) {
    return CreatePipelineResponseToJSONTyped(json, false);
}
function CreatePipelineResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (0, CreatePipelineResponseData_1.CreatePipelineResponseDataToJSON)(value['data']),
    };
}
