// @vectorize-io/vectorize-client/dist/models/CreateDestinationConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateDestinationConnectorResponse = instanceOfCreateDestinationConnectorResponse;
exports.CreateDestinationConnectorResponseFromJSON = CreateDestinationConnectorResponseFromJSON;
exports.CreateDestinationConnectorResponseFromJSONTyped = CreateDestinationConnectorResponseFromJSONTyped;
exports.CreateDestinationConnectorResponseToJSON = CreateDestinationConnectorResponseToJSON;
exports.CreateDestinationConnectorResponseToJSONTyped = CreateDestinationConnectorResponseToJSONTyped;
var CreatedDestinationConnector_1 = require("./CreatedDestinationConnector");
/**
 * Check if a given object implements the CreateDestinationConnectorResponse interface.
 */
function instanceOfCreateDestinationConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('connector' in value) || value['connector'] === undefined)
        return false;
    return true;
}
function CreateDestinationConnectorResponseFromJSON(json) {
    return CreateDestinationConnectorResponseFromJSONTyped(json, false);
}
function CreateDestinationConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'connector': (0, CreatedDestinationConnector_1.CreatedDestinationConnectorFromJSON)(json['connector']),
    };
}
function CreateDestinationConnectorResponseToJSON(json) {
    return CreateDestinationConnectorResponseToJSONTyped(json, false);
}
function CreateDestinationConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'connector': (0, CreatedDestinationConnector_1.CreatedDestinationConnectorToJSON)(value['connector']),
    };
}
