// @vectorize-io/vectorize-client/dist/models/CreateSourceConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSourceConnectorRequestFromJSON = CreateSourceConnectorRequestFromJSON;
exports.CreateSourceConnectorRequestFromJSONTyped = CreateSourceConnectorRequestFromJSONTyped;
exports.CreateSourceConnectorRequestToJSON = CreateSourceConnectorRequestToJSON;
exports.CreateSourceConnectorRequestToJSONTyped = CreateSourceConnectorRequestToJSONTyped;
function CreateSourceConnectorRequestFromJSON(json) {
    return CreateSourceConnectorRequestFromJSONTyped(json, false);
}
function CreateSourceConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    switch (json['type']) {
        default:
            throw new Error("No variant of CreateSourceConnectorRequest exists with 'type=".concat(json['type'], "'"));
    }
}
function CreateSourceConnectorRequestToJSON(json) {
    return CreateSourceConnectorRequestToJSONTyped(json, false);
}
function CreateSourceConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    switch (value['type']) {
        default:
            throw new Error("No variant of CreateSourceConnectorRequest exists with 'type=".concat(value['type'], "'"));
    }
}
