// @vectorize-io/vectorize-client/dist/models/CreateDestinationConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDestinationConnectorRequestFromJSON = CreateDestinationConnectorRequestFromJSON;
exports.CreateDestinationConnectorRequestFromJSONTyped = CreateDestinationConnectorRequestFromJSONTyped;
exports.CreateDestinationConnectorRequestToJSON = CreateDestinationConnectorRequestToJSON;
exports.CreateDestinationConnectorRequestToJSONTyped = CreateDestinationConnectorRequestToJSONTyped;
function CreateDestinationConnectorRequestFromJSON(json) {
    return CreateDestinationConnectorRequestFromJSONTyped(json, false);
}
function CreateDestinationConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    switch (json['type']) {
        default:
            throw new Error("No variant of CreateDestinationConnectorRequest exists with 'type=".concat(json['type'], "'"));
    }
}
function CreateDestinationConnectorRequestToJSON(json) {
    return CreateDestinationConnectorRequestToJSONTyped(json, false);
}
function CreateDestinationConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    switch (value['type']) {
        default:
            throw new Error("No variant of CreateDestinationConnectorRequest exists with 'type=".concat(value['type'], "'"));
    }
}
