// @vectorize-io/vectorize-client/dist/models/CreatedDestinationConnector.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatedDestinationConnector = instanceOfCreatedDestinationConnector;
exports.CreatedDestinationConnectorFromJSON = CreatedDestinationConnectorFromJSON;
exports.CreatedDestinationConnectorFromJSONTyped = CreatedDestinationConnectorFromJSONTyped;
exports.CreatedDestinationConnectorToJSON = CreatedDestinationConnectorToJSON;
exports.CreatedDestinationConnectorToJSONTyped = CreatedDestinationConnectorToJSONTyped;
/**
 * Check if a given object implements the CreatedDestinationConnector interface.
 */
function instanceOfCreatedDestinationConnector(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    return true;
}
function CreatedDestinationConnectorFromJSON(json) {
    return CreatedDestinationConnectorFromJSONTyped(json, false);
}
function CreatedDestinationConnectorFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'id': json['id'],
    };
}
function CreatedDestinationConnectorToJSON(json) {
    return CreatedDestinationConnectorToJSONTyped(json, false);
}
function CreatedDestinationConnectorToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'id': value['id'],
    };
}
