// @vectorize-io/vectorize-client/dist/models/CreateSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateSourceConnectorResponse = instanceOfCreateSourceConnectorResponse;
exports.CreateSourceConnectorResponseFromJSON = CreateSourceConnectorResponseFromJSON;
exports.CreateSourceConnectorResponseFromJSONTyped = CreateSourceConnectorResponseFromJSONTyped;
exports.CreateSourceConnectorResponseToJSON = CreateSourceConnectorResponseToJSON;
exports.CreateSourceConnectorResponseToJSONTyped = CreateSourceConnectorResponseToJSONTyped;
var CreatedSourceConnector_1 = require("./CreatedSourceConnector");
/**
 * Check if a given object implements the CreateSourceConnectorResponse interface.
 */
function instanceOfCreateSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('connector' in value) || value['connector'] === undefined)
        return false;
    return true;
}
function CreateSourceConnectorResponseFromJSON(json) {
    return CreateSourceConnectorResponseFromJSONTyped(json, false);
}
function CreateSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'connector': (0, CreatedSourceConnector_1.CreatedSourceConnectorFromJSON)(json['connector']),
    };
}
function CreateSourceConnectorResponseToJSON(json) {
    return CreateSourceConnectorResponseToJSONTyped(json, false);
}
function CreateSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'connector': (0, CreatedSourceConnector_1.CreatedSourceConnectorToJSON)(value['connector']),
    };
}
