// @vectorize-io/vectorize-client/dist/models/CreateAIPlatformConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAIPlatformConnectorRequestFromJSON = CreateAIPlatformConnectorRequestFromJSON;
exports.CreateAIPlatformConnectorRequestFromJSONTyped = CreateAIPlatformConnectorRequestFromJSONTyped;
exports.CreateAIPlatformConnectorRequestToJSON = CreateAIPlatformConnectorRequestToJSON;
exports.CreateAIPlatformConnectorRequestToJSONTyped = CreateAIPlatformConnectorRequestToJSONTyped;
function CreateAIPlatformConnectorRequestFromJSON(json) {
    return CreateAIPlatformConnectorRequestFromJSONTyped(json, false);
}
function CreateAIPlatformConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    switch (json['type']) {
        default:
            throw new Error("No variant of CreateAIPlatformConnectorRequest exists with 'type=".concat(json['type'], "'"));
    }
}
function CreateAIPlatformConnectorRequestToJSON(json) {
    return CreateAIPlatformConnectorRequestToJSONTyped(json, false);
}
function CreateAIPlatformConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    switch (value['type']) {
        default:
            throw new Error("No variant of CreateAIPlatformConnectorRequest exists with 'type=".concat(value['type'], "'"));
    }
}
