// @vectorize-io/vectorize-client/dist/models/CreatedSourceConnector.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatedSourceConnector = instanceOfCreatedSourceConnector;
exports.CreatedSourceConnectorFromJSON = CreatedSourceConnectorFromJSON;
exports.CreatedSourceConnectorFromJSONTyped = CreatedSourceConnectorFromJSONTyped;
exports.CreatedSourceConnectorToJSON = CreatedSourceConnectorToJSON;
exports.CreatedSourceConnectorToJSONTyped = CreatedSourceConnectorToJSONTyped;
/**
 * Check if a given object implements the CreatedSourceConnector interface.
 */
function instanceOfCreatedSourceConnector(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('id' in value) || value['id'] === undefined)
        return false;
    return true;
}
function CreatedSourceConnectorFromJSON(json) {
    return CreatedSourceConnectorFromJSONTyped(json, false);
}
function CreatedSourceConnectorFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'id': json['id'],
    };
}
function CreatedSourceConnectorToJSON(json) {
    return CreatedSourceConnectorToJSONTyped(json, false);
}
function CreatedSourceConnectorToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'id': value['id'],
    };
}
