// @vectorize-io/vectorize-client/dist/models/CreatePipelineResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatePipelineResponseData = instanceOfCreatePipelineResponseData;
exports.CreatePipelineResponseDataFromJSON = CreatePipelineResponseDataFromJSON;
exports.CreatePipelineResponseDataFromJSONTyped = CreatePipelineResponseDataFromJSONTyped;
exports.CreatePipelineResponseDataToJSON = CreatePipelineResponseDataToJSON;
exports.CreatePipelineResponseDataToJSONTyped = CreatePipelineResponseDataToJSONTyped;
/**
 * Check if a given object implements the CreatePipelineResponseData interface.
 */
function instanceOfCreatePipelineResponseData(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    return true;
}
function CreatePipelineResponseDataFromJSON(json) {
    return CreatePipelineResponseDataFromJSONTyped(json, false);
}
function CreatePipelineResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
    };
}
function CreatePipelineResponseDataToJSON(json) {
    return CreatePipelineResponseDataToJSONTyped(json, false);
}
function CreatePipelineResponseDataToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
    };
}
