// @vectorize-io/vectorize-client/dist/models/DATASTAXAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDATASTAXAuthConfig = instanceOfDATASTAXAuthConfig;
exports.DATASTAXAuthConfigFromJSON = DATASTAXAuthConfigFromJSON;
exports.DATASTAXAuthConfigFromJSONTyped = DATASTAXAuthConfigFromJSONTyped;
exports.DATASTAXAuthConfigToJSON = DATASTAXAuthConfigToJSON;
exports.DATASTAXAuthConfigToJSONTyped = DATASTAXAuthConfigToJSONTyped;
/**
 * Check if a given object implements the DATASTAXAuthConfig interface.
 */
function instanceOfDATASTAXAuthConfig(value) {
    if (!('endpointSecret' in value) || value['endpointSecret'] === undefined)
        return false;
    if (!('token' in value) || value['token'] === undefined)
        return false;
    return true;
}
function DATASTAXAuthConfigFromJSON(json) {
    return DATASTAXAuthConfigFromJSONTyped(json, false);
}
function DATASTAXAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'endpointSecret': json['endpoint_secret'],
        'token': json['token'],
    };
}
function DATASTAXAuthConfigToJSON(json) {
    return DATASTAXAuthConfigToJSONTyped(json, false);
}
function DATASTAXAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'endpoint_secret': value['endpointSecret'],
        'token': value['token'],
    };
}
