// @vectorize-io/vectorize-client/dist/models/DATASTAXConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDATASTAXConfig = instanceOfDATASTAXConfig;
exports.DATASTAXConfigFromJSON = DATASTAXConfigFromJSON;
exports.DATASTAXConfigFromJSONTyped = DATASTAXConfigFromJSONTyped;
exports.DATASTAXConfigToJSON = DATASTAXConfigToJSON;
exports.DATASTAXConfigToJSONTyped = DATASTAXConfigToJSONTyped;
/**
 * Check if a given object implements the DATASTAXConfig interface.
 */
function instanceOfDATASTAXConfig(value) {
    if (!('collection' in value) || value['collection'] === undefined)
        return false;
    return true;
}
function DATASTAXConfigFromJSON(json) {
    return DATASTAXConfigFromJSONTyped(json, false);
}
function DATASTAXConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'collection': json['collection'],
    };
}
function DATASTAXConfigToJSON(json) {
    return DATASTAXConfigToJSONTyped(json, false);
}
function DATASTAXConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'collection': value['collection'],
    };
}
