// @vectorize-io/vectorize-client/dist/models/Datastax1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDatastax1 = instanceOfDatastax1;
exports.Datastax1FromJSON = Datastax1FromJSON;
exports.Datastax1FromJSONTyped = Datastax1FromJSONTyped;
exports.Datastax1ToJSON = Datastax1ToJSON;
exports.Datastax1ToJSONTyped = Datastax1ToJSONTyped;
var DATASTAXAuthConfig_1 = require("./DATASTAXAuthConfig");
/**
 * Check if a given object implements the Datastax1 interface.
 */
function instanceOfDatastax1(value) {
    return true;
}
function Datastax1FromJSON(json) {
    return Datastax1FromJSONTyped(json, false);
}
function Datastax1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, DATASTAXAuthConfig_1.DATASTAXAuthConfigFromJSON)(json['config']),
    };
}
function Datastax1ToJSON(json) {
    return Datastax1ToJSONTyped(json, false);
}
function Datastax1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, DATASTAXAuthConfig_1.DATASTAXAuthConfigToJSON)(value['config']),
    };
}
