// @vectorize-io/vectorize-client/dist/models/Datastax.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatastaxTypeEnum = void 0;
exports.instanceOfDatastax = instanceOfDatastax;
exports.DatastaxFromJSON = DatastaxFromJSON;
exports.DatastaxFromJSONTyped = DatastaxFromJSONTyped;
exports.DatastaxToJSON = DatastaxToJSON;
exports.DatastaxToJSONTyped = DatastaxToJSONTyped;
var DATASTAXAuthConfig_1 = require("./DATASTAXAuthConfig");
/**
 * @export
 */
exports.DatastaxTypeEnum = {
    Datastax: 'DATASTAX'
};
/**
 * Check if a given object implements the Datastax interface.
 */
function instanceOfDatastax(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function DatastaxFromJSON(json) {
    return DatastaxFromJSONTyped(json, false);
}
function DatastaxFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, DATASTAXAuthConfig_1.DATASTAXAuthConfigFromJSON)(json['config']),
    };
}
function DatastaxToJSON(json) {
    return DatastaxToJSONTyped(json, false);
}
function DatastaxToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, DATASTAXAuthConfig_1.DATASTAXAuthConfigToJSON)(value['config']),
    };
}
