// @vectorize-io/vectorize-client/dist/models/DeepResearchResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeepResearchResult = instanceOfDeepResearchResult;
exports.DeepResearchResultFromJSON = DeepResearchResultFromJSON;
exports.DeepResearchResultFromJSONTyped = DeepResearchResultFromJSONTyped;
exports.DeepResearchResultToJSON = DeepResearchResultToJSON;
exports.DeepResearchResultToJSONTyped = DeepResearchResultToJSONTyped;
/**
 * Check if a given object implements the DeepResearchResult interface.
 */
function instanceOfDeepResearchResult(value) {
    if (!('success' in value) || value['success'] === undefined)
        return false;
    return true;
}
function DeepResearchResultFromJSON(json) {
    return DeepResearchResultFromJSONTyped(json, false);
}
function DeepResearchResultFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'success': json['success'],
        'events': json['events'] == null ? undefined : json['events'],
        'markdown': json['markdown'] == null ? undefined : json['markdown'],
        'error': json['error'] == null ? undefined : json['error'],
    };
}
function DeepResearchResultToJSON(json) {
    return DeepResearchResultToJSONTyped(json, false);
}
function DeepResearchResultToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'success': value['success'],
        'events': value['events'],
        'markdown': value['markdown'],
        'error': value['error'],
    };
}
