// @vectorize-io/vectorize-client/dist/models/DeletePipelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeletePipelineResponse = instanceOfDeletePipelineResponse;
exports.DeletePipelineResponseFromJSON = DeletePipelineResponseFromJSON;
exports.DeletePipelineResponseFromJSONTyped = DeletePipelineResponseFromJSONTyped;
exports.DeletePipelineResponseToJSON = DeletePipelineResponseToJSON;
exports.DeletePipelineResponseToJSONTyped = DeletePipelineResponseToJSONTyped;
/**
 * Check if a given object implements the DeletePipelineResponse interface.
 */
function instanceOfDeletePipelineResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function DeletePipelineResponseFromJSON(json) {
    return DeletePipelineResponseFromJSONTyped(json, false);
}
function DeletePipelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function DeletePipelineResponseToJSON(json) {
    return DeletePipelineResponseToJSONTyped(json, false);
}
function DeletePipelineResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
