// @vectorize-io/vectorize-client/dist/models/DeleteDestinationConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteDestinationConnectorResponse = instanceOfDeleteDestinationConnectorResponse;
exports.DeleteDestinationConnectorResponseFromJSON = DeleteDestinationConnectorResponseFromJSON;
exports.DeleteDestinationConnectorResponseFromJSONTyped = DeleteDestinationConnectorResponseFromJSONTyped;
exports.DeleteDestinationConnectorResponseToJSON = DeleteDestinationConnectorResponseToJSON;
exports.DeleteDestinationConnectorResponseToJSONTyped = DeleteDestinationConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteDestinationConnectorResponse interface.
 */
function instanceOfDeleteDestinationConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function DeleteDestinationConnectorResponseFromJSON(json) {
    return DeleteDestinationConnectorResponseFromJSONTyped(json, false);
}
function DeleteDestinationConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function DeleteDestinationConnectorResponseToJSON(json) {
    return DeleteDestinationConnectorResponseToJSONTyped(json, false);
}
function DeleteDestinationConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
