// @vectorize-io/vectorize-client/dist/models/DeleteFileResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteFileResponse = instanceOfDeleteFileResponse;
exports.DeleteFileResponseFromJSON = DeleteFileResponseFromJSON;
exports.DeleteFileResponseFromJSONTyped = DeleteFileResponseFromJSONTyped;
exports.DeleteFileResponseToJSON = DeleteFileResponseToJSON;
exports.DeleteFileResponseToJSONTyped = DeleteFileResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteFileResponse interface.
 */
function instanceOfDeleteFileResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('fileName' in value) || value['fileName'] === undefined)
        return false;
    return true;
}
function DeleteFileResponseFromJSON(json) {
    return DeleteFileResponseFromJSONTyped(json, false);
}
function DeleteFileResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'fileName': json['fileName'],
    };
}
function DeleteFileResponseToJSON(json) {
    return DeleteFileResponseToJSONTyped(json, false);
}
function DeleteFileResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'fileName': value['fileName'],
    };
}
