// @vectorize-io/vectorize-client/dist/models/DeleteAIPlatformConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteAIPlatformConnectorResponse = instanceOfDeleteAIPlatformConnectorResponse;
exports.DeleteAIPlatformConnectorResponseFromJSON = DeleteAIPlatformConnectorResponseFromJSON;
exports.DeleteAIPlatformConnectorResponseFromJSONTyped = DeleteAIPlatformConnectorResponseFromJSONTyped;
exports.DeleteAIPlatformConnectorResponseToJSON = DeleteAIPlatformConnectorResponseToJSON;
exports.DeleteAIPlatformConnectorResponseToJSONTyped = DeleteAIPlatformConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteAIPlatformConnectorResponse interface.
 */
function instanceOfDeleteAIPlatformConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function DeleteAIPlatformConnectorResponseFromJSON(json) {
    return DeleteAIPlatformConnectorResponseFromJSONTyped(json, false);
}
function DeleteAIPlatformConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function DeleteAIPlatformConnectorResponseToJSON(json) {
    return DeleteAIPlatformConnectorResponseToJSONTyped(json, false);
}
function DeleteAIPlatformConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
