// @vectorize-io/vectorize-client/dist/models/DeleteSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteSourceConnectorResponse = instanceOfDeleteSourceConnectorResponse;
exports.DeleteSourceConnectorResponseFromJSON = DeleteSourceConnectorResponseFromJSON;
exports.DeleteSourceConnectorResponseFromJSONTyped = DeleteSourceConnectorResponseFromJSONTyped;
exports.DeleteSourceConnectorResponseToJSON = DeleteSourceConnectorResponseToJSON;
exports.DeleteSourceConnectorResponseToJSONTyped = DeleteSourceConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the DeleteSourceConnectorResponse interface.
 */
function instanceOfDeleteSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function DeleteSourceConnectorResponseFromJSON(json) {
    return DeleteSourceConnectorResponseFromJSONTyped(json, false);
}
function DeleteSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function DeleteSourceConnectorResponseToJSON(json) {
    return DeleteSourceConnectorResponseToJSONTyped(json, false);
}
function DeleteSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
