// @vectorize-io/vectorize-client/dist/models/DestinationConnectorTypeForPipeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DestinationConnectorTypeForPipeline = void 0;
exports.instanceOfDestinationConnectorTypeForPipeline = instanceOfDestinationConnectorTypeForPipeline;
exports.DestinationConnectorTypeForPipelineFromJSON = DestinationConnectorTypeForPipelineFromJSON;
exports.DestinationConnectorTypeForPipelineFromJSONTyped = DestinationConnectorTypeForPipelineFromJSONTyped;
exports.DestinationConnectorTypeForPipelineToJSON = DestinationConnectorTypeForPipelineToJSON;
exports.DestinationConnectorTypeForPipelineToJSONTyped = DestinationConnectorTypeForPipelineToJSONTyped;
/**
 *
 * @export
 */
exports.DestinationConnectorTypeForPipeline = {
    Capella: 'CAPELLA',
    Datastax: 'DATASTAX',
    Elastic: 'ELASTIC',
    Pinecone: 'PINECONE',
    Singlestore: 'SINGLESTORE',
    Milvus: 'MILVUS',
    Postgresql: 'POSTGRESQL',
    Qdrant: 'QDRANT',
    Supabase: 'SUPABASE',
    Weaviate: 'WEAVIATE',
    Azureaisearch: 'AZUREAISEARCH',
    Turbopuffer: 'TURBOPUFFER',
    Vectorize: 'VECTORIZE'
};
function instanceOfDestinationConnectorTypeForPipeline(value) {
    for (var key in exports.DestinationConnectorTypeForPipeline) {
        if (Object.prototype.hasOwnProperty.call(exports.DestinationConnectorTypeForPipeline, key)) {
            if (exports.DestinationConnectorTypeForPipeline[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function DestinationConnectorTypeForPipelineFromJSON(json) {
    return DestinationConnectorTypeForPipelineFromJSONTyped(json, false);
}
function DestinationConnectorTypeForPipelineFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function DestinationConnectorTypeForPipelineToJSON(value) {
    return value;
}
function DestinationConnectorTypeForPipelineToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
