// @vectorize-io/vectorize-client/dist/models/DestinationConnectorType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DestinationConnectorType = void 0;
exports.instanceOfDestinationConnectorType = instanceOfDestinationConnectorType;
exports.DestinationConnectorTypeFromJSON = DestinationConnectorTypeFromJSON;
exports.DestinationConnectorTypeFromJSONTyped = DestinationConnectorTypeFromJSONTyped;
exports.DestinationConnectorTypeToJSON = DestinationConnectorTypeToJSON;
exports.DestinationConnectorTypeToJSONTyped = DestinationConnectorTypeToJSONTyped;
/**
 *
 * @export
 */
exports.DestinationConnectorType = {
    Capella: 'CAPELLA',
    Datastax: 'DATASTAX',
    Elastic: 'ELASTIC',
    Pinecone: 'PINECONE',
    Singlestore: 'SINGLESTORE',
    Milvus: 'MILVUS',
    Postgresql: 'POSTGRESQL',
    Qdrant: 'QDRANT',
    Supabase: 'SUPABASE',
    Weaviate: 'WEAVIATE',
    Azureaisearch: 'AZUREAISEARCH',
    Turbopuffer: 'TURBOPUFFER'
};
function instanceOfDestinationConnectorType(value) {
    for (var key in exports.DestinationConnectorType) {
        if (Object.prototype.hasOwnProperty.call(exports.DestinationConnectorType, key)) {
            if (exports.DestinationConnectorType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function DestinationConnectorTypeFromJSON(json) {
    return DestinationConnectorTypeFromJSONTyped(json, false);
}
function DestinationConnectorTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function DestinationConnectorTypeToJSON(value) {
    return value;
}
function DestinationConnectorTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
