// @vectorize-io/vectorize-client/dist/models/DestinationConnector.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDestinationConnector = instanceOfDestinationConnector;
exports.DestinationConnectorFromJSON = DestinationConnectorFromJSON;
exports.DestinationConnectorFromJSONTyped = DestinationConnectorFromJSONTyped;
exports.DestinationConnectorToJSON = DestinationConnectorToJSON;
exports.DestinationConnectorToJSONTyped = DestinationConnectorToJSONTyped;
/**
 * Check if a given object implements the DestinationConnector interface.
 */
function instanceOfDestinationConnector(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    return true;
}
function DestinationConnectorFromJSON(json) {
    return DestinationConnectorFromJSONTyped(json, false);
}
function DestinationConnectorFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'type': json['type'],
        'name': json['name'],
        'configDoc': json['configDoc'] == null ? undefined : json['configDoc'],
        'createdAt': json['createdAt'] == null ? undefined : json['createdAt'],
        'createdById': json['createdById'] == null ? undefined : json['createdById'],
        'lastUpdatedById': json['lastUpdatedById'] == null ? undefined : json['lastUpdatedById'],
        'createdByEmail': json['createdByEmail'] == null ? undefined : json['createdByEmail'],
        'lastUpdatedByEmail': json['lastUpdatedByEmail'] == null ? undefined : json['lastUpdatedByEmail'],
        'errorMessage': json['errorMessage'] == null ? undefined : json['errorMessage'],
        'verificationStatus': json['verificationStatus'] == null ? undefined : json['verificationStatus'],
    };
}
function DestinationConnectorToJSON(json) {
    return DestinationConnectorToJSONTyped(json, false);
}
function DestinationConnectorToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'type': value['type'],
        'name': value['name'],
        'configDoc': value['configDoc'],
        'createdAt': value['createdAt'],
        'createdById': value['createdById'],
        'lastUpdatedById': value['lastUpdatedById'],
        'createdByEmail': value['createdByEmail'],
        'lastUpdatedByEmail': value['lastUpdatedByEmail'],
        'errorMessage': value['errorMessage'],
        'verificationStatus': value['verificationStatus'],
    };
}
