// @vectorize-io/vectorize-client/dist/models/DestinationConnectorInputConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DestinationConnectorInputConfigFromJSON = DestinationConnectorInputConfigFromJSON;
exports.DestinationConnectorInputConfigFromJSONTyped = DestinationConnectorInputConfigFromJSONTyped;
exports.DestinationConnectorInputConfigToJSON = DestinationConnectorInputConfigToJSON;
exports.DestinationConnectorInputConfigToJSONTyped = DestinationConnectorInputConfigToJSONTyped;
var AZUREAISEARCHConfig_1 = require("./AZUREAISEARCHConfig");
var CAPELLAConfig_1 = require("./CAPELLAConfig");
var DATASTAXConfig_1 = require("./DATASTAXConfig");
var ELASTICConfig_1 = require("./ELASTICConfig");
var MILVUSConfig_1 = require("./MILVUSConfig");
var PINECONEConfig_1 = require("./PINECONEConfig");
var POSTGRESQLConfig_1 = require("./POSTGRESQLConfig");
var QDRANTConfig_1 = require("./QDRANTConfig");
var SINGLESTOREConfig_1 = require("./SINGLESTOREConfig");
var SUPABASEConfig_1 = require("./SUPABASEConfig");
var TURBOPUFFERConfig_1 = require("./TURBOPUFFERConfig");
var WEAVIATEConfig_1 = require("./WEAVIATEConfig");
function DestinationConnectorInputConfigFromJSON(json) {
    return DestinationConnectorInputConfigFromJSONTyped(json, false);
}
function DestinationConnectorInputConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    if ((0, AZUREAISEARCHConfig_1.instanceOfAZUREAISEARCHConfig)(json)) {
        return (0, AZUREAISEARCHConfig_1.AZUREAISEARCHConfigFromJSONTyped)(json, true);
    }
    if ((0, CAPELLAConfig_1.instanceOfCAPELLAConfig)(json)) {
        return (0, CAPELLAConfig_1.CAPELLAConfigFromJSONTyped)(json, true);
    }
    if ((0, DATASTAXConfig_1.instanceOfDATASTAXConfig)(json)) {
        return (0, DATASTAXConfig_1.DATASTAXConfigFromJSONTyped)(json, true);
    }
    if ((0, ELASTICConfig_1.instanceOfELASTICConfig)(json)) {
        return (0, ELASTICConfig_1.ELASTICConfigFromJSONTyped)(json, true);
    }
    if ((0, MILVUSConfig_1.instanceOfMILVUSConfig)(json)) {
        return (0, MILVUSConfig_1.MILVUSConfigFromJSONTyped)(json, true);
    }
    if ((0, PINECONEConfig_1.instanceOfPINECONEConfig)(json)) {
        return (0, PINECONEConfig_1.PINECONEConfigFromJSONTyped)(json, true);
    }
    if ((0, POSTGRESQLConfig_1.instanceOfPOSTGRESQLConfig)(json)) {
        return (0, POSTGRESQLConfig_1.POSTGRESQLConfigFromJSONTyped)(json, true);
    }
    if ((0, QDRANTConfig_1.instanceOfQDRANTConfig)(json)) {
        return (0, QDRANTConfig_1.QDRANTConfigFromJSONTyped)(json, true);
    }
    if ((0, SINGLESTOREConfig_1.instanceOfSINGLESTOREConfig)(json)) {
        return (0, SINGLESTOREConfig_1.SINGLESTOREConfigFromJSONTyped)(json, true);
    }
    if ((0, SUPABASEConfig_1.instanceOfSUPABASEConfig)(json)) {
        return (0, SUPABASEConfig_1.SUPABASEConfigFromJSONTyped)(json, true);
    }
    if ((0, TURBOPUFFERConfig_1.instanceOfTURBOPUFFERConfig)(json)) {
        return (0, TURBOPUFFERConfig_1.TURBOPUFFERConfigFromJSONTyped)(json, true);
    }
    if ((0, WEAVIATEConfig_1.instanceOfWEAVIATEConfig)(json)) {
        return (0, WEAVIATEConfig_1.WEAVIATEConfigFromJSONTyped)(json, true);
    }
    return {};
}
function DestinationConnectorInputConfigToJSON(json) {
    return DestinationConnectorInputConfigToJSONTyped(json, false);
}
function DestinationConnectorInputConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    if ((0, AZUREAISEARCHConfig_1.instanceOfAZUREAISEARCHConfig)(value)) {
        return (0, AZUREAISEARCHConfig_1.AZUREAISEARCHConfigToJSON)(value);
    }
    if ((0, CAPELLAConfig_1.instanceOfCAPELLAConfig)(value)) {
        return (0, CAPELLAConfig_1.CAPELLAConfigToJSON)(value);
    }
    if ((0, DATASTAXConfig_1.instanceOfDATASTAXConfig)(value)) {
        return (0, DATASTAXConfig_1.DATASTAXConfigToJSON)(value);
    }
    if ((0, ELASTICConfig_1.instanceOfELASTICConfig)(value)) {
        return (0, ELASTICConfig_1.ELASTICConfigToJSON)(value);
    }
    if ((0, MILVUSConfig_1.instanceOfMILVUSConfig)(value)) {
        return (0, MILVUSConfig_1.MILVUSConfigToJSON)(value);
    }
    if ((0, PINECONEConfig_1.instanceOfPINECONEConfig)(value)) {
        return (0, PINECONEConfig_1.PINECONEConfigToJSON)(value);
    }
    if ((0, POSTGRESQLConfig_1.instanceOfPOSTGRESQLConfig)(value)) {
        return (0, POSTGRESQLConfig_1.POSTGRESQLConfigToJSON)(value);
    }
    if ((0, QDRANTConfig_1.instanceOfQDRANTConfig)(value)) {
        return (0, QDRANTConfig_1.QDRANTConfigToJSON)(value);
    }
    if ((0, SINGLESTOREConfig_1.instanceOfSINGLESTOREConfig)(value)) {
        return (0, SINGLESTOREConfig_1.SINGLESTOREConfigToJSON)(value);
    }
    if ((0, SUPABASEConfig_1.instanceOfSUPABASEConfig)(value)) {
        return (0, SUPABASEConfig_1.SUPABASEConfigToJSON)(value);
    }
    if ((0, TURBOPUFFERConfig_1.instanceOfTURBOPUFFERConfig)(value)) {
        return (0, TURBOPUFFERConfig_1.TURBOPUFFERConfigToJSON)(value);
    }
    if ((0, WEAVIATEConfig_1.instanceOfWEAVIATEConfig)(value)) {
        return (0, WEAVIATEConfig_1.WEAVIATEConfigToJSON)(value);
    }
    return {};
}
