// @vectorize-io/vectorize-client/dist/models/Discord1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDiscord1 = instanceOfDiscord1;
exports.Discord1FromJSON = Discord1FromJSON;
exports.Discord1FromJSONTyped = Discord1FromJSONTyped;
exports.Discord1ToJSON = Discord1ToJSON;
exports.Discord1ToJSONTyped = Discord1ToJSONTyped;
var DISCORDAuthConfig_1 = require("./DISCORDAuthConfig");
/**
 * Check if a given object implements the Discord1 interface.
 */
function instanceOfDiscord1(value) {
    return true;
}
function Discord1FromJSON(json) {
    return Discord1FromJSONTyped(json, false);
}
function Discord1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, DISCORDAuthConfig_1.DISCORDAuthConfigFromJSON)(json['config']),
    };
}
function Discord1ToJSON(json) {
    return Discord1ToJSONTyped(json, false);
}
function Discord1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, DISCORDAuthConfig_1.DISCORDAuthConfigToJSON)(value['config']),
    };
}
