// @vectorize-io/vectorize-client/dist/models/DISCORDAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDISCORDAuthConfig = instanceOfDISCORDAuthConfig;
exports.DISCORDAuthConfigFromJSON = DISCORDAuthConfigFromJSON;
exports.DISCORDAuthConfigFromJSONTyped = DISCORDAuthConfigFromJSONTyped;
exports.DISCORDAuthConfigToJSON = DISCORDAuthConfigToJSON;
exports.DISCORDAuthConfigToJSONTyped = DISCORDAuthConfigToJSONTyped;
/**
 * Check if a given object implements the DISCORDAuthConfig interface.
 */
function instanceOfDISCORDAuthConfig(value) {
    if (!('serverId' in value) || value['serverId'] === undefined)
        return false;
    if (!('botToken' in value) || value['botToken'] === undefined)
        return false;
    if (!('channelIds' in value) || value['channelIds'] === undefined)
        return false;
    return true;
}
function DISCORDAuthConfigFromJSON(json) {
    return DISCORDAuthConfigFromJSONTyped(json, false);
}
function DISCORDAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'serverId': json['server-id'],
        'botToken': json['bot-token'],
        'channelIds': json['channel-ids'],
    };
}
function DISCORDAuthConfigToJSON(json) {
    return DISCORDAuthConfigToJSONTyped(json, false);
}
function DISCORDAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'server-id': value['serverId'],
        'bot-token': value['botToken'],
        'channel-ids': value['channelIds'],
    };
}
