// @vectorize-io/vectorize-client/dist/models/Discord.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscordTypeEnum = void 0;
exports.instanceOfDiscord = instanceOfDiscord;
exports.DiscordFromJSON = DiscordFromJSON;
exports.DiscordFromJSONTyped = DiscordFromJSONTyped;
exports.DiscordToJSON = DiscordToJSON;
exports.DiscordToJSONTyped = DiscordToJSONTyped;
var DISCORDAuthConfig_1 = require("./DISCORDAuthConfig");
/**
 * @export
 */
exports.DiscordTypeEnum = {
    Discord: 'DISCORD'
};
/**
 * Check if a given object implements the Discord interface.
 */
function instanceOfDiscord(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function DiscordFromJSON(json) {
    return DiscordFromJSONTyped(json, false);
}
function DiscordFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, DISCORDAuthConfig_1.DISCORDAuthConfigFromJSON)(json['config']),
    };
}
function DiscordToJSON(json) {
    return DiscordToJSONTyped(json, false);
}
function DiscordToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, DISCORDAuthConfig_1.DISCORDAuthConfigToJSON)(value['config']),
    };
}
