// @vectorize-io/vectorize-client/dist/models/DOCUSIGNConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DOCUSIGNConfigEnvelopeStatusesEnum = void 0;
exports.instanceOfDOCUSIGNConfig = instanceOfDOCUSIGNConfig;
exports.DOCUSIGNConfigFromJSON = DOCUSIGNConfigFromJSON;
exports.DOCUSIGNConfigFromJSONTyped = DOCUSIGNConfigFromJSONTyped;
exports.DOCUSIGNConfigToJSON = DOCUSIGNConfigToJSON;
exports.DOCUSIGNConfigToJSONTyped = DOCUSIGNConfigToJSONTyped;
/**
 * @export
 */
exports.DOCUSIGNConfigEnvelopeStatusesEnum = {
    Completed: 'completed',
    Correct: 'correct',
    Created: 'created',
    Declined: 'declined',
    Delivered: 'delivered',
    Sent: 'sent',
    Signed: 'signed',
    Voided: 'voided',
    All: 'all'
};
/**
 * Check if a given object implements the DOCUSIGNConfig interface.
 */
function instanceOfDOCUSIGNConfig(value) {
    if (!('fromDate' in value) || value['fromDate'] === undefined)
        return false;
    return true;
}
function DOCUSIGNConfigFromJSON(json) {
    return DOCUSIGNConfigFromJSONTyped(json, false);
}
function DOCUSIGNConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'envelopeStatuses': json['envelope-statuses'] == null ? undefined : json['envelope-statuses'],
        'fromDate': (new Date(json['from-date'])),
        'toDate': json['to-date'] == null ? undefined : (new Date(json['to-date'])),
        'maxDocuments': json['max-documents'] == null ? undefined : json['max-documents'],
        'searchText': json['search-text'] == null ? undefined : json['search-text'],
    };
}
function DOCUSIGNConfigToJSON(json) {
    return DOCUSIGNConfigToJSONTyped(json, false);
}
function DOCUSIGNConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'envelope-statuses': value['envelopeStatuses'],
        'from-date': ((value['fromDate']).toISOString().substring(0, 10)),
        'to-date': value['toDate'] == null ? undefined : ((value['toDate']).toISOString().substring(0, 10)),
        'max-documents': value['maxDocuments'],
        'search-text': value['searchText'],
    };
}
