// @vectorize-io/vectorize-client/dist/models/DROPBOXConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDROPBOXConfig = instanceOfDROPBOXConfig;
exports.DROPBOXConfigFromJSON = DROPBOXConfigFromJSON;
exports.DROPBOXConfigFromJSONTyped = DROPBOXConfigFromJSONTyped;
exports.DROPBOXConfigToJSON = DROPBOXConfigToJSON;
exports.DROPBOXConfigToJSONTyped = DROPBOXConfigToJSONTyped;
/**
 * Check if a given object implements the DROPBOXConfig interface.
 */
function instanceOfDROPBOXConfig(value) {
    return true;
}
function DROPBOXConfigFromJSON(json) {
    return DROPBOXConfigFromJSONTyped(json, false);
}
function DROPBOXConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'pathPrefix': json['path-prefix'] == null ? undefined : json['path-prefix'],
    };
}
function DROPBOXConfigToJSON(json) {
    return DROPBOXConfigToJSONTyped(json, false);
}
function DROPBOXConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'path-prefix': value['pathPrefix'],
    };
}
