// @vectorize-io/vectorize-client/dist/models/Elastic.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElasticTypeEnum = void 0;
exports.instanceOfElastic = instanceOfElastic;
exports.ElasticFromJSON = ElasticFromJSON;
exports.ElasticFromJSONTyped = ElasticFromJSONTyped;
exports.ElasticToJSON = ElasticToJSON;
exports.ElasticToJSONTyped = ElasticToJSONTyped;
var ELASTICAuthConfig_1 = require("./ELASTICAuthConfig");
/**
 * @export
 */
exports.ElasticTypeEnum = {
    Elastic: 'ELASTIC'
};
/**
 * Check if a given object implements the Elastic interface.
 */
function instanceOfElastic(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function ElasticFromJSON(json) {
    return ElasticFromJSONTyped(json, false);
}
function ElasticFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, ELASTICAuthConfig_1.ELASTICAuthConfigFromJSON)(json['config']),
    };
}
function ElasticToJSON(json) {
    return ElasticToJSONTyped(json, false);
}
function ElasticToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, ELASTICAuthConfig_1.ELASTICAuthConfigToJSON)(value['config']),
    };
}
