// @vectorize-io/vectorize-client/dist/models/ELASTICAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfELASTICAuthConfig = instanceOfELASTICAuthConfig;
exports.ELASTICAuthConfigFromJSON = ELASTICAuthConfigFromJSON;
exports.ELASTICAuthConfigFromJSONTyped = ELASTICAuthConfigFromJSONTyped;
exports.ELASTICAuthConfigToJSON = ELASTICAuthConfigToJSON;
exports.ELASTICAuthConfigToJSONTyped = ELASTICAuthConfigToJSONTyped;
/**
 * Check if a given object implements the ELASTICAuthConfig interface.
 */
function instanceOfELASTICAuthConfig(value) {
    if (!('host' in value) || value['host'] === undefined)
        return false;
    if (!('port' in value) || value['port'] === undefined)
        return false;
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function ELASTICAuthConfigFromJSON(json) {
    return ELASTICAuthConfigFromJSONTyped(json, false);
}
function ELASTICAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'host': json['host'],
        'port': json['port'],
        'apiKey': json['api-key'],
    };
}
function ELASTICAuthConfigToJSON(json) {
    return ELASTICAuthConfigToJSONTyped(json, false);
}
function ELASTICAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'host': value['host'],
        'port': value['port'],
        'api-key': value['apiKey'],
    };
}
