// @vectorize-io/vectorize-client/dist/models/Elastic1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfElastic1 = instanceOfElastic1;
exports.Elastic1FromJSON = Elastic1FromJSON;
exports.Elastic1FromJSONTyped = Elastic1FromJSONTyped;
exports.Elastic1ToJSON = Elastic1ToJSON;
exports.Elastic1ToJSONTyped = Elastic1ToJSONTyped;
var ELASTICAuthConfig_1 = require("./ELASTICAuthConfig");
/**
 * Check if a given object implements the Elastic1 interface.
 */
function instanceOfElastic1(value) {
    return true;
}
function Elastic1FromJSON(json) {
    return Elastic1FromJSONTyped(json, false);
}
function Elastic1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, ELASTICAuthConfig_1.ELASTICAuthConfigFromJSON)(json['config']),
    };
}
function Elastic1ToJSON(json) {
    return Elastic1ToJSONTyped(json, false);
}
function Elastic1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, ELASTICAuthConfig_1.ELASTICAuthConfigToJSON)(value['config']),
    };
}
