// @vectorize-io/vectorize-client/dist/models/ELASTICConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfELASTICConfig = instanceOfELASTICConfig;
exports.ELASTICConfigFromJSON = ELASTICConfigFromJSON;
exports.ELASTICConfigFromJSONTyped = ELASTICConfigFromJSONTyped;
exports.ELASTICConfigToJSON = ELASTICConfigToJSON;
exports.ELASTICConfigToJSONTyped = ELASTICConfigToJSONTyped;
/**
 * Check if a given object implements the ELASTICConfig interface.
 */
function instanceOfELASTICConfig(value) {
    if (!('index' in value) || value['index'] === undefined)
        return false;
    return true;
}
function ELASTICConfigFromJSON(json) {
    return ELASTICConfigFromJSONTyped(json, false);
}
function ELASTICConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'index': json['index'],
    };
}
function ELASTICConfigToJSON(json) {
    return ELASTICConfigToJSONTyped(json, false);
}
function ELASTICConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'index': value['index'],
    };
}
