// @vectorize-io/vectorize-client/dist/models/ExtractionType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtractionType = void 0;
exports.instanceOfExtractionType = instanceOfExtractionType;
exports.ExtractionTypeFromJSON = ExtractionTypeFromJSON;
exports.ExtractionTypeFromJSONTyped = ExtractionTypeFromJSONTyped;
exports.ExtractionTypeToJSON = ExtractionTypeToJSON;
exports.ExtractionTypeToJSONTyped = ExtractionTypeToJSONTyped;
/**
 *
 * @export
 */
exports.ExtractionType = {
    Iris: 'iris'
};
function instanceOfExtractionType(value) {
    for (var key in exports.ExtractionType) {
        if (Object.prototype.hasOwnProperty.call(exports.ExtractionType, key)) {
            if (exports.ExtractionType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function ExtractionTypeFromJSON(json) {
    return ExtractionTypeFromJSONTyped(json, false);
}
function ExtractionTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function ExtractionTypeToJSON(value) {
    return value;
}
function ExtractionTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
