// @vectorize-io/vectorize-client/dist/models/ExtractionResultResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfExtractionResultResponse = instanceOfExtractionResultResponse;
exports.ExtractionResultResponseFromJSON = ExtractionResultResponseFromJSON;
exports.ExtractionResultResponseFromJSONTyped = ExtractionResultResponseFromJSONTyped;
exports.ExtractionResultResponseToJSON = ExtractionResultResponseToJSON;
exports.ExtractionResultResponseToJSONTyped = ExtractionResultResponseToJSONTyped;
var ExtractionResult_1 = require("./ExtractionResult");
/**
 * Check if a given object implements the ExtractionResultResponse interface.
 */
function instanceOfExtractionResultResponse(value) {
    if (!('ready' in value) || value['ready'] === undefined)
        return false;
    return true;
}
function ExtractionResultResponseFromJSON(json) {
    return ExtractionResultResponseFromJSONTyped(json, false);
}
function ExtractionResultResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'ready': json['ready'],
        'data': json['data'] == null ? undefined : (0, ExtractionResult_1.ExtractionResultFromJSON)(json['data']),
    };
}
function ExtractionResultResponseToJSON(json) {
    return ExtractionResultResponseToJSONTyped(json, false);
}
function ExtractionResultResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'ready': value['ready'],
        'data': (0, ExtractionResult_1.ExtractionResultToJSON)(value['data']),
    };
}
