// @vectorize-io/vectorize-client/dist/models/ExtractionChunkingStrategy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtractionChunkingStrategy = void 0;
exports.instanceOfExtractionChunkingStrategy = instanceOfExtractionChunkingStrategy;
exports.ExtractionChunkingStrategyFromJSON = ExtractionChunkingStrategyFromJSON;
exports.ExtractionChunkingStrategyFromJSONTyped = ExtractionChunkingStrategyFromJSONTyped;
exports.ExtractionChunkingStrategyToJSON = ExtractionChunkingStrategyToJSON;
exports.ExtractionChunkingStrategyToJSONTyped = ExtractionChunkingStrategyToJSONTyped;
/**
 *
 * @export
 */
exports.ExtractionChunkingStrategy = {
    Markdown: 'markdown'
};
function instanceOfExtractionChunkingStrategy(value) {
    for (var key in exports.ExtractionChunkingStrategy) {
        if (Object.prototype.hasOwnProperty.call(exports.ExtractionChunkingStrategy, key)) {
            if (exports.ExtractionChunkingStrategy[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function ExtractionChunkingStrategyFromJSON(json) {
    return ExtractionChunkingStrategyFromJSONTyped(json, false);
}
function ExtractionChunkingStrategyFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function ExtractionChunkingStrategyToJSON(value) {
    return value;
}
function ExtractionChunkingStrategyToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
