// @vectorize-io/vectorize-client/dist/models/FileUpload1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFileUpload1 = instanceOfFileUpload1;
exports.FileUpload1FromJSON = FileUpload1FromJSON;
exports.FileUpload1FromJSONTyped = FileUpload1FromJSONTyped;
exports.FileUpload1ToJSON = FileUpload1ToJSON;
exports.FileUpload1ToJSONTyped = FileUpload1ToJSONTyped;
var FILEUPLOADAuthConfig_1 = require("./FILEUPLOADAuthConfig");
/**
 * Check if a given object implements the FileUpload1 interface.
 */
function instanceOfFileUpload1(value) {
    return true;
}
function FileUpload1FromJSON(json) {
    return FileUpload1FromJSONTyped(json, false);
}
function FileUpload1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, FILEUPLOADAuthConfig_1.FILEUPLOADAuthConfigFromJSON)(json['config']),
    };
}
function FileUpload1ToJSON(json) {
    return FileUpload1ToJSONTyped(json, false);
}
function FileUpload1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, FILEUPLOADAuthConfig_1.FILEUPLOADAuthConfigToJSON)(value['config']),
    };
}
