// @vectorize-io/vectorize-client/dist/models/FileUpload.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileUploadTypeEnum = void 0;
exports.instanceOfFileUpload = instanceOfFileUpload;
exports.FileUploadFromJSON = FileUploadFromJSON;
exports.FileUploadFromJSONTyped = FileUploadFromJSONTyped;
exports.FileUploadToJSON = FileUploadToJSON;
exports.FileUploadToJSONTyped = FileUploadToJSONTyped;
/**
 * @export
 */
exports.FileUploadTypeEnum = {
    FileUpload: 'FILE_UPLOAD'
};
/**
 * Check if a given object implements the FileUpload interface.
 */
function instanceOfFileUpload(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    return true;
}
function FileUploadFromJSON(json) {
    return FileUploadFromJSONTyped(json, false);
}
function FileUploadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
    };
}
function FileUploadToJSON(json) {
    return FileUploadToJSONTyped(json, false);
}
function FileUploadToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
    };
}
