// @vectorize-io/vectorize-client/dist/models/FILEUPLOADAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFILEUPLOADAuthConfig = instanceOfFILEUPLOADAuthConfig;
exports.FILEUPLOADAuthConfigFromJSON = FILEUPLOADAuthConfigFromJSON;
exports.FILEUPLOADAuthConfigFromJSONTyped = FILEUPLOADAuthConfigFromJSONTyped;
exports.FILEUPLOADAuthConfigToJSON = FILEUPLOADAuthConfigToJSON;
exports.FILEUPLOADAuthConfigToJSONTyped = FILEUPLOADAuthConfigToJSONTyped;
/**
 * Check if a given object implements the FILEUPLOADAuthConfig interface.
 */
function instanceOfFILEUPLOADAuthConfig(value) {
    return true;
}
function FILEUPLOADAuthConfigFromJSON(json) {
    return FILEUPLOADAuthConfigFromJSONTyped(json, false);
}
function FILEUPLOADAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'pathPrefix': json['path-prefix'] == null ? undefined : json['path-prefix'],
        'files': json['files'] == null ? undefined : json['files'],
    };
}
function FILEUPLOADAuthConfigToJSON(json) {
    return FILEUPLOADAuthConfigToJSONTyped(json, false);
}
function FILEUPLOADAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'path-prefix': value['pathPrefix'],
        'files': value['files'],
    };
}
