// @vectorize-io/vectorize-client/dist/models/Fireflies1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFireflies1 = instanceOfFireflies1;
exports.Fireflies1FromJSON = Fireflies1FromJSON;
exports.Fireflies1FromJSONTyped = Fireflies1FromJSONTyped;
exports.Fireflies1ToJSON = Fireflies1ToJSON;
exports.Fireflies1ToJSONTyped = Fireflies1ToJSONTyped;
var FIREFLIESAuthConfig_1 = require("./FIREFLIESAuthConfig");
/**
 * Check if a given object implements the Fireflies1 interface.
 */
function instanceOfFireflies1(value) {
    return true;
}
function Fireflies1FromJSON(json) {
    return Fireflies1FromJSONTyped(json, false);
}
function Fireflies1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, FIREFLIESAuthConfig_1.FIREFLIESAuthConfigFromJSON)(json['config']),
    };
}
function Fireflies1ToJSON(json) {
    return Fireflies1ToJSONTyped(json, false);
}
function Fireflies1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, FIREFLIESAuthConfig_1.FIREFLIESAuthConfigToJSON)(value['config']),
    };
}
