// @vectorize-io/vectorize-client/dist/models/FIRECRAWLAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFIRECRAWLAuthConfig = instanceOfFIRECRAWLAuthConfig;
exports.FIRECRAWLAuthConfigFromJSON = FIRECRAWLAuthConfigFromJSON;
exports.FIRECRAWLAuthConfigFromJSONTyped = FIRECRAWLAuthConfigFromJSONTyped;
exports.FIRECRAWLAuthConfigToJSON = FIRECRAWLAuthConfigToJSON;
exports.FIRECRAWLAuthConfigToJSONTyped = FIRECRAWLAuthConfigToJSONTyped;
/**
 * Check if a given object implements the FIRECRAWLAuthConfig interface.
 */
function instanceOfFIRECRAWLAuthConfig(value) {
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function FIRECRAWLAuthConfigFromJSON(json) {
    return FIRECRAWLAuthConfigFromJSONTyped(json, false);
}
function FIRECRAWLAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'apiKey': json['api-key'],
    };
}
function FIRECRAWLAuthConfigToJSON(json) {
    return FIRECRAWLAuthConfigToJSONTyped(json, false);
}
function FIRECRAWLAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'api-key': value['apiKey'],
    };
}
