// @vectorize-io/vectorize-client/dist/models/FIREFLIESAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFIREFLIESAuthConfig = instanceOfFIREFLIESAuthConfig;
exports.FIREFLIESAuthConfigFromJSON = FIREFLIESAuthConfigFromJSON;
exports.FIREFLIESAuthConfigFromJSONTyped = FIREFLIESAuthConfigFromJSONTyped;
exports.FIREFLIESAuthConfigToJSON = FIREFLIESAuthConfigToJSON;
exports.FIREFLIESAuthConfigToJSONTyped = FIREFLIESAuthConfigToJSONTyped;
/**
 * Check if a given object implements the FIREFLIESAuthConfig interface.
 */
function instanceOfFIREFLIESAuthConfig(value) {
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function FIREFLIESAuthConfigFromJSON(json) {
    return FIREFLIESAuthConfigFromJSONTyped(json, false);
}
function FIREFLIESAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'apiKey': json['api-key'],
    };
}
function FIREFLIESAuthConfigToJSON(json) {
    return FIREFLIESAuthConfigToJSONTyped(json, false);
}
function FIREFLIESAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'api-key': value['apiKey'],
    };
}
