// @vectorize-io/vectorize-client/dist/models/FIREFLIESConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFIREFLIESConfig = instanceOfFIREFLIESConfig;
exports.FIREFLIESConfigFromJSON = FIREFLIESConfigFromJSON;
exports.FIREFLIESConfigFromJSONTyped = FIREFLIESConfigFromJSONTyped;
exports.FIREFLIESConfigToJSON = FIREFLIESConfigToJSON;
exports.FIREFLIESConfigToJSONTyped = FIREFLIESConfigToJSONTyped;
/**
 * Check if a given object implements the FIREFLIESConfig interface.
 */
function instanceOfFIREFLIESConfig(value) {
    if (!('startDate' in value) || value['startDate'] === undefined)
        return false;
    if (!('participantFilterType' in value) || value['participantFilterType'] === undefined)
        return false;
    return true;
}
function FIREFLIESConfigFromJSON(json) {
    return FIREFLIESConfigFromJSONTyped(json, false);
}
function FIREFLIESConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'startDate': (new Date(json['start-date'])),
        'endDate': json['end-date'] == null ? undefined : (new Date(json['end-date'])),
        'titleFilter': json['title-filter'] == null ? undefined : json['title-filter'],
        'participantFilterType': json['participant-filter-type'],
        'participantFilter': json['participant-filter'] == null ? undefined : json['participant-filter'],
        'maxMeetings': json['max-meetings'] == null ? undefined : json['max-meetings'],
    };
}
function FIREFLIESConfigToJSON(json) {
    return FIREFLIESConfigToJSONTyped(json, false);
}
function FIREFLIESConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'start-date': ((value['startDate']).toISOString().substring(0, 10)),
        'end-date': value['endDate'] == null ? undefined : ((value['endDate']).toISOString().substring(0, 10)),
        'title-filter': value['titleFilter'],
        'participant-filter-type': value['participantFilterType'],
        'participant-filter': value['participantFilter'],
        'max-meetings': value['maxMeetings'],
    };
}
