// @vectorize-io/vectorize-client/dist/models/Firecrawl.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirecrawlTypeEnum = void 0;
exports.instanceOfFirecrawl = instanceOfFirecrawl;
exports.FirecrawlFromJSON = FirecrawlFromJSON;
exports.FirecrawlFromJSONTyped = FirecrawlFromJSONTyped;
exports.FirecrawlToJSON = FirecrawlToJSON;
exports.FirecrawlToJSONTyped = FirecrawlToJSONTyped;
var FIRECRAWLAuthConfig_1 = require("./FIRECRAWLAuthConfig");
/**
 * @export
 */
exports.FirecrawlTypeEnum = {
    Firecrawl: 'FIRECRAWL'
};
/**
 * Check if a given object implements the Firecrawl interface.
 */
function instanceOfFirecrawl(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function FirecrawlFromJSON(json) {
    return FirecrawlFromJSONTyped(json, false);
}
function FirecrawlFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, FIRECRAWLAuthConfig_1.FIRECRAWLAuthConfigFromJSON)(json['config']),
    };
}
function FirecrawlToJSON(json) {
    return FirecrawlToJSONTyped(json, false);
}
function FirecrawlToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, FIRECRAWLAuthConfig_1.FIRECRAWLAuthConfigToJSON)(value['config']),
    };
}
