// @vectorize-io/vectorize-client/dist/models/FIRECRAWLConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FIRECRAWLConfigEndpointEnum = void 0;
exports.instanceOfFIRECRAWLConfig = instanceOfFIRECRAWLConfig;
exports.FIRECRAWLConfigFromJSON = FIRECRAWLConfigFromJSON;
exports.FIRECRAWLConfigFromJSONTyped = FIRECRAWLConfigFromJSONTyped;
exports.FIRECRAWLConfigToJSON = FIRECRAWLConfigToJSON;
exports.FIRECRAWLConfigToJSONTyped = FIRECRAWLConfigToJSONTyped;
/**
 * @export
 */
exports.FIRECRAWLConfigEndpointEnum = {
    Crawl: 'Crawl',
    Scrape: 'Scrape'
};
/**
 * Check if a given object implements the FIRECRAWLConfig interface.
 */
function instanceOfFIRECRAWLConfig(value) {
    if (!('endpoint' in value) || value['endpoint'] === undefined)
        return false;
    if (!('request' in value) || value['request'] === undefined)
        return false;
    return true;
}
function FIRECRAWLConfigFromJSON(json) {
    return FIRECRAWLConfigFromJSONTyped(json, false);
}
function FIRECRAWLConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'endpoint': json['endpoint'],
        'request': json['request'],
    };
}
function FIRECRAWLConfigToJSON(json) {
    return FIRECRAWLConfigToJSONTyped(json, false);
}
function FIRECRAWLConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'endpoint': value['endpoint'],
        'request': value['request'],
    };
}
