// @vectorize-io/vectorize-client/dist/models/Fireflies.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirefliesTypeEnum = void 0;
exports.instanceOfFireflies = instanceOfFireflies;
exports.FirefliesFromJSON = FirefliesFromJSON;
exports.FirefliesFromJSONTyped = FirefliesFromJSONTyped;
exports.FirefliesToJSON = FirefliesToJSON;
exports.FirefliesToJSONTyped = FirefliesToJSONTyped;
var FIREFLIESAuthConfig_1 = require("./FIREFLIESAuthConfig");
/**
 * @export
 */
exports.FirefliesTypeEnum = {
    Fireflies: 'FIREFLIES'
};
/**
 * Check if a given object implements the Fireflies interface.
 */
function instanceOfFireflies(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function FirefliesFromJSON(json) {
    return FirefliesFromJSONTyped(json, false);
}
function FirefliesFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, FIREFLIESAuthConfig_1.FIREFLIESAuthConfigFromJSON)(json['config']),
    };
}
function FirefliesToJSON(json) {
    return FirefliesToJSONTyped(json, false);
}
function FirefliesToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, FIREFLIESAuthConfig_1.FIREFLIESAuthConfigToJSON)(value['config']),
    };
}
