// @vectorize-io/vectorize-client/dist/models/Firecrawl1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFirecrawl1 = instanceOfFirecrawl1;
exports.Firecrawl1FromJSON = Firecrawl1FromJSON;
exports.Firecrawl1FromJSONTyped = Firecrawl1FromJSONTyped;
exports.Firecrawl1ToJSON = Firecrawl1ToJSON;
exports.Firecrawl1ToJSONTyped = Firecrawl1ToJSONTyped;
var FIRECRAWLAuthConfig_1 = require("./FIRECRAWLAuthConfig");
/**
 * Check if a given object implements the Firecrawl1 interface.
 */
function instanceOfFirecrawl1(value) {
    return true;
}
function Firecrawl1FromJSON(json) {
    return Firecrawl1FromJSONTyped(json, false);
}
function Firecrawl1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, FIRECRAWLAuthConfig_1.FIRECRAWLAuthConfigFromJSON)(json['config']),
    };
}
function Firecrawl1ToJSON(json) {
    return Firecrawl1ToJSONTyped(json, false);
}
function Firecrawl1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, FIRECRAWLAuthConfig_1.FIRECRAWLAuthConfigToJSON)(value['config']),
    };
}
