// @vectorize-io/vectorize-client/dist/models/Gcs1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGcs1 = instanceOfGcs1;
exports.Gcs1FromJSON = Gcs1FromJSON;
exports.Gcs1FromJSONTyped = Gcs1FromJSONTyped;
exports.Gcs1ToJSON = Gcs1ToJSON;
exports.Gcs1ToJSONTyped = Gcs1ToJSONTyped;
var GCSAuthConfig_1 = require("./GCSAuthConfig");
/**
 * Check if a given object implements the Gcs1 interface.
 */
function instanceOfGcs1(value) {
    return true;
}
function Gcs1FromJSON(json) {
    return Gcs1FromJSONTyped(json, false);
}
function Gcs1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, GCSAuthConfig_1.GCSAuthConfigFromJSON)(json['config']),
    };
}
function Gcs1ToJSON(json) {
    return Gcs1ToJSONTyped(json, false);
}
function Gcs1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, GCSAuthConfig_1.GCSAuthConfigToJSON)(value['config']),
    };
}
