// @vectorize-io/vectorize-client/dist/models/GCSAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGCSAuthConfig = instanceOfGCSAuthConfig;
exports.GCSAuthConfigFromJSON = GCSAuthConfigFromJSON;
exports.GCSAuthConfigFromJSONTyped = GCSAuthConfigFromJSONTyped;
exports.GCSAuthConfigToJSON = GCSAuthConfigToJSON;
exports.GCSAuthConfigToJSONTyped = GCSAuthConfigToJSONTyped;
/**
 * Check if a given object implements the GCSAuthConfig interface.
 */
function instanceOfGCSAuthConfig(value) {
    if (!('serviceAccountJson' in value) || value['serviceAccountJson'] === undefined)
        return false;
    if (!('bucketName' in value) || value['bucketName'] === undefined)
        return false;
    return true;
}
function GCSAuthConfigFromJSON(json) {
    return GCSAuthConfigFromJSONTyped(json, false);
}
function GCSAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'serviceAccountJson': json['service-account-json'],
        'bucketName': json['bucket-name'],
    };
}
function GCSAuthConfigToJSON(json) {
    return GCSAuthConfigToJSONTyped(json, false);
}
function GCSAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'service-account-json': value['serviceAccountJson'],
        'bucket-name': value['bucketName'],
    };
}
