// @vectorize-io/vectorize-client/dist/models/GetAIPlatformConnectors200Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetAIPlatformConnectors200Response = instanceOfGetAIPlatformConnectors200Response;
exports.GetAIPlatformConnectors200ResponseFromJSON = GetAIPlatformConnectors200ResponseFromJSON;
exports.GetAIPlatformConnectors200ResponseFromJSONTyped = GetAIPlatformConnectors200ResponseFromJSONTyped;
exports.GetAIPlatformConnectors200ResponseToJSON = GetAIPlatformConnectors200ResponseToJSON;
exports.GetAIPlatformConnectors200ResponseToJSONTyped = GetAIPlatformConnectors200ResponseToJSONTyped;
var AIPlatformConnector_1 = require("./AIPlatformConnector");
/**
 * Check if a given object implements the GetAIPlatformConnectors200Response interface.
 */
function instanceOfGetAIPlatformConnectors200Response(value) {
    if (!('aiPlatformConnectors' in value) || value['aiPlatformConnectors'] === undefined)
        return false;
    return true;
}
function GetAIPlatformConnectors200ResponseFromJSON(json) {
    return GetAIPlatformConnectors200ResponseFromJSONTyped(json, false);
}
function GetAIPlatformConnectors200ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'aiPlatformConnectors': (json['aiPlatformConnectors'].map(AIPlatformConnector_1.AIPlatformConnectorFromJSON)),
    };
}
function GetAIPlatformConnectors200ResponseToJSON(json) {
    return GetAIPlatformConnectors200ResponseToJSONTyped(json, false);
}
function GetAIPlatformConnectors200ResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'aiPlatformConnectors': (value['aiPlatformConnectors'].map(AIPlatformConnector_1.AIPlatformConnectorToJSON)),
    };
}
