// @vectorize-io/vectorize-client/dist/models/GetDeepResearchResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetDeepResearchResponse = instanceOfGetDeepResearchResponse;
exports.GetDeepResearchResponseFromJSON = GetDeepResearchResponseFromJSON;
exports.GetDeepResearchResponseFromJSONTyped = GetDeepResearchResponseFromJSONTyped;
exports.GetDeepResearchResponseToJSON = GetDeepResearchResponseToJSON;
exports.GetDeepResearchResponseToJSONTyped = GetDeepResearchResponseToJSONTyped;
var DeepResearchResult_1 = require("./DeepResearchResult");
/**
 * Check if a given object implements the GetDeepResearchResponse interface.
 */
function instanceOfGetDeepResearchResponse(value) {
    if (!('ready' in value) || value['ready'] === undefined)
        return false;
    return true;
}
function GetDeepResearchResponseFromJSON(json) {
    return GetDeepResearchResponseFromJSONTyped(json, false);
}
function GetDeepResearchResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'ready': json['ready'],
        'data': json['data'] == null ? undefined : (0, DeepResearchResult_1.DeepResearchResultFromJSON)(json['data']),
    };
}
function GetDeepResearchResponseToJSON(json) {
    return GetDeepResearchResponseToJSONTyped(json, false);
}
function GetDeepResearchResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'ready': value['ready'],
        'data': (0, DeepResearchResult_1.DeepResearchResultToJSON)(value['data']),
    };
}
