// @vectorize-io/vectorize-client/dist/models/GetDestinationConnectors200Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetDestinationConnectors200Response = instanceOfGetDestinationConnectors200Response;
exports.GetDestinationConnectors200ResponseFromJSON = GetDestinationConnectors200ResponseFromJSON;
exports.GetDestinationConnectors200ResponseFromJSONTyped = GetDestinationConnectors200ResponseFromJSONTyped;
exports.GetDestinationConnectors200ResponseToJSON = GetDestinationConnectors200ResponseToJSON;
exports.GetDestinationConnectors200ResponseToJSONTyped = GetDestinationConnectors200ResponseToJSONTyped;
var DestinationConnector_1 = require("./DestinationConnector");
/**
 * Check if a given object implements the GetDestinationConnectors200Response interface.
 */
function instanceOfGetDestinationConnectors200Response(value) {
    if (!('destinationConnectors' in value) || value['destinationConnectors'] === undefined)
        return false;
    return true;
}
function GetDestinationConnectors200ResponseFromJSON(json) {
    return GetDestinationConnectors200ResponseFromJSONTyped(json, false);
}
function GetDestinationConnectors200ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'destinationConnectors': (json['destinationConnectors'].map(DestinationConnector_1.DestinationConnectorFromJSON)),
    };
}
function GetDestinationConnectors200ResponseToJSON(json) {
    return GetDestinationConnectors200ResponseToJSONTyped(json, false);
}
function GetDestinationConnectors200ResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'destinationConnectors': (value['destinationConnectors'].map(DestinationConnector_1.DestinationConnectorToJSON)),
    };
}
