// @vectorize-io/vectorize-client/dist/models/GetPipelineEventsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetPipelineEventsResponse = instanceOfGetPipelineEventsResponse;
exports.GetPipelineEventsResponseFromJSON = GetPipelineEventsResponseFromJSON;
exports.GetPipelineEventsResponseFromJSONTyped = GetPipelineEventsResponseFromJSONTyped;
exports.GetPipelineEventsResponseToJSON = GetPipelineEventsResponseToJSON;
exports.GetPipelineEventsResponseToJSONTyped = GetPipelineEventsResponseToJSONTyped;
var PipelineEvents_1 = require("./PipelineEvents");
/**
 * Check if a given object implements the GetPipelineEventsResponse interface.
 */
function instanceOfGetPipelineEventsResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function GetPipelineEventsResponseFromJSON(json) {
    return GetPipelineEventsResponseFromJSONTyped(json, false);
}
function GetPipelineEventsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'nextToken': json['nextToken'] == null ? undefined : json['nextToken'],
        'data': (json['data'].map(PipelineEvents_1.PipelineEventsFromJSON)),
    };
}
function GetPipelineEventsResponseToJSON(json) {
    return GetPipelineEventsResponseToJSONTyped(json, false);
}
function GetPipelineEventsResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'nextToken': value['nextToken'],
        'data': (value['data'].map(PipelineEvents_1.PipelineEventsToJSON)),
    };
}
