// @vectorize-io/vectorize-client/dist/models/GetPipelinesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetPipelinesResponse = instanceOfGetPipelinesResponse;
exports.GetPipelinesResponseFromJSON = GetPipelinesResponseFromJSON;
exports.GetPipelinesResponseFromJSONTyped = GetPipelinesResponseFromJSONTyped;
exports.GetPipelinesResponseToJSON = GetPipelinesResponseToJSON;
exports.GetPipelinesResponseToJSONTyped = GetPipelinesResponseToJSONTyped;
var PipelineListSummary_1 = require("./PipelineListSummary");
/**
 * Check if a given object implements the GetPipelinesResponse interface.
 */
function instanceOfGetPipelinesResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function GetPipelinesResponseFromJSON(json) {
    return GetPipelinesResponseFromJSONTyped(json, false);
}
function GetPipelinesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (json['data'].map(PipelineListSummary_1.PipelineListSummaryFromJSON)),
    };
}
function GetPipelinesResponseToJSON(json) {
    return GetPipelinesResponseToJSONTyped(json, false);
}
function GetPipelinesResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (value['data'].map(PipelineListSummary_1.PipelineListSummaryToJSON)),
    };
}
