// @vectorize-io/vectorize-client/dist/models/GetPipelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetPipelineResponse = instanceOfGetPipelineResponse;
exports.GetPipelineResponseFromJSON = GetPipelineResponseFromJSON;
exports.GetPipelineResponseFromJSONTyped = GetPipelineResponseFromJSONTyped;
exports.GetPipelineResponseToJSON = GetPipelineResponseToJSON;
exports.GetPipelineResponseToJSONTyped = GetPipelineResponseToJSONTyped;
var PipelineSummary_1 = require("./PipelineSummary");
/**
 * Check if a given object implements the GetPipelineResponse interface.
 */
function instanceOfGetPipelineResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function GetPipelineResponseFromJSON(json) {
    return GetPipelineResponseFromJSONTyped(json, false);
}
function GetPipelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (0, PipelineSummary_1.PipelineSummaryFromJSON)(json['data']),
    };
}
function GetPipelineResponseToJSON(json) {
    return GetPipelineResponseToJSONTyped(json, false);
}
function GetPipelineResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (0, PipelineSummary_1.PipelineSummaryToJSON)(value['data']),
    };
}
