// @vectorize-io/vectorize-client/dist/models/GetPipelineMetricsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetPipelineMetricsResponse = instanceOfGetPipelineMetricsResponse;
exports.GetPipelineMetricsResponseFromJSON = GetPipelineMetricsResponseFromJSON;
exports.GetPipelineMetricsResponseFromJSONTyped = GetPipelineMetricsResponseFromJSONTyped;
exports.GetPipelineMetricsResponseToJSON = GetPipelineMetricsResponseToJSON;
exports.GetPipelineMetricsResponseToJSONTyped = GetPipelineMetricsResponseToJSONTyped;
var PipelineMetrics_1 = require("./PipelineMetrics");
/**
 * Check if a given object implements the GetPipelineMetricsResponse interface.
 */
function instanceOfGetPipelineMetricsResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function GetPipelineMetricsResponseFromJSON(json) {
    return GetPipelineMetricsResponseFromJSONTyped(json, false);
}
function GetPipelineMetricsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (json['data'].map(PipelineMetrics_1.PipelineMetricsFromJSON)),
    };
}
function GetPipelineMetricsResponseToJSON(json) {
    return GetPipelineMetricsResponseToJSONTyped(json, false);
}
function GetPipelineMetricsResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (value['data'].map(PipelineMetrics_1.PipelineMetricsToJSON)),
    };
}
