// @vectorize-io/vectorize-client/dist/models/GetUploadFilesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetUploadFilesResponse = instanceOfGetUploadFilesResponse;
exports.GetUploadFilesResponseFromJSON = GetUploadFilesResponseFromJSON;
exports.GetUploadFilesResponseFromJSONTyped = GetUploadFilesResponseFromJSONTyped;
exports.GetUploadFilesResponseToJSON = GetUploadFilesResponseToJSON;
exports.GetUploadFilesResponseToJSONTyped = GetUploadFilesResponseToJSONTyped;
var UploadFile_1 = require("./UploadFile");
/**
 * Check if a given object implements the GetUploadFilesResponse interface.
 */
function instanceOfGetUploadFilesResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('files' in value) || value['files'] === undefined)
        return false;
    return true;
}
function GetUploadFilesResponseFromJSON(json) {
    return GetUploadFilesResponseFromJSONTyped(json, false);
}
function GetUploadFilesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'files': (json['files'].map(UploadFile_1.UploadFileFromJSON)),
    };
}
function GetUploadFilesResponseToJSON(json) {
    return GetUploadFilesResponseToJSONTyped(json, false);
}
function GetUploadFilesResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'files': (value['files'].map(UploadFile_1.UploadFileToJSON)),
    };
}
