// @vectorize-io/vectorize-client/dist/models/GetWorkspacesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetWorkspacesResponse = instanceOfGetWorkspacesResponse;
exports.GetWorkspacesResponseFromJSON = GetWorkspacesResponseFromJSON;
exports.GetWorkspacesResponseFromJSONTyped = GetWorkspacesResponseFromJSONTyped;
exports.GetWorkspacesResponseToJSON = GetWorkspacesResponseToJSON;
exports.GetWorkspacesResponseToJSONTyped = GetWorkspacesResponseToJSONTyped;
var WorkspaceSchema_1 = require("./WorkspaceSchema");
/**
 * Check if a given object implements the GetWorkspacesResponse interface.
 */
function instanceOfGetWorkspacesResponse(value) {
    if (!('workspaces' in value) || value['workspaces'] === undefined)
        return false;
    return true;
}
function GetWorkspacesResponseFromJSON(json) {
    return GetWorkspacesResponseFromJSONTyped(json, false);
}
function GetWorkspacesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'workspaces': (json['workspaces'].map(WorkspaceSchema_1.WorkspaceSchemaFromJSON)),
        'nextToken': json['nextToken'] == null ? undefined : json['nextToken'],
    };
}
function GetWorkspacesResponseToJSON(json) {
    return GetWorkspacesResponseToJSONTyped(json, false);
}
function GetWorkspacesResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'workspaces': (value['workspaces'].map(WorkspaceSchema_1.WorkspaceSchemaToJSON)),
        'nextToken': value['nextToken'],
    };
}
