// @vectorize-io/vectorize-client/dist/models/GetWorkspaceByIdResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetWorkspaceByIdResponse = instanceOfGetWorkspaceByIdResponse;
exports.GetWorkspaceByIdResponseFromJSON = GetWorkspaceByIdResponseFromJSON;
exports.GetWorkspaceByIdResponseFromJSONTyped = GetWorkspaceByIdResponseFromJSONTyped;
exports.GetWorkspaceByIdResponseToJSON = GetWorkspaceByIdResponseToJSON;
exports.GetWorkspaceByIdResponseToJSONTyped = GetWorkspaceByIdResponseToJSONTyped;
/**
 * Check if a given object implements the GetWorkspaceByIdResponse interface.
 */
function instanceOfGetWorkspaceByIdResponse(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    return true;
}
function GetWorkspaceByIdResponseFromJSON(json) {
    return GetWorkspaceByIdResponseFromJSONTyped(json, false);
}
function GetWorkspaceByIdResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'name': json['name'],
        'createdAt': json['createdAt'] == null ? undefined : json['createdAt'],
        'updatedAt': json['updatedAt'] == null ? undefined : json['updatedAt'],
    };
}
function GetWorkspaceByIdResponseToJSON(json) {
    return GetWorkspaceByIdResponseToJSONTyped(json, false);
}
function GetWorkspaceByIdResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'name': value['name'],
        'createdAt': value['createdAt'],
        'updatedAt': value['updatedAt'],
    };
}
