// @vectorize-io/vectorize-client/dist/models/GetWorkspaces400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGetWorkspaces400Response = instanceOfGetWorkspaces400Response;
exports.GetWorkspaces400ResponseFromJSON = GetWorkspaces400ResponseFromJSON;
exports.GetWorkspaces400ResponseFromJSONTyped = GetWorkspaces400ResponseFromJSONTyped;
exports.GetWorkspaces400ResponseToJSON = GetWorkspaces400ResponseToJSON;
exports.GetWorkspaces400ResponseToJSONTyped = GetWorkspaces400ResponseToJSONTyped;
/**
 * Check if a given object implements the GetWorkspaces400Response interface.
 */
function instanceOfGetWorkspaces400Response(value) {
    if (!('error' in value) || value['error'] === undefined)
        return false;
    return true;
}
function GetWorkspaces400ResponseFromJSON(json) {
    return GetWorkspaces400ResponseFromJSONTyped(json, false);
}
function GetWorkspaces400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'error': json['error'],
        'details': json['details'] == null ? undefined : json['details'],
        'failedUpdates': json['failedUpdates'] == null ? undefined : json['failedUpdates'],
        'successfulUpdates': json['successfulUpdates'] == null ? undefined : json['successfulUpdates'],
    };
}
function GetWorkspaces400ResponseToJSON(json) {
    return GetWorkspaces400ResponseToJSONTyped(json, false);
}
function GetWorkspaces400ResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'error': value['error'],
        'details': value['details'],
        'failedUpdates': value['failedUpdates'],
        'successfulUpdates': value['successfulUpdates'],
    };
}
