// @vectorize-io/vectorize-client/dist/models/Github.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GithubTypeEnum = void 0;
exports.instanceOfGithub = instanceOfGithub;
exports.GithubFromJSON = GithubFromJSON;
exports.GithubFromJSONTyped = GithubFromJSONTyped;
exports.GithubToJSON = GithubToJSON;
exports.GithubToJSONTyped = GithubToJSONTyped;
var GITHUBAuthConfig_1 = require("./GITHUBAuthConfig");
/**
 * @export
 */
exports.GithubTypeEnum = {
    Github: 'GITHUB'
};
/**
 * Check if a given object implements the Github interface.
 */
function instanceOfGithub(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function GithubFromJSON(json) {
    return GithubFromJSONTyped(json, false);
}
function GithubFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, GITHUBAuthConfig_1.GITHUBAuthConfigFromJSON)(json['config']),
    };
}
function GithubToJSON(json) {
    return GithubToJSONTyped(json, false);
}
function GithubToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, GITHUBAuthConfig_1.GITHUBAuthConfigToJSON)(value['config']),
    };
}
