// @vectorize-io/vectorize-client/dist/models/Github1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGithub1 = instanceOfGithub1;
exports.Github1FromJSON = Github1FromJSON;
exports.Github1FromJSONTyped = Github1FromJSONTyped;
exports.Github1ToJSON = Github1ToJSON;
exports.Github1ToJSONTyped = Github1ToJSONTyped;
var GITHUBAuthConfig_1 = require("./GITHUBAuthConfig");
/**
 * Check if a given object implements the Github1 interface.
 */
function instanceOfGithub1(value) {
    return true;
}
function Github1FromJSON(json) {
    return Github1FromJSONTyped(json, false);
}
function Github1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, GITHUBAuthConfig_1.GITHUBAuthConfigFromJSON)(json['config']),
    };
}
function Github1ToJSON(json) {
    return Github1ToJSONTyped(json, false);
}
function Github1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, GITHUBAuthConfig_1.GITHUBAuthConfigToJSON)(value['config']),
    };
}
