// @vectorize-io/vectorize-client/dist/models/GITHUBAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGITHUBAuthConfig = instanceOfGITHUBAuthConfig;
exports.GITHUBAuthConfigFromJSON = GITHUBAuthConfigFromJSON;
exports.GITHUBAuthConfigFromJSONTyped = GITHUBAuthConfigFromJSONTyped;
exports.GITHUBAuthConfigToJSON = GITHUBAuthConfigToJSON;
exports.GITHUBAuthConfigToJSONTyped = GITHUBAuthConfigToJSONTyped;
/**
 * Check if a given object implements the GITHUBAuthConfig interface.
 */
function instanceOfGITHUBAuthConfig(value) {
    if (!('oauthToken' in value) || value['oauthToken'] === undefined)
        return false;
    return true;
}
function GITHUBAuthConfigFromJSON(json) {
    return GITHUBAuthConfigFromJSONTyped(json, false);
}
function GITHUBAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'oauthToken': json['oauth-token'],
    };
}
function GITHUBAuthConfigToJSON(json) {
    return GITHUBAuthConfigToJSONTyped(json, false);
}
function GITHUBAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'oauth-token': value['oauthToken'],
    };
}
