// @vectorize-io/vectorize-client/dist/models/GoogleDrive1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGoogleDrive1 = instanceOfGoogleDrive1;
exports.GoogleDrive1FromJSON = GoogleDrive1FromJSON;
exports.GoogleDrive1FromJSONTyped = GoogleDrive1FromJSONTyped;
exports.GoogleDrive1ToJSON = GoogleDrive1ToJSON;
exports.GoogleDrive1ToJSONTyped = GoogleDrive1ToJSONTyped;
var GOOGLEDRIVEAuthConfig_1 = require("./GOOGLEDRIVEAuthConfig");
/**
 * Check if a given object implements the GoogleDrive1 interface.
 */
function instanceOfGoogleDrive1(value) {
    return true;
}
function GoogleDrive1FromJSON(json) {
    return GoogleDrive1FromJSONTyped(json, false);
}
function GoogleDrive1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, GOOGLEDRIVEAuthConfig_1.GOOGLEDRIVEAuthConfigFromJSON)(json['config']),
    };
}
function GoogleDrive1ToJSON(json) {
    return GoogleDrive1ToJSONTyped(json, false);
}
function GoogleDrive1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, GOOGLEDRIVEAuthConfig_1.GOOGLEDRIVEAuthConfigToJSON)(value['config']),
    };
}
