// @vectorize-io/vectorize-client/dist/models/GOOGLEDRIVEOAUTHConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GOOGLEDRIVEOAUTHConfigFileExtensionsEnum = void 0;
exports.instanceOfGOOGLEDRIVEOAUTHConfig = instanceOfGOOGLEDRIVEOAUTHConfig;
exports.GOOGLEDRIVEOAUTHConfigFromJSON = GOOGLEDRIVEOAUTHConfigFromJSON;
exports.GOOGLEDRIVEOAUTHConfigFromJSONTyped = GOOGLEDRIVEOAUTHConfigFromJSONTyped;
exports.GOOGLEDRIVEOAUTHConfigToJSON = GOOGLEDRIVEOAUTHConfigToJSON;
exports.GOOGLEDRIVEOAUTHConfigToJSONTyped = GOOGLEDRIVEOAUTHConfigToJSONTyped;
/**
 * @export
 */
exports.GOOGLEDRIVEOAUTHConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Md: 'md',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the GOOGLEDRIVEOAUTHConfig interface.
 */
function instanceOfGOOGLEDRIVEOAUTHConfig(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    return true;
}
function GOOGLEDRIVEOAUTHConfigFromJSON(json) {
    return GOOGLEDRIVEOAUTHConfigFromJSONTyped(json, false);
}
function GOOGLEDRIVEOAUTHConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'idleTime': json['idle-time'] == null ? undefined : json['idle-time'],
    };
}
function GOOGLEDRIVEOAUTHConfigToJSON(json) {
    return GOOGLEDRIVEOAUTHConfigToJSONTyped(json, false);
}
function GOOGLEDRIVEOAUTHConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'idle-time': value['idleTime'],
    };
}
