// @vectorize-io/vectorize-client/dist/models/GOOGLEDRIVEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GOOGLEDRIVEConfigFileExtensionsEnum = void 0;
exports.instanceOfGOOGLEDRIVEConfig = instanceOfGOOGLEDRIVEConfig;
exports.GOOGLEDRIVEConfigFromJSON = GOOGLEDRIVEConfigFromJSON;
exports.GOOGLEDRIVEConfigFromJSONTyped = GOOGLEDRIVEConfigFromJSONTyped;
exports.GOOGLEDRIVEConfigToJSON = GOOGLEDRIVEConfigToJSON;
exports.GOOGLEDRIVEConfigToJSONTyped = GOOGLEDRIVEConfigToJSONTyped;
/**
 * @export
 */
exports.GOOGLEDRIVEConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Md: 'md',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the GOOGLEDRIVEConfig interface.
 */
function instanceOfGOOGLEDRIVEConfig(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    return true;
}
function GOOGLEDRIVEConfigFromJSON(json) {
    return GOOGLEDRIVEConfigFromJSONTyped(json, false);
}
function GOOGLEDRIVEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'rootParents': json['root-parents'] == null ? undefined : json['root-parents'],
        'idleTime': json['idle-time'] == null ? undefined : json['idle-time'],
    };
}
function GOOGLEDRIVEConfigToJSON(json) {
    return GOOGLEDRIVEConfigToJSONTyped(json, false);
}
function GOOGLEDRIVEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'root-parents': value['rootParents'],
        'idle-time': value['idleTime'],
    };
}
