// @vectorize-io/vectorize-client/dist/models/GoogleDrive.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleDriveTypeEnum = void 0;
exports.instanceOfGoogleDrive = instanceOfGoogleDrive;
exports.GoogleDriveFromJSON = GoogleDriveFromJSON;
exports.GoogleDriveFromJSONTyped = GoogleDriveFromJSONTyped;
exports.GoogleDriveToJSON = GoogleDriveToJSON;
exports.GoogleDriveToJSONTyped = GoogleDriveToJSONTyped;
var GOOGLEDRIVEAuthConfig_1 = require("./GOOGLEDRIVEAuthConfig");
/**
 * @export
 */
exports.GoogleDriveTypeEnum = {
    GoogleDrive: 'GOOGLE_DRIVE'
};
/**
 * Check if a given object implements the GoogleDrive interface.
 */
function instanceOfGoogleDrive(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function GoogleDriveFromJSON(json) {
    return GoogleDriveFromJSONTyped(json, false);
}
function GoogleDriveFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, GOOGLEDRIVEAuthConfig_1.GOOGLEDRIVEAuthConfigFromJSON)(json['config']),
    };
}
function GoogleDriveToJSON(json) {
    return GoogleDriveToJSONTyped(json, false);
}
function GoogleDriveToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, GOOGLEDRIVEAuthConfig_1.GOOGLEDRIVEAuthConfigToJSON)(value['config']),
    };
}
