// @vectorize-io/vectorize-client/dist/models/GOOGLEDRIVEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGOOGLEDRIVEAuthConfig = instanceOfGOOGLEDRIVEAuthConfig;
exports.GOOGLEDRIVEAuthConfigFromJSON = GOOGLEDRIVEAuthConfigFromJSON;
exports.GOOGLEDRIVEAuthConfigFromJSONTyped = GOOGLEDRIVEAuthConfigFromJSONTyped;
exports.GOOGLEDRIVEAuthConfigToJSON = GOOGLEDRIVEAuthConfigToJSON;
exports.GOOGLEDRIVEAuthConfigToJSONTyped = GOOGLEDRIVEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the GOOGLEDRIVEAuthConfig interface.
 */
function instanceOfGOOGLEDRIVEAuthConfig(value) {
    if (!('serviceAccountJson' in value) || value['serviceAccountJson'] === undefined)
        return false;
    return true;
}
function GOOGLEDRIVEAuthConfigFromJSON(json) {
    return GOOGLEDRIVEAuthConfigFromJSONTyped(json, false);
}
function GOOGLEDRIVEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'serviceAccountJson': json['service-account-json'],
    };
}
function GOOGLEDRIVEAuthConfigToJSON(json) {
    return GOOGLEDRIVEAuthConfigToJSONTyped(json, false);
}
function GOOGLEDRIVEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'service-account-json': value['serviceAccountJson'],
    };
}
