// @vectorize-io/vectorize-client/dist/models/GOOGLEDRIVEOAUTHMULTICUSTOMConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigFileExtensionsEnum = void 0;
exports.instanceOfGOOGLEDRIVEOAUTHMULTICUSTOMConfig = instanceOfGOOGLEDRIVEOAUTHMULTICUSTOMConfig;
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSON = GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSON;
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSONTyped = GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSONTyped;
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSON = GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSON;
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSONTyped = GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSONTyped;
/**
 * @export
 */
exports.GOOGLEDRIVEOAUTHMULTICUSTOMConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Md: 'md',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the GOOGLEDRIVEOAUTHMULTICUSTOMConfig interface.
 */
function instanceOfGOOGLEDRIVEOAUTHMULTICUSTOMConfig(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    return true;
}
function GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSON(json) {
    return GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSONTyped(json, false);
}
function GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'idleTime': json['idle-time'] == null ? undefined : json['idle-time'],
    };
}
function GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSON(json) {
    return GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSONTyped(json, false);
}
function GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'idle-time': value['idleTime'],
    };
}
