// @vectorize-io/vectorize-client/dist/models/GROQAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGROQAuthConfig = instanceOfGROQAuthConfig;
exports.GROQAuthConfigFromJSON = GROQAuthConfigFromJSON;
exports.GROQAuthConfigFromJSONTyped = GROQAuthConfigFromJSONTyped;
exports.GROQAuthConfigToJSON = GROQAuthConfigToJSON;
exports.GROQAuthConfigToJSONTyped = GROQAuthConfigToJSONTyped;
/**
 * Check if a given object implements the GROQAuthConfig interface.
 */
function instanceOfGROQAuthConfig(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    return true;
}
function GROQAuthConfigFromJSON(json) {
    return GROQAuthConfigFromJSONTyped(json, false);
}
function GROQAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
function GROQAuthConfigToJSON(json) {
    return GROQAuthConfigToJSONTyped(json, false);
}
function GROQAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
    };
}
