// @vectorize-io/vectorize-client/dist/models/Groq1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGroq1 = instanceOfGroq1;
exports.Groq1FromJSON = Groq1FromJSON;
exports.Groq1FromJSONTyped = Groq1FromJSONTyped;
exports.Groq1ToJSON = Groq1ToJSON;
exports.Groq1ToJSONTyped = Groq1ToJSONTyped;
/**
 * Check if a given object implements the Groq1 interface.
 */
function instanceOfGroq1(value) {
    return true;
}
function Groq1FromJSON(json) {
    return Groq1FromJSONTyped(json, false);
}
function Groq1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Groq1ToJSON(json) {
    return Groq1ToJSONTyped(json, false);
}
function Groq1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
