// @vectorize-io/vectorize-client/dist/models/Groq.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroqTypeEnum = void 0;
exports.instanceOfGroq = instanceOfGroq;
exports.GroqFromJSON = GroqFromJSON;
exports.GroqFromJSONTyped = GroqFromJSONTyped;
exports.GroqToJSON = GroqToJSON;
exports.GroqToJSONTyped = GroqToJSONTyped;
var GROQAuthConfig_1 = require("./GROQAuthConfig");
/**
 * @export
 */
exports.GroqTypeEnum = {
    Groq: 'GROQ'
};
/**
 * Check if a given object implements the Groq interface.
 */
function instanceOfGroq(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function GroqFromJSON(json) {
    return GroqFromJSONTyped(json, false);
}
function GroqFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, GROQAuthConfig_1.GROQAuthConfigFromJSON)(json['config']),
    };
}
function GroqToJSON(json) {
    return GroqToJSONTyped(json, false);
}
function GroqToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, GROQAuthConfig_1.GROQAuthConfigToJSON)(value['config']),
    };
}
