// @vectorize-io/vectorize-client/dist/models/MetadataExtractionStrategySchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMetadataExtractionStrategySchema = instanceOfMetadataExtractionStrategySchema;
exports.MetadataExtractionStrategySchemaFromJSON = MetadataExtractionStrategySchemaFromJSON;
exports.MetadataExtractionStrategySchemaFromJSONTyped = MetadataExtractionStrategySchemaFromJSONTyped;
exports.MetadataExtractionStrategySchemaToJSON = MetadataExtractionStrategySchemaToJSON;
exports.MetadataExtractionStrategySchemaToJSONTyped = MetadataExtractionStrategySchemaToJSONTyped;
/**
 * Check if a given object implements the MetadataExtractionStrategySchema interface.
 */
function instanceOfMetadataExtractionStrategySchema(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('schema' in value) || value['schema'] === undefined)
        return false;
    return true;
}
function MetadataExtractionStrategySchemaFromJSON(json) {
    return MetadataExtractionStrategySchemaFromJSONTyped(json, false);
}
function MetadataExtractionStrategySchemaFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'schema': json['schema'],
    };
}
function MetadataExtractionStrategySchemaToJSON(json) {
    return MetadataExtractionStrategySchemaToJSONTyped(json, false);
}
function MetadataExtractionStrategySchemaToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'schema': value['schema'],
    };
}
