// @vectorize-io/vectorize-client/dist/models/MetadataExtractionStrategy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMetadataExtractionStrategy = instanceOfMetadataExtractionStrategy;
exports.MetadataExtractionStrategyFromJSON = MetadataExtractionStrategyFromJSON;
exports.MetadataExtractionStrategyFromJSONTyped = MetadataExtractionStrategyFromJSONTyped;
exports.MetadataExtractionStrategyToJSON = MetadataExtractionStrategyToJSON;
exports.MetadataExtractionStrategyToJSONTyped = MetadataExtractionStrategyToJSONTyped;
var MetadataExtractionStrategySchema_1 = require("./MetadataExtractionStrategySchema");
/**
 * Check if a given object implements the MetadataExtractionStrategy interface.
 */
function instanceOfMetadataExtractionStrategy(value) {
    return true;
}
function MetadataExtractionStrategyFromJSON(json) {
    return MetadataExtractionStrategyFromJSONTyped(json, false);
}
function MetadataExtractionStrategyFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'schemas': json['schemas'] == null ? undefined : (json['schemas'].map(MetadataExtractionStrategySchema_1.MetadataExtractionStrategySchemaFromJSON)),
        'inferSchema': json['inferSchema'] == null ? undefined : json['inferSchema'],
    };
}
function MetadataExtractionStrategyToJSON(json) {
    return MetadataExtractionStrategyToJSONTyped(json, false);
}
function MetadataExtractionStrategyToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'schemas': value['schemas'] == null ? undefined : (value['schemas'].map(MetadataExtractionStrategySchema_1.MetadataExtractionStrategySchemaToJSON)),
        'inferSchema': value['inferSchema'],
    };
}
