// @vectorize-io/vectorize-client/dist/models/Milvus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MilvusTypeEnum = void 0;
exports.instanceOfMilvus = instanceOfMilvus;
exports.MilvusFromJSON = MilvusFromJSON;
exports.MilvusFromJSONTyped = MilvusFromJSONTyped;
exports.MilvusToJSON = MilvusToJSON;
exports.MilvusToJSONTyped = MilvusToJSONTyped;
var MILVUSAuthConfig_1 = require("./MILVUSAuthConfig");
/**
 * @export
 */
exports.MilvusTypeEnum = {
    Milvus: 'MILVUS'
};
/**
 * Check if a given object implements the Milvus interface.
 */
function instanceOfMilvus(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function MilvusFromJSON(json) {
    return MilvusFromJSONTyped(json, false);
}
function MilvusFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, MILVUSAuthConfig_1.MILVUSAuthConfigFromJSON)(json['config']),
    };
}
function MilvusToJSON(json) {
    return MilvusToJSONTyped(json, false);
}
function MilvusToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, MILVUSAuthConfig_1.MILVUSAuthConfigToJSON)(value['config']),
    };
}
