// @vectorize-io/vectorize-client/dist/models/MILVUSAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMILVUSAuthConfig = instanceOfMILVUSAuthConfig;
exports.MILVUSAuthConfigFromJSON = MILVUSAuthConfigFromJSON;
exports.MILVUSAuthConfigFromJSONTyped = MILVUSAuthConfigFromJSONTyped;
exports.MILVUSAuthConfigToJSON = MILVUSAuthConfigToJSON;
exports.MILVUSAuthConfigToJSONTyped = MILVUSAuthConfigToJSONTyped;
/**
 * Check if a given object implements the MILVUSAuthConfig interface.
 */
function instanceOfMILVUSAuthConfig(value) {
    if (!('url' in value) || value['url'] === undefined)
        return false;
    return true;
}
function MILVUSAuthConfigFromJSON(json) {
    return MILVUSAuthConfigFromJSONTyped(json, false);
}
function MILVUSAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'url': json['url'],
        'token': json['token'] == null ? undefined : json['token'],
        'username': json['username'] == null ? undefined : json['username'],
        'password': json['password'] == null ? undefined : json['password'],
    };
}
function MILVUSAuthConfigToJSON(json) {
    return MILVUSAuthConfigToJSONTyped(json, false);
}
function MILVUSAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'url': value['url'],
        'token': value['token'],
        'username': value['username'],
        'password': value['password'],
    };
}
