// @vectorize-io/vectorize-client/dist/models/Milvus1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMilvus1 = instanceOfMilvus1;
exports.Milvus1FromJSON = Milvus1FromJSON;
exports.Milvus1FromJSONTyped = Milvus1FromJSONTyped;
exports.Milvus1ToJSON = Milvus1ToJSON;
exports.Milvus1ToJSONTyped = Milvus1ToJSONTyped;
var MILVUSAuthConfig_1 = require("./MILVUSAuthConfig");
/**
 * Check if a given object implements the Milvus1 interface.
 */
function instanceOfMilvus1(value) {
    return true;
}
function Milvus1FromJSON(json) {
    return Milvus1FromJSONTyped(json, false);
}
function Milvus1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, MILVUSAuthConfig_1.MILVUSAuthConfigFromJSON)(json['config']),
    };
}
function Milvus1ToJSON(json) {
    return Milvus1ToJSONTyped(json, false);
}
function Milvus1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, MILVUSAuthConfig_1.MILVUSAuthConfigToJSON)(value['config']),
    };
}
