// @vectorize-io/vectorize-client/dist/models/N8NConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfN8NConfig = instanceOfN8NConfig;
exports.N8NConfigFromJSON = N8NConfigFromJSON;
exports.N8NConfigFromJSONTyped = N8NConfigFromJSONTyped;
exports.N8NConfigToJSON = N8NConfigToJSON;
exports.N8NConfigToJSONTyped = N8NConfigToJSONTyped;
/**
 * Check if a given object implements the N8NConfig interface.
 */
function instanceOfN8NConfig(value) {
    if (!('account' in value) || value['account'] === undefined)
        return false;
    if (!('webhookPath' in value) || value['webhookPath'] === undefined)
        return false;
    return true;
}
function N8NConfigFromJSON(json) {
    return N8NConfigFromJSONTyped(json, false);
}
function N8NConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'account': json['account'],
        'webhookPath': json['webhookPath'],
        'headers': json['headers'] == null ? undefined : json['headers'],
    };
}
function N8NConfigToJSON(json) {
    return N8NConfigToJSONTyped(json, false);
}
function N8NConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'account': value['account'],
        'webhookPath': value['webhookPath'],
        'headers': value['headers'],
    };
}
