// @vectorize-io/vectorize-client/dist/models/NOTIONConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfNOTIONConfig = instanceOfNOTIONConfig;
exports.NOTIONConfigFromJSON = NOTIONConfigFromJSON;
exports.NOTIONConfigFromJSONTyped = NOTIONConfigFromJSONTyped;
exports.NOTIONConfigToJSON = NOTIONConfigToJSON;
exports.NOTIONConfigToJSONTyped = NOTIONConfigToJSONTyped;
/**
 * Check if a given object implements the NOTIONConfig interface.
 */
function instanceOfNOTIONConfig(value) {
    if (!('selectResources' in value) || value['selectResources'] === undefined)
        return false;
    if (!('databaseIds' in value) || value['databaseIds'] === undefined)
        return false;
    if (!('databaseNames' in value) || value['databaseNames'] === undefined)
        return false;
    if (!('pageIds' in value) || value['pageIds'] === undefined)
        return false;
    if (!('pageNames' in value) || value['pageNames'] === undefined)
        return false;
    return true;
}
function NOTIONConfigFromJSON(json) {
    return NOTIONConfigFromJSONTyped(json, false);
}
function NOTIONConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'selectResources': json['select-resources'],
        'databaseIds': json['database-ids'],
        'databaseNames': json['database-names'],
        'pageIds': json['page-ids'],
        'pageNames': json['page-names'],
    };
}
function NOTIONConfigToJSON(json) {
    return NOTIONConfigToJSONTyped(json, false);
}
function NOTIONConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'select-resources': value['selectResources'],
        'database-ids': value['databaseIds'],
        'database-names': value['databaseNames'],
        'page-ids': value['pageIds'],
        'page-names': value['pageNames'],
    };
}
