// @vectorize-io/vectorize-client/dist/models/ONEDRIVEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ONEDRIVEConfigFileExtensionsEnum = void 0;
exports.instanceOfONEDRIVEConfig = instanceOfONEDRIVEConfig;
exports.ONEDRIVEConfigFromJSON = ONEDRIVEConfigFromJSON;
exports.ONEDRIVEConfigFromJSONTyped = ONEDRIVEConfigFromJSONTyped;
exports.ONEDRIVEConfigToJSON = ONEDRIVEConfigToJSON;
exports.ONEDRIVEConfigToJSONTyped = ONEDRIVEConfigToJSONTyped;
/**
 * @export
 */
exports.ONEDRIVEConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Md: 'md',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the ONEDRIVEConfig interface.
 */
function instanceOfONEDRIVEConfig(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    return true;
}
function ONEDRIVEConfigFromJSON(json) {
    return ONEDRIVEConfigFromJSONTyped(json, false);
}
function ONEDRIVEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'pathPrefix': json['path-prefix'] == null ? undefined : json['path-prefix'],
    };
}
function ONEDRIVEConfigToJSON(json) {
    return ONEDRIVEConfigToJSONTyped(json, false);
}
function ONEDRIVEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'path-prefix': value['pathPrefix'],
    };
}
