// @vectorize-io/vectorize-client/dist/models/OneDrive1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfOneDrive1 = instanceOfOneDrive1;
exports.OneDrive1FromJSON = OneDrive1FromJSON;
exports.OneDrive1FromJSONTyped = OneDrive1FromJSONTyped;
exports.OneDrive1ToJSON = OneDrive1ToJSON;
exports.OneDrive1ToJSONTyped = OneDrive1ToJSONTyped;
var ONEDRIVEAuthConfig_1 = require("./ONEDRIVEAuthConfig");
/**
 * Check if a given object implements the OneDrive1 interface.
 */
function instanceOfOneDrive1(value) {
    return true;
}
function OneDrive1FromJSON(json) {
    return OneDrive1FromJSONTyped(json, false);
}
function OneDrive1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, ONEDRIVEAuthConfig_1.ONEDRIVEAuthConfigFromJSON)(json['config']),
    };
}
function OneDrive1ToJSON(json) {
    return OneDrive1ToJSONTyped(json, false);
}
function OneDrive1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, ONEDRIVEAuthConfig_1.ONEDRIVEAuthConfigToJSON)(value['config']),
    };
}
