// @vectorize-io/vectorize-client/dist/models/ONEDRIVEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfONEDRIVEAuthConfig = instanceOfONEDRIVEAuthConfig;
exports.ONEDRIVEAuthConfigFromJSON = ONEDRIVEAuthConfigFromJSON;
exports.ONEDRIVEAuthConfigFromJSONTyped = ONEDRIVEAuthConfigFromJSONTyped;
exports.ONEDRIVEAuthConfigToJSON = ONEDRIVEAuthConfigToJSON;
exports.ONEDRIVEAuthConfigToJSONTyped = ONEDRIVEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the ONEDRIVEAuthConfig interface.
 */
function instanceOfONEDRIVEAuthConfig(value) {
    if (!('msClientId' in value) || value['msClientId'] === undefined)
        return false;
    if (!('msTenantId' in value) || value['msTenantId'] === undefined)
        return false;
    if (!('msClientSecret' in value) || value['msClientSecret'] === undefined)
        return false;
    if (!('users' in value) || value['users'] === undefined)
        return false;
    return true;
}
function ONEDRIVEAuthConfigFromJSON(json) {
    return ONEDRIVEAuthConfigFromJSONTyped(json, false);
}
function ONEDRIVEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'msClientId': json['ms-client-id'],
        'msTenantId': json['ms-tenant-id'],
        'msClientSecret': json['ms-client-secret'],
        'users': json['users'],
    };
}
function ONEDRIVEAuthConfigToJSON(json) {
    return ONEDRIVEAuthConfigToJSONTyped(json, false);
}
function ONEDRIVEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'ms-client-id': value['msClientId'],
        'ms-tenant-id': value['msTenantId'],
        'ms-client-secret': value['msClientSecret'],
        'users': value['users'],
    };
}
