// @vectorize-io/vectorize-client/dist/models/OneDrive.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OneDriveTypeEnum = void 0;
exports.instanceOfOneDrive = instanceOfOneDrive;
exports.OneDriveFromJSON = OneDriveFromJSON;
exports.OneDriveFromJSONTyped = OneDriveFromJSONTyped;
exports.OneDriveToJSON = OneDriveToJSON;
exports.OneDriveToJSONTyped = OneDriveToJSONTyped;
var ONEDRIVEAuthConfig_1 = require("./ONEDRIVEAuthConfig");
/**
 * @export
 */
exports.OneDriveTypeEnum = {
    OneDrive: 'ONE_DRIVE'
};
/**
 * Check if a given object implements the OneDrive interface.
 */
function instanceOfOneDrive(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function OneDriveFromJSON(json) {
    return OneDriveFromJSONTyped(json, false);
}
function OneDriveFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, ONEDRIVEAuthConfig_1.ONEDRIVEAuthConfigFromJSON)(json['config']),
    };
}
function OneDriveToJSON(json) {
    return OneDriveToJSONTyped(json, false);
}
function OneDriveToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, ONEDRIVEAuthConfig_1.ONEDRIVEAuthConfigToJSON)(value['config']),
    };
}
