// @vectorize-io/vectorize-client/dist/models/Openai1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfOpenai1 = instanceOfOpenai1;
exports.Openai1FromJSON = Openai1FromJSON;
exports.Openai1FromJSONTyped = Openai1FromJSONTyped;
exports.Openai1ToJSON = Openai1ToJSON;
exports.Openai1ToJSONTyped = Openai1ToJSONTyped;
/**
 * Check if a given object implements the Openai1 interface.
 */
function instanceOfOpenai1(value) {
    return true;
}
function Openai1FromJSON(json) {
    return Openai1FromJSONTyped(json, false);
}
function Openai1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Openai1ToJSON(json) {
    return Openai1ToJSONTyped(json, false);
}
function Openai1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
