// @vectorize-io/vectorize-client/dist/models/Openai.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenaiTypeEnum = void 0;
exports.instanceOfOpenai = instanceOfOpenai;
exports.OpenaiFromJSON = OpenaiFromJSON;
exports.OpenaiFromJSONTyped = OpenaiFromJSONTyped;
exports.OpenaiToJSON = OpenaiToJSON;
exports.OpenaiToJSONTyped = OpenaiToJSONTyped;
var OPENAIAuthConfig_1 = require("./OPENAIAuthConfig");
/**
 * @export
 */
exports.OpenaiTypeEnum = {
    Openai: 'OPENAI'
};
/**
 * Check if a given object implements the Openai interface.
 */
function instanceOfOpenai(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function OpenaiFromJSON(json) {
    return OpenaiFromJSONTyped(json, false);
}
function OpenaiFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, OPENAIAuthConfig_1.OPENAIAuthConfigFromJSON)(json['config']),
    };
}
function OpenaiToJSON(json) {
    return OpenaiToJSONTyped(json, false);
}
function OpenaiToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, OPENAIAuthConfig_1.OPENAIAuthConfigToJSON)(value['config']),
    };
}
