// @vectorize-io/vectorize-client/dist/models/OPENAIAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfOPENAIAuthConfig = instanceOfOPENAIAuthConfig;
exports.OPENAIAuthConfigFromJSON = OPENAIAuthConfigFromJSON;
exports.OPENAIAuthConfigFromJSONTyped = OPENAIAuthConfigFromJSONTyped;
exports.OPENAIAuthConfigToJSON = OPENAIAuthConfigToJSON;
exports.OPENAIAuthConfigToJSONTyped = OPENAIAuthConfigToJSONTyped;
/**
 * Check if a given object implements the OPENAIAuthConfig interface.
 */
function instanceOfOPENAIAuthConfig(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    return true;
}
function OPENAIAuthConfigFromJSON(json) {
    return OPENAIAuthConfigFromJSONTyped(json, false);
}
function OPENAIAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
function OPENAIAuthConfigToJSON(json) {
    return OPENAIAuthConfigToJSONTyped(json, false);
}
function OPENAIAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
    };
}
