// @vectorize-io/vectorize-client/dist/models/Pinecone1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPinecone1 = instanceOfPinecone1;
exports.Pinecone1FromJSON = Pinecone1FromJSON;
exports.Pinecone1FromJSONTyped = Pinecone1FromJSONTyped;
exports.Pinecone1ToJSON = Pinecone1ToJSON;
exports.Pinecone1ToJSONTyped = Pinecone1ToJSONTyped;
var PINECONEAuthConfig_1 = require("./PINECONEAuthConfig");
/**
 * Check if a given object implements the Pinecone1 interface.
 */
function instanceOfPinecone1(value) {
    return true;
}
function Pinecone1FromJSON(json) {
    return Pinecone1FromJSONTyped(json, false);
}
function Pinecone1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, PINECONEAuthConfig_1.PINECONEAuthConfigFromJSON)(json['config']),
    };
}
function Pinecone1ToJSON(json) {
    return Pinecone1ToJSONTyped(json, false);
}
function Pinecone1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, PINECONEAuthConfig_1.PINECONEAuthConfigToJSON)(value['config']),
    };
}
