// @vectorize-io/vectorize-client/dist/models/Pinecone.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PineconeTypeEnum = void 0;
exports.instanceOfPinecone = instanceOfPinecone;
exports.PineconeFromJSON = PineconeFromJSON;
exports.PineconeFromJSONTyped = PineconeFromJSONTyped;
exports.PineconeToJSON = PineconeToJSON;
exports.PineconeToJSONTyped = PineconeToJSONTyped;
var PINECONEAuthConfig_1 = require("./PINECONEAuthConfig");
/**
 * @export
 */
exports.PineconeTypeEnum = {
    Pinecone: 'PINECONE'
};
/**
 * Check if a given object implements the Pinecone interface.
 */
function instanceOfPinecone(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function PineconeFromJSON(json) {
    return PineconeFromJSONTyped(json, false);
}
function PineconeFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, PINECONEAuthConfig_1.PINECONEAuthConfigFromJSON)(json['config']),
    };
}
function PineconeToJSON(json) {
    return PineconeToJSONTyped(json, false);
}
function PineconeToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, PINECONEAuthConfig_1.PINECONEAuthConfigToJSON)(value['config']),
    };
}
