// @vectorize-io/vectorize-client/dist/models/PINECONEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPINECONEConfig = instanceOfPINECONEConfig;
exports.PINECONEConfigFromJSON = PINECONEConfigFromJSON;
exports.PINECONEConfigFromJSONTyped = PINECONEConfigFromJSONTyped;
exports.PINECONEConfigToJSON = PINECONEConfigToJSON;
exports.PINECONEConfigToJSONTyped = PINECONEConfigToJSONTyped;
/**
 * Check if a given object implements the PINECONEConfig interface.
 */
function instanceOfPINECONEConfig(value) {
    if (!('index' in value) || value['index'] === undefined)
        return false;
    return true;
}
function PINECONEConfigFromJSON(json) {
    return PINECONEConfigFromJSONTyped(json, false);
}
function PINECONEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'index': json['index'],
        'namespace': json['namespace'] == null ? undefined : json['namespace'],
    };
}
function PINECONEConfigToJSON(json) {
    return PINECONEConfigToJSONTyped(json, false);
}
function PINECONEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'index': value['index'],
        'namespace': value['namespace'],
    };
}
