// @vectorize-io/vectorize-client/dist/models/PINECONEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPINECONEAuthConfig = instanceOfPINECONEAuthConfig;
exports.PINECONEAuthConfigFromJSON = PINECONEAuthConfigFromJSON;
exports.PINECONEAuthConfigFromJSONTyped = PINECONEAuthConfigFromJSONTyped;
exports.PINECONEAuthConfigToJSON = PINECONEAuthConfigToJSON;
exports.PINECONEAuthConfigToJSONTyped = PINECONEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the PINECONEAuthConfig interface.
 */
function instanceOfPINECONEAuthConfig(value) {
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function PINECONEAuthConfigFromJSON(json) {
    return PINECONEAuthConfigFromJSONTyped(json, false);
}
function PINECONEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'apiKey': json['api-key'],
    };
}
function PINECONEAuthConfigToJSON(json) {
    return PINECONEAuthConfigToJSONTyped(json, false);
}
function PINECONEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'api-key': value['apiKey'],
    };
}
