// @vectorize-io/vectorize-client/dist/models/PipelineSummary.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPipelineSummary = instanceOfPipelineSummary;
exports.PipelineSummaryFromJSON = PipelineSummaryFromJSON;
exports.PipelineSummaryFromJSONTyped = PipelineSummaryFromJSONTyped;
exports.PipelineSummaryToJSON = PipelineSummaryToJSON;
exports.PipelineSummaryToJSONTyped = PipelineSummaryToJSONTyped;
var AIPlatformConnector_1 = require("./AIPlatformConnector");
var SourceConnector_1 = require("./SourceConnector");
var DestinationConnector_1 = require("./DestinationConnector");
/**
 * Check if a given object implements the PipelineSummary interface.
 */
function instanceOfPipelineSummary(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('documentCount' in value) || value['documentCount'] === undefined)
        return false;
    if (!('sourceConnectorAuthIds' in value) || value['sourceConnectorAuthIds'] === undefined)
        return false;
    if (!('destinationConnectorAuthIds' in value) || value['destinationConnectorAuthIds'] === undefined)
        return false;
    if (!('aiPlatformAuthIds' in value) || value['aiPlatformAuthIds'] === undefined)
        return false;
    if (!('sourceConnectorTypes' in value) || value['sourceConnectorTypes'] === undefined)
        return false;
    if (!('destinationConnectorTypes' in value) || value['destinationConnectorTypes'] === undefined)
        return false;
    if (!('aiPlatformTypes' in value) || value['aiPlatformTypes'] === undefined)
        return false;
    if (!('createdAt' in value) || value['createdAt'] === undefined)
        return false;
    if (!('createdBy' in value) || value['createdBy'] === undefined)
        return false;
    if (!('sourceConnectors' in value) || value['sourceConnectors'] === undefined)
        return false;
    if (!('destinationConnectors' in value) || value['destinationConnectors'] === undefined)
        return false;
    if (!('aiPlatforms' in value) || value['aiPlatforms'] === undefined)
        return false;
    return true;
}
function PipelineSummaryFromJSON(json) {
    return PipelineSummaryFromJSONTyped(json, false);
}
function PipelineSummaryFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'name': json['name'],
        'documentCount': json['documentCount'],
        'sourceConnectorAuthIds': json['sourceConnectorAuthIds'],
        'destinationConnectorAuthIds': json['destinationConnectorAuthIds'],
        'aiPlatformAuthIds': json['aiPlatformAuthIds'],
        'sourceConnectorTypes': json['sourceConnectorTypes'],
        'destinationConnectorTypes': json['destinationConnectorTypes'],
        'aiPlatformTypes': json['aiPlatformTypes'],
        'createdAt': json['createdAt'],
        'createdBy': json['createdBy'],
        'status': json['status'] == null ? undefined : json['status'],
        'configDoc': json['configDoc'] == null ? undefined : json['configDoc'],
        'sourceConnectors': (json['sourceConnectors'].map(SourceConnector_1.SourceConnectorFromJSON)),
        'destinationConnectors': (json['destinationConnectors'].map(DestinationConnector_1.DestinationConnectorFromJSON)),
        'aiPlatforms': (json['aiPlatforms'].map(AIPlatformConnector_1.AIPlatformConnectorFromJSON)),
    };
}
function PipelineSummaryToJSON(json) {
    return PipelineSummaryToJSONTyped(json, false);
}
function PipelineSummaryToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'name': value['name'],
        'documentCount': value['documentCount'],
        'sourceConnectorAuthIds': value['sourceConnectorAuthIds'],
        'destinationConnectorAuthIds': value['destinationConnectorAuthIds'],
        'aiPlatformAuthIds': value['aiPlatformAuthIds'],
        'sourceConnectorTypes': value['sourceConnectorTypes'],
        'destinationConnectorTypes': value['destinationConnectorTypes'],
        'aiPlatformTypes': value['aiPlatformTypes'],
        'createdAt': value['createdAt'],
        'createdBy': value['createdBy'],
        'status': value['status'],
        'configDoc': value['configDoc'],
        'sourceConnectors': (value['sourceConnectors'].map(SourceConnector_1.SourceConnectorToJSON)),
        'destinationConnectors': (value['destinationConnectors'].map(DestinationConnector_1.DestinationConnectorToJSON)),
        'aiPlatforms': (value['aiPlatforms'].map(AIPlatformConnector_1.AIPlatformConnectorToJSON)),
    };
}
