// @vectorize-io/vectorize-client/dist/models/PipelineMetrics.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPipelineMetrics = instanceOfPipelineMetrics;
exports.PipelineMetricsFromJSON = PipelineMetricsFromJSON;
exports.PipelineMetricsFromJSONTyped = PipelineMetricsFromJSONTyped;
exports.PipelineMetricsToJSON = PipelineMetricsToJSON;
exports.PipelineMetricsToJSONTyped = PipelineMetricsToJSONTyped;
/**
 * Check if a given object implements the PipelineMetrics interface.
 */
function instanceOfPipelineMetrics(value) {
    if (!('timestamp' in value) || value['timestamp'] === undefined)
        return false;
    if (!('newObjects' in value) || value['newObjects'] === undefined)
        return false;
    if (!('changedObjects' in value) || value['changedObjects'] === undefined)
        return false;
    if (!('deletedObjects' in value) || value['deletedObjects'] === undefined)
        return false;
    return true;
}
function PipelineMetricsFromJSON(json) {
    return PipelineMetricsFromJSONTyped(json, false);
}
function PipelineMetricsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'timestamp': json['timestamp'],
        'newObjects': json['newObjects'],
        'changedObjects': json['changedObjects'],
        'deletedObjects': json['deletedObjects'],
    };
}
function PipelineMetricsToJSON(json) {
    return PipelineMetricsToJSONTyped(json, false);
}
function PipelineMetricsToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'timestamp': value['timestamp'],
        'newObjects': value['newObjects'],
        'changedObjects': value['changedObjects'],
        'deletedObjects': value['deletedObjects'],
    };
}
