// @vectorize-io/vectorize-client/dist/models/PipelineDestinationConnectorSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPipelineDestinationConnectorSchema = instanceOfPipelineDestinationConnectorSchema;
exports.PipelineDestinationConnectorSchemaFromJSON = PipelineDestinationConnectorSchemaFromJSON;
exports.PipelineDestinationConnectorSchemaFromJSONTyped = PipelineDestinationConnectorSchemaFromJSONTyped;
exports.PipelineDestinationConnectorSchemaToJSON = PipelineDestinationConnectorSchemaToJSON;
exports.PipelineDestinationConnectorSchemaToJSONTyped = PipelineDestinationConnectorSchemaToJSONTyped;
var DestinationConnectorTypeForPipeline_1 = require("./DestinationConnectorTypeForPipeline");
/**
 * Check if a given object implements the PipelineDestinationConnectorSchema interface.
 */
function instanceOfPipelineDestinationConnectorSchema(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    return true;
}
function PipelineDestinationConnectorSchemaFromJSON(json) {
    return PipelineDestinationConnectorSchemaFromJSONTyped(json, false);
}
function PipelineDestinationConnectorSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'type': (0, DestinationConnectorTypeForPipeline_1.DestinationConnectorTypeForPipelineFromJSON)(json['type']),
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function PipelineDestinationConnectorSchemaToJSON(json) {
    return PipelineDestinationConnectorSchemaToJSONTyped(json, false);
}
function PipelineDestinationConnectorSchemaToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'type': (0, DestinationConnectorTypeForPipeline_1.DestinationConnectorTypeForPipelineToJSON)(value['type']),
        'config': value['config'],
    };
}
