// @vectorize-io/vectorize-client/dist/models/PipelineEvents.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPipelineEvents = instanceOfPipelineEvents;
exports.PipelineEventsFromJSON = PipelineEventsFromJSON;
exports.PipelineEventsFromJSONTyped = PipelineEventsFromJSONTyped;
exports.PipelineEventsToJSON = PipelineEventsToJSON;
exports.PipelineEventsToJSONTyped = PipelineEventsToJSONTyped;
/**
 * Check if a given object implements the PipelineEvents interface.
 */
function instanceOfPipelineEvents(value) {
    if (!('id' in value) || value['id'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('timestamp' in value) || value['timestamp'] === undefined)
        return false;
    return true;
}
function PipelineEventsFromJSON(json) {
    return PipelineEventsFromJSONTyped(json, false);
}
function PipelineEventsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'type': json['type'],
        'timestamp': json['timestamp'],
        'details': json['details'] == null ? undefined : json['details'],
        'summary': json['summary'] == null ? undefined : json['summary'],
    };
}
function PipelineEventsToJSON(json) {
    return PipelineEventsToJSONTyped(json, false);
}
function PipelineEventsToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'type': value['type'],
        'timestamp': value['timestamp'],
        'details': value['details'],
        'summary': value['summary'],
    };
}
