// @vectorize-io/vectorize-client/dist/models/POSTGRESQLConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPOSTGRESQLConfig = instanceOfPOSTGRESQLConfig;
exports.POSTGRESQLConfigFromJSON = POSTGRESQLConfigFromJSON;
exports.POSTGRESQLConfigFromJSONTyped = POSTGRESQLConfigFromJSONTyped;
exports.POSTGRESQLConfigToJSON = POSTGRESQLConfigToJSON;
exports.POSTGRESQLConfigToJSONTyped = POSTGRESQLConfigToJSONTyped;
/**
 * Check if a given object implements the POSTGRESQLConfig interface.
 */
function instanceOfPOSTGRESQLConfig(value) {
    if (!('table' in value) || value['table'] === undefined)
        return false;
    return true;
}
function POSTGRESQLConfigFromJSON(json) {
    return POSTGRESQLConfigFromJSONTyped(json, false);
}
function POSTGRESQLConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'table': json['table'],
    };
}
function POSTGRESQLConfigToJSON(json) {
    return POSTGRESQLConfigToJSONTyped(json, false);
}
function POSTGRESQLConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'table': value['table'],
    };
}
