// @vectorize-io/vectorize-client/dist/models/Postgresql1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPostgresql1 = instanceOfPostgresql1;
exports.Postgresql1FromJSON = Postgresql1FromJSON;
exports.Postgresql1FromJSONTyped = Postgresql1FromJSONTyped;
exports.Postgresql1ToJSON = Postgresql1ToJSON;
exports.Postgresql1ToJSONTyped = Postgresql1ToJSONTyped;
var POSTGRESQLAuthConfig_1 = require("./POSTGRESQLAuthConfig");
/**
 * Check if a given object implements the Postgresql1 interface.
 */
function instanceOfPostgresql1(value) {
    return true;
}
function Postgresql1FromJSON(json) {
    return Postgresql1FromJSONTyped(json, false);
}
function Postgresql1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, POSTGRESQLAuthConfig_1.POSTGRESQLAuthConfigFromJSON)(json['config']),
    };
}
function Postgresql1ToJSON(json) {
    return Postgresql1ToJSONTyped(json, false);
}
function Postgresql1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, POSTGRESQLAuthConfig_1.POSTGRESQLAuthConfigToJSON)(value['config']),
    };
}
