// @vectorize-io/vectorize-client/dist/models/Postgresql.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostgresqlTypeEnum = void 0;
exports.instanceOfPostgresql = instanceOfPostgresql;
exports.PostgresqlFromJSON = PostgresqlFromJSON;
exports.PostgresqlFromJSONTyped = PostgresqlFromJSONTyped;
exports.PostgresqlToJSON = PostgresqlToJSON;
exports.PostgresqlToJSONTyped = PostgresqlToJSONTyped;
var POSTGRESQLAuthConfig_1 = require("./POSTGRESQLAuthConfig");
/**
 * @export
 */
exports.PostgresqlTypeEnum = {
    Postgresql: 'POSTGRESQL'
};
/**
 * Check if a given object implements the Postgresql interface.
 */
function instanceOfPostgresql(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function PostgresqlFromJSON(json) {
    return PostgresqlFromJSONTyped(json, false);
}
function PostgresqlFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, POSTGRESQLAuthConfig_1.POSTGRESQLAuthConfigFromJSON)(json['config']),
    };
}
function PostgresqlToJSON(json) {
    return PostgresqlToJSONTyped(json, false);
}
function PostgresqlToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, POSTGRESQLAuthConfig_1.POSTGRESQLAuthConfigToJSON)(value['config']),
    };
}
