// @vectorize-io/vectorize-client/dist/models/POSTGRESQLAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPOSTGRESQLAuthConfig = instanceOfPOSTGRESQLAuthConfig;
exports.POSTGRESQLAuthConfigFromJSON = POSTGRESQLAuthConfigFromJSON;
exports.POSTGRESQLAuthConfigFromJSONTyped = POSTGRESQLAuthConfigFromJSONTyped;
exports.POSTGRESQLAuthConfigToJSON = POSTGRESQLAuthConfigToJSON;
exports.POSTGRESQLAuthConfigToJSONTyped = POSTGRESQLAuthConfigToJSONTyped;
/**
 * Check if a given object implements the POSTGRESQLAuthConfig interface.
 */
function instanceOfPOSTGRESQLAuthConfig(value) {
    if (!('host' in value) || value['host'] === undefined)
        return false;
    if (!('database' in value) || value['database'] === undefined)
        return false;
    if (!('username' in value) || value['username'] === undefined)
        return false;
    if (!('password' in value) || value['password'] === undefined)
        return false;
    return true;
}
function POSTGRESQLAuthConfigFromJSON(json) {
    return POSTGRESQLAuthConfigFromJSONTyped(json, false);
}
function POSTGRESQLAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'host': json['host'],
        'port': json['port'] == null ? undefined : json['port'],
        'database': json['database'],
        'username': json['username'],
        'password': json['password'],
    };
}
function POSTGRESQLAuthConfigToJSON(json) {
    return POSTGRESQLAuthConfigToJSONTyped(json, false);
}
function POSTGRESQLAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'host': value['host'],
        'port': value['port'],
        'database': value['database'],
        'username': value['username'],
        'password': value['password'],
    };
}
