// @vectorize-io/vectorize-client/dist/models/QDRANTConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQDRANTConfig = instanceOfQDRANTConfig;
exports.QDRANTConfigFromJSON = QDRANTConfigFromJSON;
exports.QDRANTConfigFromJSONTyped = QDRANTConfigFromJSONTyped;
exports.QDRANTConfigToJSON = QDRANTConfigToJSON;
exports.QDRANTConfigToJSONTyped = QDRANTConfigToJSONTyped;
/**
 * Check if a given object implements the QDRANTConfig interface.
 */
function instanceOfQDRANTConfig(value) {
    if (!('collection' in value) || value['collection'] === undefined)
        return false;
    return true;
}
function QDRANTConfigFromJSON(json) {
    return QDRANTConfigFromJSONTyped(json, false);
}
function QDRANTConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'collection': json['collection'],
    };
}
function QDRANTConfigToJSON(json) {
    return QDRANTConfigToJSONTyped(json, false);
}
function QDRANTConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'collection': value['collection'],
    };
}
