// @vectorize-io/vectorize-client/dist/models/Qdrant1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQdrant1 = instanceOfQdrant1;
exports.Qdrant1FromJSON = Qdrant1FromJSON;
exports.Qdrant1FromJSONTyped = Qdrant1FromJSONTyped;
exports.Qdrant1ToJSON = Qdrant1ToJSON;
exports.Qdrant1ToJSONTyped = Qdrant1ToJSONTyped;
var QDRANTAuthConfig_1 = require("./QDRANTAuthConfig");
/**
 * Check if a given object implements the Qdrant1 interface.
 */
function instanceOfQdrant1(value) {
    return true;
}
function Qdrant1FromJSON(json) {
    return Qdrant1FromJSONTyped(json, false);
}
function Qdrant1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, QDRANTAuthConfig_1.QDRANTAuthConfigFromJSON)(json['config']),
    };
}
function Qdrant1ToJSON(json) {
    return Qdrant1ToJSONTyped(json, false);
}
function Qdrant1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, QDRANTAuthConfig_1.QDRANTAuthConfigToJSON)(value['config']),
    };
}
