// @vectorize-io/vectorize-client/dist/models/Qdrant.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QdrantTypeEnum = void 0;
exports.instanceOfQdrant = instanceOfQdrant;
exports.QdrantFromJSON = QdrantFromJSON;
exports.QdrantFromJSONTyped = QdrantFromJSONTyped;
exports.QdrantToJSON = QdrantToJSON;
exports.QdrantToJSONTyped = QdrantToJSONTyped;
var QDRANTAuthConfig_1 = require("./QDRANTAuthConfig");
/**
 * @export
 */
exports.QdrantTypeEnum = {
    Qdrant: 'QDRANT'
};
/**
 * Check if a given object implements the Qdrant interface.
 */
function instanceOfQdrant(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function QdrantFromJSON(json) {
    return QdrantFromJSONTyped(json, false);
}
function QdrantFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, QDRANTAuthConfig_1.QDRANTAuthConfigFromJSON)(json['config']),
    };
}
function QdrantToJSON(json) {
    return QdrantToJSONTyped(json, false);
}
function QdrantToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, QDRANTAuthConfig_1.QDRANTAuthConfigToJSON)(value['config']),
    };
}
