// @vectorize-io/vectorize-client/dist/models/QDRANTAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQDRANTAuthConfig = instanceOfQDRANTAuthConfig;
exports.QDRANTAuthConfigFromJSON = QDRANTAuthConfigFromJSON;
exports.QDRANTAuthConfigFromJSONTyped = QDRANTAuthConfigFromJSONTyped;
exports.QDRANTAuthConfigToJSON = QDRANTAuthConfigToJSON;
exports.QDRANTAuthConfigToJSONTyped = QDRANTAuthConfigToJSONTyped;
/**
 * Check if a given object implements the QDRANTAuthConfig interface.
 */
function instanceOfQDRANTAuthConfig(value) {
    if (!('host' in value) || value['host'] === undefined)
        return false;
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function QDRANTAuthConfigFromJSON(json) {
    return QDRANTAuthConfigFromJSONTyped(json, false);
}
function QDRANTAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'host': json['host'],
        'apiKey': json['api-key'],
    };
}
function QDRANTAuthConfigToJSON(json) {
    return QDRANTAuthConfigToJSONTyped(json, false);
}
function QDRANTAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'host': value['host'],
        'api-key': value['apiKey'],
    };
}
