// @vectorize-io/vectorize-client/dist/models/RemoveUserFromSourceConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRemoveUserFromSourceConnectorRequest = instanceOfRemoveUserFromSourceConnectorRequest;
exports.RemoveUserFromSourceConnectorRequestFromJSON = RemoveUserFromSourceConnectorRequestFromJSON;
exports.RemoveUserFromSourceConnectorRequestFromJSONTyped = RemoveUserFromSourceConnectorRequestFromJSONTyped;
exports.RemoveUserFromSourceConnectorRequestToJSON = RemoveUserFromSourceConnectorRequestToJSON;
exports.RemoveUserFromSourceConnectorRequestToJSONTyped = RemoveUserFromSourceConnectorRequestToJSONTyped;
/**
 * Check if a given object implements the RemoveUserFromSourceConnectorRequest interface.
 */
function instanceOfRemoveUserFromSourceConnectorRequest(value) {
    if (!('userId' in value) || value['userId'] === undefined)
        return false;
    return true;
}
function RemoveUserFromSourceConnectorRequestFromJSON(json) {
    return RemoveUserFromSourceConnectorRequestFromJSONTyped(json, false);
}
function RemoveUserFromSourceConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'userId': json['userId'],
    };
}
function RemoveUserFromSourceConnectorRequestToJSON(json) {
    return RemoveUserFromSourceConnectorRequestToJSONTyped(json, false);
}
function RemoveUserFromSourceConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'userId': value['userId'],
    };
}
