// @vectorize-io/vectorize-client/dist/models/RemoveUserFromSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRemoveUserFromSourceConnectorResponse = instanceOfRemoveUserFromSourceConnectorResponse;
exports.RemoveUserFromSourceConnectorResponseFromJSON = RemoveUserFromSourceConnectorResponseFromJSON;
exports.RemoveUserFromSourceConnectorResponseFromJSONTyped = RemoveUserFromSourceConnectorResponseFromJSONTyped;
exports.RemoveUserFromSourceConnectorResponseToJSON = RemoveUserFromSourceConnectorResponseToJSON;
exports.RemoveUserFromSourceConnectorResponseToJSONTyped = RemoveUserFromSourceConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the RemoveUserFromSourceConnectorResponse interface.
 */
function instanceOfRemoveUserFromSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function RemoveUserFromSourceConnectorResponseFromJSON(json) {
    return RemoveUserFromSourceConnectorResponseFromJSONTyped(json, false);
}
function RemoveUserFromSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function RemoveUserFromSourceConnectorResponseToJSON(json) {
    return RemoveUserFromSourceConnectorResponseToJSONTyped(json, false);
}
function RemoveUserFromSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
