// @vectorize-io/vectorize-client/dist/models/RetrieveContext.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRetrieveContext = instanceOfRetrieveContext;
exports.RetrieveContextFromJSON = RetrieveContextFromJSON;
exports.RetrieveContextFromJSONTyped = RetrieveContextFromJSONTyped;
exports.RetrieveContextToJSON = RetrieveContextToJSON;
exports.RetrieveContextToJSONTyped = RetrieveContextToJSONTyped;
var RetrieveContextMessage_1 = require("./RetrieveContextMessage");
/**
 * Check if a given object implements the RetrieveContext interface.
 */
function instanceOfRetrieveContext(value) {
    if (!('messages' in value) || value['messages'] === undefined)
        return false;
    return true;
}
function RetrieveContextFromJSON(json) {
    return RetrieveContextFromJSONTyped(json, false);
}
function RetrieveContextFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'messages': (json['messages'].map(RetrieveContextMessage_1.RetrieveContextMessageFromJSON)),
    };
}
function RetrieveContextToJSON(json) {
    return RetrieveContextToJSONTyped(json, false);
}
function RetrieveContextToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'messages': (value['messages'].map(RetrieveContextMessage_1.RetrieveContextMessageToJSON)),
    };
}
