// @vectorize-io/vectorize-client/dist/models/RetrieveDocumentsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRetrieveDocumentsResponse = instanceOfRetrieveDocumentsResponse;
exports.RetrieveDocumentsResponseFromJSON = RetrieveDocumentsResponseFromJSON;
exports.RetrieveDocumentsResponseFromJSONTyped = RetrieveDocumentsResponseFromJSONTyped;
exports.RetrieveDocumentsResponseToJSON = RetrieveDocumentsResponseToJSON;
exports.RetrieveDocumentsResponseToJSONTyped = RetrieveDocumentsResponseToJSONTyped;
var Document_1 = require("./Document");
/**
 * Check if a given object implements the RetrieveDocumentsResponse interface.
 */
function instanceOfRetrieveDocumentsResponse(value) {
    if (!('question' in value) || value['question'] === undefined)
        return false;
    if (!('documents' in value) || value['documents'] === undefined)
        return false;
    if (!('averageRelevancy' in value) || value['averageRelevancy'] === undefined)
        return false;
    if (!('ndcg' in value) || value['ndcg'] === undefined)
        return false;
    return true;
}
function RetrieveDocumentsResponseFromJSON(json) {
    return RetrieveDocumentsResponseFromJSONTyped(json, false);
}
function RetrieveDocumentsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'question': json['question'],
        'documents': (json['documents'].map(Document_1.DocumentFromJSON)),
        'averageRelevancy': json['average_relevancy'],
        'ndcg': json['ndcg'],
    };
}
function RetrieveDocumentsResponseToJSON(json) {
    return RetrieveDocumentsResponseToJSONTyped(json, false);
}
function RetrieveDocumentsResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'question': value['question'],
        'documents': (value['documents'].map(Document_1.DocumentToJSON)),
        'average_relevancy': value['averageRelevancy'],
        'ndcg': value['ndcg'],
    };
}
