// @vectorize-io/vectorize-client/dist/models/ScheduleSchemaType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleSchemaType = void 0;
exports.instanceOfScheduleSchemaType = instanceOfScheduleSchemaType;
exports.ScheduleSchemaTypeFromJSON = ScheduleSchemaTypeFromJSON;
exports.ScheduleSchemaTypeFromJSONTyped = ScheduleSchemaTypeFromJSONTyped;
exports.ScheduleSchemaTypeToJSON = ScheduleSchemaTypeToJSON;
exports.ScheduleSchemaTypeToJSONTyped = ScheduleSchemaTypeToJSONTyped;
/**
 *
 * @export
 */
exports.ScheduleSchemaType = {
    Manual: 'manual',
    Realtime: 'realtime',
    Custom: 'custom'
};
function instanceOfScheduleSchemaType(value) {
    for (var key in exports.ScheduleSchemaType) {
        if (Object.prototype.hasOwnProperty.call(exports.ScheduleSchemaType, key)) {
            if (exports.ScheduleSchemaType[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function ScheduleSchemaTypeFromJSON(json) {
    return ScheduleSchemaTypeFromJSONTyped(json, false);
}
function ScheduleSchemaTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function ScheduleSchemaTypeToJSON(value) {
    return value;
}
function ScheduleSchemaTypeToJSONTyped(value, ignoreDiscriminator) {
    return value;
}
