// @vectorize-io/vectorize-client/dist/models/ScheduleSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfScheduleSchema = instanceOfScheduleSchema;
exports.ScheduleSchemaFromJSON = ScheduleSchemaFromJSON;
exports.ScheduleSchemaFromJSONTyped = ScheduleSchemaFromJSONTyped;
exports.ScheduleSchemaToJSON = ScheduleSchemaToJSON;
exports.ScheduleSchemaToJSONTyped = ScheduleSchemaToJSONTyped;
var ScheduleSchemaType_1 = require("./ScheduleSchemaType");
/**
 * Check if a given object implements the ScheduleSchema interface.
 */
function instanceOfScheduleSchema(value) {
    if (!('type' in value) || value['type'] === undefined)
        return false;
    return true;
}
function ScheduleSchemaFromJSON(json) {
    return ScheduleSchemaFromJSONTyped(json, false);
}
function ScheduleSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'type': (0, ScheduleSchemaType_1.ScheduleSchemaTypeFromJSON)(json['type']),
    };
}
function ScheduleSchemaToJSON(json) {
    return ScheduleSchemaToJSONTyped(json, false);
}
function ScheduleSchemaToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'type': (0, ScheduleSchemaType_1.ScheduleSchemaTypeToJSON)(value['type']),
    };
}
