// @vectorize-io/vectorize-client/dist/models/Sharepoint.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharepointTypeEnum = void 0;
exports.instanceOfSharepoint = instanceOfSharepoint;
exports.SharepointFromJSON = SharepointFromJSON;
exports.SharepointFromJSONTyped = SharepointFromJSONTyped;
exports.SharepointToJSON = SharepointToJSON;
exports.SharepointToJSONTyped = SharepointToJSONTyped;
var SHAREPOINTAuthConfig_1 = require("./SHAREPOINTAuthConfig");
/**
 * @export
 */
exports.SharepointTypeEnum = {
    Sharepoint: 'SHAREPOINT'
};
/**
 * Check if a given object implements the Sharepoint interface.
 */
function instanceOfSharepoint(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function SharepointFromJSON(json) {
    return SharepointFromJSONTyped(json, false);
}
function SharepointFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, SHAREPOINTAuthConfig_1.SHAREPOINTAuthConfigFromJSON)(json['config']),
    };
}
function SharepointToJSON(json) {
    return SharepointToJSONTyped(json, false);
}
function SharepointToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, SHAREPOINTAuthConfig_1.SHAREPOINTAuthConfigToJSON)(value['config']),
    };
}
