// @vectorize-io/vectorize-client/dist/models/SHAREPOINTConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SHAREPOINTConfigFileExtensionsEnum = void 0;
exports.instanceOfSHAREPOINTConfig = instanceOfSHAREPOINTConfig;
exports.SHAREPOINTConfigFromJSON = SHAREPOINTConfigFromJSON;
exports.SHAREPOINTConfigFromJSONTyped = SHAREPOINTConfigFromJSONTyped;
exports.SHAREPOINTConfigToJSON = SHAREPOINTConfigToJSON;
exports.SHAREPOINTConfigToJSONTyped = SHAREPOINTConfigToJSONTyped;
/**
 * @export
 */
exports.SHAREPOINTConfigFileExtensionsEnum = {
    Pdf: 'pdf',
    Docdocxgdocodtrtfepub: 'doc,docx,gdoc,odt,rtf,epub',
    Pptpptxgslides: 'ppt,pptx,gslides',
    Xlsxlsxgsheetsods: 'xls,xlsx,gsheets,ods',
    Emlmsg: 'eml,msg',
    Txt: 'txt',
    Htmlhtm: 'html,htm',
    Jpgjpegpngwebpsvggif: 'jpg,jpeg,png,webp,svg,gif',
    Json: 'json',
    Csv: 'csv'
};
/**
 * Check if a given object implements the SHAREPOINTConfig interface.
 */
function instanceOfSHAREPOINTConfig(value) {
    if (!('fileExtensions' in value) || value['fileExtensions'] === undefined)
        return false;
    return true;
}
function SHAREPOINTConfigFromJSON(json) {
    return SHAREPOINTConfigFromJSONTyped(json, false);
}
function SHAREPOINTConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileExtensions': json['file-extensions'],
        'sites': json['sites'] == null ? undefined : json['sites'],
        'folderPath': json['folder-path'] == null ? undefined : json['folder-path'],
    };
}
function SHAREPOINTConfigToJSON(json) {
    return SHAREPOINTConfigToJSONTyped(json, false);
}
function SHAREPOINTConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'file-extensions': value['fileExtensions'],
        'sites': value['sites'],
        'folder-path': value['folderPath'],
    };
}
