// @vectorize-io/vectorize-client/dist/models/Sharepoint1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSharepoint1 = instanceOfSharepoint1;
exports.Sharepoint1FromJSON = Sharepoint1FromJSON;
exports.Sharepoint1FromJSONTyped = Sharepoint1FromJSONTyped;
exports.Sharepoint1ToJSON = Sharepoint1ToJSON;
exports.Sharepoint1ToJSONTyped = Sharepoint1ToJSONTyped;
var SHAREPOINTAuthConfig_1 = require("./SHAREPOINTAuthConfig");
/**
 * Check if a given object implements the Sharepoint1 interface.
 */
function instanceOfSharepoint1(value) {
    return true;
}
function Sharepoint1FromJSON(json) {
    return Sharepoint1FromJSONTyped(json, false);
}
function Sharepoint1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, SHAREPOINTAuthConfig_1.SHAREPOINTAuthConfigFromJSON)(json['config']),
    };
}
function Sharepoint1ToJSON(json) {
    return Sharepoint1ToJSONTyped(json, false);
}
function Sharepoint1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, SHAREPOINTAuthConfig_1.SHAREPOINTAuthConfigToJSON)(value['config']),
    };
}
