// @vectorize-io/vectorize-client/dist/models/SHAREPOINTAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSHAREPOINTAuthConfig = instanceOfSHAREPOINTAuthConfig;
exports.SHAREPOINTAuthConfigFromJSON = SHAREPOINTAuthConfigFromJSON;
exports.SHAREPOINTAuthConfigFromJSONTyped = SHAREPOINTAuthConfigFromJSONTyped;
exports.SHAREPOINTAuthConfigToJSON = SHAREPOINTAuthConfigToJSON;
exports.SHAREPOINTAuthConfigToJSONTyped = SHAREPOINTAuthConfigToJSONTyped;
/**
 * Check if a given object implements the SHAREPOINTAuthConfig interface.
 */
function instanceOfSHAREPOINTAuthConfig(value) {
    if (!('msClientId' in value) || value['msClientId'] === undefined)
        return false;
    if (!('msTenantId' in value) || value['msTenantId'] === undefined)
        return false;
    if (!('msClientSecret' in value) || value['msClientSecret'] === undefined)
        return false;
    return true;
}
function SHAREPOINTAuthConfigFromJSON(json) {
    return SHAREPOINTAuthConfigFromJSONTyped(json, false);
}
function SHAREPOINTAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'msClientId': json['ms-client-id'],
        'msTenantId': json['ms-tenant-id'],
        'msClientSecret': json['ms-client-secret'],
    };
}
function SHAREPOINTAuthConfigToJSON(json) {
    return SHAREPOINTAuthConfigToJSONTyped(json, false);
}
function SHAREPOINTAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'ms-client-id': value['msClientId'],
        'ms-tenant-id': value['msTenantId'],
        'ms-client-secret': value['msClientSecret'],
    };
}
