// @vectorize-io/vectorize-client/dist/models/Singlestore.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SinglestoreTypeEnum = void 0;
exports.instanceOfSinglestore = instanceOfSinglestore;
exports.SinglestoreFromJSON = SinglestoreFromJSON;
exports.SinglestoreFromJSONTyped = SinglestoreFromJSONTyped;
exports.SinglestoreToJSON = SinglestoreToJSON;
exports.SinglestoreToJSONTyped = SinglestoreToJSONTyped;
var SINGLESTOREAuthConfig_1 = require("./SINGLESTOREAuthConfig");
/**
 * @export
 */
exports.SinglestoreTypeEnum = {
    Singlestore: 'SINGLESTORE'
};
/**
 * Check if a given object implements the Singlestore interface.
 */
function instanceOfSinglestore(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function SinglestoreFromJSON(json) {
    return SinglestoreFromJSONTyped(json, false);
}
function SinglestoreFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, SINGLESTOREAuthConfig_1.SINGLESTOREAuthConfigFromJSON)(json['config']),
    };
}
function SinglestoreToJSON(json) {
    return SinglestoreToJSONTyped(json, false);
}
function SinglestoreToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, SINGLESTOREAuthConfig_1.SINGLESTOREAuthConfigToJSON)(value['config']),
    };
}
