// @vectorize-io/vectorize-client/dist/models/Singlestore1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSinglestore1 = instanceOfSinglestore1;
exports.Singlestore1FromJSON = Singlestore1FromJSON;
exports.Singlestore1FromJSONTyped = Singlestore1FromJSONTyped;
exports.Singlestore1ToJSON = Singlestore1ToJSON;
exports.Singlestore1ToJSONTyped = Singlestore1ToJSONTyped;
var SINGLESTOREAuthConfig_1 = require("./SINGLESTOREAuthConfig");
/**
 * Check if a given object implements the Singlestore1 interface.
 */
function instanceOfSinglestore1(value) {
    return true;
}
function Singlestore1FromJSON(json) {
    return Singlestore1FromJSONTyped(json, false);
}
function Singlestore1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, SINGLESTOREAuthConfig_1.SINGLESTOREAuthConfigFromJSON)(json['config']),
    };
}
function Singlestore1ToJSON(json) {
    return Singlestore1ToJSONTyped(json, false);
}
function Singlestore1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, SINGLESTOREAuthConfig_1.SINGLESTOREAuthConfigToJSON)(value['config']),
    };
}
