// @vectorize-io/vectorize-client/dist/models/SINGLESTOREConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSINGLESTOREConfig = instanceOfSINGLESTOREConfig;
exports.SINGLESTOREConfigFromJSON = SINGLESTOREConfigFromJSON;
exports.SINGLESTOREConfigFromJSONTyped = SINGLESTOREConfigFromJSONTyped;
exports.SINGLESTOREConfigToJSON = SINGLESTOREConfigToJSON;
exports.SINGLESTOREConfigToJSONTyped = SINGLESTOREConfigToJSONTyped;
/**
 * Check if a given object implements the SINGLESTOREConfig interface.
 */
function instanceOfSINGLESTOREConfig(value) {
    if (!('table' in value) || value['table'] === undefined)
        return false;
    return true;
}
function SINGLESTOREConfigFromJSON(json) {
    return SINGLESTOREConfigFromJSONTyped(json, false);
}
function SINGLESTOREConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'table': json['table'],
    };
}
function SINGLESTOREConfigToJSON(json) {
    return SINGLESTOREConfigToJSONTyped(json, false);
}
function SINGLESTOREConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'table': value['table'],
    };
}
