// @vectorize-io/vectorize-client/dist/models/SourceConnectorInputConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceConnectorInputConfigFromJSON = SourceConnectorInputConfigFromJSON;
exports.SourceConnectorInputConfigFromJSONTyped = SourceConnectorInputConfigFromJSONTyped;
exports.SourceConnectorInputConfigToJSON = SourceConnectorInputConfigToJSON;
exports.SourceConnectorInputConfigToJSONTyped = SourceConnectorInputConfigToJSONTyped;
var AWSS3Config_1 = require("./AWSS3Config");
var AZUREBLOBConfig_1 = require("./AZUREBLOBConfig");
var CONFLUENCEConfig_1 = require("./CONFLUENCEConfig");
var DISCORDConfig_1 = require("./DISCORDConfig");
var DOCUSIGNConfig_1 = require("./DOCUSIGNConfig");
var DROPBOXConfig_1 = require("./DROPBOXConfig");
var FIRECRAWLConfig_1 = require("./FIRECRAWLConfig");
var FIREFLIESConfig_1 = require("./FIREFLIESConfig");
var GCSConfig_1 = require("./GCSConfig");
var GITHUBConfig_1 = require("./GITHUBConfig");
var GMAILConfig_1 = require("./GMAILConfig");
var GOOGLEDRIVEConfig_1 = require("./GOOGLEDRIVEConfig");
var GOOGLEDRIVEOAUTHConfig_1 = require("./GOOGLEDRIVEOAUTHConfig");
var GOOGLEDRIVEOAUTHMULTICUSTOMConfig_1 = require("./GOOGLEDRIVEOAUTHMULTICUSTOMConfig");
var GOOGLEDRIVEOAUTHMULTIConfig_1 = require("./GOOGLEDRIVEOAUTHMULTIConfig");
var INTERCOMConfig_1 = require("./INTERCOMConfig");
var NOTIONConfig_1 = require("./NOTIONConfig");
var ONEDRIVEConfig_1 = require("./ONEDRIVEConfig");
var SHAREPOINTConfig_1 = require("./SHAREPOINTConfig");
var WEBCRAWLERConfig_1 = require("./WEBCRAWLERConfig");
var ZOOMADMINConfig_1 = require("./ZOOMADMINConfig");
var ZOOMConfig_1 = require("./ZOOMConfig");
function SourceConnectorInputConfigFromJSON(json) {
    return SourceConnectorInputConfigFromJSONTyped(json, false);
}
function SourceConnectorInputConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    if ((0, AWSS3Config_1.instanceOfAWSS3Config)(json)) {
        return (0, AWSS3Config_1.AWSS3ConfigFromJSONTyped)(json, true);
    }
    if ((0, AZUREBLOBConfig_1.instanceOfAZUREBLOBConfig)(json)) {
        return (0, AZUREBLOBConfig_1.AZUREBLOBConfigFromJSONTyped)(json, true);
    }
    if ((0, CONFLUENCEConfig_1.instanceOfCONFLUENCEConfig)(json)) {
        return (0, CONFLUENCEConfig_1.CONFLUENCEConfigFromJSONTyped)(json, true);
    }
    if ((0, DISCORDConfig_1.instanceOfDISCORDConfig)(json)) {
        return (0, DISCORDConfig_1.DISCORDConfigFromJSONTyped)(json, true);
    }
    if ((0, DOCUSIGNConfig_1.instanceOfDOCUSIGNConfig)(json)) {
        return (0, DOCUSIGNConfig_1.DOCUSIGNConfigFromJSONTyped)(json, true);
    }
    if ((0, DROPBOXConfig_1.instanceOfDROPBOXConfig)(json)) {
        return (0, DROPBOXConfig_1.DROPBOXConfigFromJSONTyped)(json, true);
    }
    if ((0, FIRECRAWLConfig_1.instanceOfFIRECRAWLConfig)(json)) {
        return (0, FIRECRAWLConfig_1.FIRECRAWLConfigFromJSONTyped)(json, true);
    }
    if ((0, FIREFLIESConfig_1.instanceOfFIREFLIESConfig)(json)) {
        return (0, FIREFLIESConfig_1.FIREFLIESConfigFromJSONTyped)(json, true);
    }
    if ((0, GCSConfig_1.instanceOfGCSConfig)(json)) {
        return (0, GCSConfig_1.GCSConfigFromJSONTyped)(json, true);
    }
    if ((0, GITHUBConfig_1.instanceOfGITHUBConfig)(json)) {
        return (0, GITHUBConfig_1.GITHUBConfigFromJSONTyped)(json, true);
    }
    if ((0, GMAILConfig_1.instanceOfGMAILConfig)(json)) {
        return (0, GMAILConfig_1.GMAILConfigFromJSONTyped)(json, true);
    }
    if ((0, GOOGLEDRIVEConfig_1.instanceOfGOOGLEDRIVEConfig)(json)) {
        return (0, GOOGLEDRIVEConfig_1.GOOGLEDRIVEConfigFromJSONTyped)(json, true);
    }
    if ((0, GOOGLEDRIVEOAUTHConfig_1.instanceOfGOOGLEDRIVEOAUTHConfig)(json)) {
        return (0, GOOGLEDRIVEOAUTHConfig_1.GOOGLEDRIVEOAUTHConfigFromJSONTyped)(json, true);
    }
    if ((0, GOOGLEDRIVEOAUTHMULTICUSTOMConfig_1.instanceOfGOOGLEDRIVEOAUTHMULTICUSTOMConfig)(json)) {
        return (0, GOOGLEDRIVEOAUTHMULTICUSTOMConfig_1.GOOGLEDRIVEOAUTHMULTICUSTOMConfigFromJSONTyped)(json, true);
    }
    if ((0, GOOGLEDRIVEOAUTHMULTIConfig_1.instanceOfGOOGLEDRIVEOAUTHMULTIConfig)(json)) {
        return (0, GOOGLEDRIVEOAUTHMULTIConfig_1.GOOGLEDRIVEOAUTHMULTIConfigFromJSONTyped)(json, true);
    }
    if ((0, INTERCOMConfig_1.instanceOfINTERCOMConfig)(json)) {
        return (0, INTERCOMConfig_1.INTERCOMConfigFromJSONTyped)(json, true);
    }
    if ((0, NOTIONConfig_1.instanceOfNOTIONConfig)(json)) {
        return (0, NOTIONConfig_1.NOTIONConfigFromJSONTyped)(json, true);
    }
    if ((0, ONEDRIVEConfig_1.instanceOfONEDRIVEConfig)(json)) {
        return (0, ONEDRIVEConfig_1.ONEDRIVEConfigFromJSONTyped)(json, true);
    }
    if ((0, SHAREPOINTConfig_1.instanceOfSHAREPOINTConfig)(json)) {
        return (0, SHAREPOINTConfig_1.SHAREPOINTConfigFromJSONTyped)(json, true);
    }
    if ((0, WEBCRAWLERConfig_1.instanceOfWEBCRAWLERConfig)(json)) {
        return (0, WEBCRAWLERConfig_1.WEBCRAWLERConfigFromJSONTyped)(json, true);
    }
    if ((0, ZOOMADMINConfig_1.instanceOfZOOMADMINConfig)(json)) {
        return (0, ZOOMADMINConfig_1.ZOOMADMINConfigFromJSONTyped)(json, true);
    }
    if ((0, ZOOMConfig_1.instanceOfZOOMConfig)(json)) {
        return (0, ZOOMConfig_1.ZOOMConfigFromJSONTyped)(json, true);
    }
    return {};
}
function SourceConnectorInputConfigToJSON(json) {
    return SourceConnectorInputConfigToJSONTyped(json, false);
}
function SourceConnectorInputConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    if ((0, AWSS3Config_1.instanceOfAWSS3Config)(value)) {
        return (0, AWSS3Config_1.AWSS3ConfigToJSON)(value);
    }
    if ((0, AZUREBLOBConfig_1.instanceOfAZUREBLOBConfig)(value)) {
        return (0, AZUREBLOBConfig_1.AZUREBLOBConfigToJSON)(value);
    }
    if ((0, CONFLUENCEConfig_1.instanceOfCONFLUENCEConfig)(value)) {
        return (0, CONFLUENCEConfig_1.CONFLUENCEConfigToJSON)(value);
    }
    if ((0, DISCORDConfig_1.instanceOfDISCORDConfig)(value)) {
        return (0, DISCORDConfig_1.DISCORDConfigToJSON)(value);
    }
    if ((0, DOCUSIGNConfig_1.instanceOfDOCUSIGNConfig)(value)) {
        return (0, DOCUSIGNConfig_1.DOCUSIGNConfigToJSON)(value);
    }
    if ((0, DROPBOXConfig_1.instanceOfDROPBOXConfig)(value)) {
        return (0, DROPBOXConfig_1.DROPBOXConfigToJSON)(value);
    }
    if ((0, FIRECRAWLConfig_1.instanceOfFIRECRAWLConfig)(value)) {
        return (0, FIRECRAWLConfig_1.FIRECRAWLConfigToJSON)(value);
    }
    if ((0, FIREFLIESConfig_1.instanceOfFIREFLIESConfig)(value)) {
        return (0, FIREFLIESConfig_1.FIREFLIESConfigToJSON)(value);
    }
    if ((0, GCSConfig_1.instanceOfGCSConfig)(value)) {
        return (0, GCSConfig_1.GCSConfigToJSON)(value);
    }
    if ((0, GITHUBConfig_1.instanceOfGITHUBConfig)(value)) {
        return (0, GITHUBConfig_1.GITHUBConfigToJSON)(value);
    }
    if ((0, GMAILConfig_1.instanceOfGMAILConfig)(value)) {
        return (0, GMAILConfig_1.GMAILConfigToJSON)(value);
    }
    if ((0, GOOGLEDRIVEConfig_1.instanceOfGOOGLEDRIVEConfig)(value)) {
        return (0, GOOGLEDRIVEConfig_1.GOOGLEDRIVEConfigToJSON)(value);
    }
    if ((0, GOOGLEDRIVEOAUTHConfig_1.instanceOfGOOGLEDRIVEOAUTHConfig)(value)) {
        return (0, GOOGLEDRIVEOAUTHConfig_1.GOOGLEDRIVEOAUTHConfigToJSON)(value);
    }
    if ((0, GOOGLEDRIVEOAUTHMULTICUSTOMConfig_1.instanceOfGOOGLEDRIVEOAUTHMULTICUSTOMConfig)(value)) {
        return (0, GOOGLEDRIVEOAUTHMULTICUSTOMConfig_1.GOOGLEDRIVEOAUTHMULTICUSTOMConfigToJSON)(value);
    }
    if ((0, GOOGLEDRIVEOAUTHMULTIConfig_1.instanceOfGOOGLEDRIVEOAUTHMULTIConfig)(value)) {
        return (0, GOOGLEDRIVEOAUTHMULTIConfig_1.GOOGLEDRIVEOAUTHMULTIConfigToJSON)(value);
    }
    if ((0, INTERCOMConfig_1.instanceOfINTERCOMConfig)(value)) {
        return (0, INTERCOMConfig_1.INTERCOMConfigToJSON)(value);
    }
    if ((0, NOTIONConfig_1.instanceOfNOTIONConfig)(value)) {
        return (0, NOTIONConfig_1.NOTIONConfigToJSON)(value);
    }
    if ((0, ONEDRIVEConfig_1.instanceOfONEDRIVEConfig)(value)) {
        return (0, ONEDRIVEConfig_1.ONEDRIVEConfigToJSON)(value);
    }
    if ((0, SHAREPOINTConfig_1.instanceOfSHAREPOINTConfig)(value)) {
        return (0, SHAREPOINTConfig_1.SHAREPOINTConfigToJSON)(value);
    }
    if ((0, WEBCRAWLERConfig_1.instanceOfWEBCRAWLERConfig)(value)) {
        return (0, WEBCRAWLERConfig_1.WEBCRAWLERConfigToJSON)(value);
    }
    if ((0, ZOOMADMINConfig_1.instanceOfZOOMADMINConfig)(value)) {
        return (0, ZOOMADMINConfig_1.ZOOMADMINConfigToJSON)(value);
    }
    if ((0, ZOOMConfig_1.instanceOfZOOMConfig)(value)) {
        return (0, ZOOMConfig_1.ZOOMConfigToJSON)(value);
    }
    return {};
}
