// @vectorize-io/vectorize-client/dist/models/StartFileUploadToConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartFileUploadToConnectorRequest = instanceOfStartFileUploadToConnectorRequest;
exports.StartFileUploadToConnectorRequestFromJSON = StartFileUploadToConnectorRequestFromJSON;
exports.StartFileUploadToConnectorRequestFromJSONTyped = StartFileUploadToConnectorRequestFromJSONTyped;
exports.StartFileUploadToConnectorRequestToJSON = StartFileUploadToConnectorRequestToJSON;
exports.StartFileUploadToConnectorRequestToJSONTyped = StartFileUploadToConnectorRequestToJSONTyped;
/**
 * Check if a given object implements the StartFileUploadToConnectorRequest interface.
 */
function instanceOfStartFileUploadToConnectorRequest(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('contentType' in value) || value['contentType'] === undefined)
        return false;
    return true;
}
function StartFileUploadToConnectorRequestFromJSON(json) {
    return StartFileUploadToConnectorRequestFromJSONTyped(json, false);
}
function StartFileUploadToConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'contentType': json['contentType'],
        'metadata': json['metadata'] == null ? undefined : json['metadata'],
    };
}
function StartFileUploadToConnectorRequestToJSON(json) {
    return StartFileUploadToConnectorRequestToJSONTyped(json, false);
}
function StartFileUploadToConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'contentType': value['contentType'],
        'metadata': value['metadata'],
    };
}
