// @vectorize-io/vectorize-client/dist/models/StartExtractionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartExtractionResponse = instanceOfStartExtractionResponse;
exports.StartExtractionResponseFromJSON = StartExtractionResponseFromJSON;
exports.StartExtractionResponseFromJSONTyped = StartExtractionResponseFromJSONTyped;
exports.StartExtractionResponseToJSON = StartExtractionResponseToJSON;
exports.StartExtractionResponseToJSONTyped = StartExtractionResponseToJSONTyped;
/**
 * Check if a given object implements the StartExtractionResponse interface.
 */
function instanceOfStartExtractionResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('extractionId' in value) || value['extractionId'] === undefined)
        return false;
    return true;
}
function StartExtractionResponseFromJSON(json) {
    return StartExtractionResponseFromJSONTyped(json, false);
}
function StartExtractionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'extractionId': json['extractionId'],
    };
}
function StartExtractionResponseToJSON(json) {
    return StartExtractionResponseToJSONTyped(json, false);
}
function StartExtractionResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'extractionId': value['extractionId'],
    };
}
