// @vectorize-io/vectorize-client/dist/models/StartPipelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartPipelineResponse = instanceOfStartPipelineResponse;
exports.StartPipelineResponseFromJSON = StartPipelineResponseFromJSON;
exports.StartPipelineResponseFromJSONTyped = StartPipelineResponseFromJSONTyped;
exports.StartPipelineResponseToJSON = StartPipelineResponseToJSON;
exports.StartPipelineResponseToJSONTyped = StartPipelineResponseToJSONTyped;
/**
 * Check if a given object implements the StartPipelineResponse interface.
 */
function instanceOfStartPipelineResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function StartPipelineResponseFromJSON(json) {
    return StartPipelineResponseFromJSONTyped(json, false);
}
function StartPipelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function StartPipelineResponseToJSON(json) {
    return StartPipelineResponseToJSONTyped(json, false);
}
function StartPipelineResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
