// @vectorize-io/vectorize-client/dist/models/StartFileUploadRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartFileUploadRequest = instanceOfStartFileUploadRequest;
exports.StartFileUploadRequestFromJSON = StartFileUploadRequestFromJSON;
exports.StartFileUploadRequestFromJSONTyped = StartFileUploadRequestFromJSONTyped;
exports.StartFileUploadRequestToJSON = StartFileUploadRequestToJSON;
exports.StartFileUploadRequestToJSONTyped = StartFileUploadRequestToJSONTyped;
/**
 * Check if a given object implements the StartFileUploadRequest interface.
 */
function instanceOfStartFileUploadRequest(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('contentType' in value) || value['contentType'] === undefined)
        return false;
    return true;
}
function StartFileUploadRequestFromJSON(json) {
    return StartFileUploadRequestFromJSONTyped(json, false);
}
function StartFileUploadRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'contentType': json['contentType'],
    };
}
function StartFileUploadRequestToJSON(json) {
    return StartFileUploadRequestToJSONTyped(json, false);
}
function StartFileUploadRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'contentType': value['contentType'],
    };
}
