// @vectorize-io/vectorize-client/dist/models/StartFileUploadToConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartFileUploadToConnectorResponse = instanceOfStartFileUploadToConnectorResponse;
exports.StartFileUploadToConnectorResponseFromJSON = StartFileUploadToConnectorResponseFromJSON;
exports.StartFileUploadToConnectorResponseFromJSONTyped = StartFileUploadToConnectorResponseFromJSONTyped;
exports.StartFileUploadToConnectorResponseToJSON = StartFileUploadToConnectorResponseToJSON;
exports.StartFileUploadToConnectorResponseToJSONTyped = StartFileUploadToConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the StartFileUploadToConnectorResponse interface.
 */
function instanceOfStartFileUploadToConnectorResponse(value) {
    if (!('uploadUrl' in value) || value['uploadUrl'] === undefined)
        return false;
    return true;
}
function StartFileUploadToConnectorResponseFromJSON(json) {
    return StartFileUploadToConnectorResponseFromJSONTyped(json, false);
}
function StartFileUploadToConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'uploadUrl': json['uploadUrl'],
    };
}
function StartFileUploadToConnectorResponseToJSON(json) {
    return StartFileUploadToConnectorResponseToJSONTyped(json, false);
}
function StartFileUploadToConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'uploadUrl': value['uploadUrl'],
    };
}
