// @vectorize-io/vectorize-client/dist/models/StartDeepResearchRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartDeepResearchRequest = instanceOfStartDeepResearchRequest;
exports.StartDeepResearchRequestFromJSON = StartDeepResearchRequestFromJSON;
exports.StartDeepResearchRequestFromJSONTyped = StartDeepResearchRequestFromJSONTyped;
exports.StartDeepResearchRequestToJSON = StartDeepResearchRequestToJSON;
exports.StartDeepResearchRequestToJSONTyped = StartDeepResearchRequestToJSONTyped;
var N8NConfig_1 = require("./N8NConfig");
/**
 * Check if a given object implements the StartDeepResearchRequest interface.
 */
function instanceOfStartDeepResearchRequest(value) {
    if (!('query' in value) || value['query'] === undefined)
        return false;
    return true;
}
function StartDeepResearchRequestFromJSON(json) {
    return StartDeepResearchRequestFromJSONTyped(json, false);
}
function StartDeepResearchRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'query': json['query'],
        'webSearch': json['webSearch'] == null ? undefined : json['webSearch'],
        'schema': json['schema'] == null ? undefined : json['schema'],
        'n8n': json['n8n'] == null ? undefined : (0, N8NConfig_1.N8NConfigFromJSON)(json['n8n']),
    };
}
function StartDeepResearchRequestToJSON(json) {
    return StartDeepResearchRequestToJSONTyped(json, false);
}
function StartDeepResearchRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'query': value['query'],
        'webSearch': value['webSearch'],
        'schema': value['schema'],
        'n8n': (0, N8NConfig_1.N8NConfigToJSON)(value['n8n']),
    };
}
