// @vectorize-io/vectorize-client/dist/models/StartDeepResearchResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartDeepResearchResponse = instanceOfStartDeepResearchResponse;
exports.StartDeepResearchResponseFromJSON = StartDeepResearchResponseFromJSON;
exports.StartDeepResearchResponseFromJSONTyped = StartDeepResearchResponseFromJSONTyped;
exports.StartDeepResearchResponseToJSON = StartDeepResearchResponseToJSON;
exports.StartDeepResearchResponseToJSONTyped = StartDeepResearchResponseToJSONTyped;
/**
 * Check if a given object implements the StartDeepResearchResponse interface.
 */
function instanceOfStartDeepResearchResponse(value) {
    if (!('researchId' in value) || value['researchId'] === undefined)
        return false;
    return true;
}
function StartDeepResearchResponseFromJSON(json) {
    return StartDeepResearchResponseFromJSONTyped(json, false);
}
function StartDeepResearchResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'researchId': json['researchId'],
    };
}
function StartDeepResearchResponseToJSON(json) {
    return StartDeepResearchResponseToJSONTyped(json, false);
}
function StartDeepResearchResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'researchId': value['researchId'],
    };
}
