// @vectorize-io/vectorize-client/dist/models/StartFileUploadResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStartFileUploadResponse = instanceOfStartFileUploadResponse;
exports.StartFileUploadResponseFromJSON = StartFileUploadResponseFromJSON;
exports.StartFileUploadResponseFromJSONTyped = StartFileUploadResponseFromJSONTyped;
exports.StartFileUploadResponseToJSON = StartFileUploadResponseToJSON;
exports.StartFileUploadResponseToJSONTyped = StartFileUploadResponseToJSONTyped;
/**
 * Check if a given object implements the StartFileUploadResponse interface.
 */
function instanceOfStartFileUploadResponse(value) {
    if (!('fileId' in value) || value['fileId'] === undefined)
        return false;
    if (!('uploadUrl' in value) || value['uploadUrl'] === undefined)
        return false;
    return true;
}
function StartFileUploadResponseFromJSON(json) {
    return StartFileUploadResponseFromJSONTyped(json, false);
}
function StartFileUploadResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fileId': json['fileId'],
        'uploadUrl': json['uploadUrl'],
    };
}
function StartFileUploadResponseToJSON(json) {
    return StartFileUploadResponseToJSONTyped(json, false);
}
function StartFileUploadResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'fileId': value['fileId'],
        'uploadUrl': value['uploadUrl'],
    };
}
