// @vectorize-io/vectorize-client/dist/models/StopPipelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStopPipelineResponse = instanceOfStopPipelineResponse;
exports.StopPipelineResponseFromJSON = StopPipelineResponseFromJSON;
exports.StopPipelineResponseFromJSONTyped = StopPipelineResponseFromJSONTyped;
exports.StopPipelineResponseToJSON = StopPipelineResponseToJSON;
exports.StopPipelineResponseToJSONTyped = StopPipelineResponseToJSONTyped;
/**
 * Check if a given object implements the StopPipelineResponse interface.
 */
function instanceOfStopPipelineResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function StopPipelineResponseFromJSON(json) {
    return StopPipelineResponseFromJSONTyped(json, false);
}
function StopPipelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function StopPipelineResponseToJSON(json) {
    return StopPipelineResponseToJSONTyped(json, false);
}
function StopPipelineResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
