// @vectorize-io/vectorize-client/dist/models/Supabase1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSupabase1 = instanceOfSupabase1;
exports.Supabase1FromJSON = Supabase1FromJSON;
exports.Supabase1FromJSONTyped = Supabase1FromJSONTyped;
exports.Supabase1ToJSON = Supabase1ToJSON;
exports.Supabase1ToJSONTyped = Supabase1ToJSONTyped;
var SUPABASEAuthConfig_1 = require("./SUPABASEAuthConfig");
/**
 * Check if a given object implements the Supabase1 interface.
 */
function instanceOfSupabase1(value) {
    return true;
}
function Supabase1FromJSON(json) {
    return Supabase1FromJSONTyped(json, false);
}
function Supabase1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, SUPABASEAuthConfig_1.SUPABASEAuthConfigFromJSON)(json['config']),
    };
}
function Supabase1ToJSON(json) {
    return Supabase1ToJSONTyped(json, false);
}
function Supabase1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, SUPABASEAuthConfig_1.SUPABASEAuthConfigToJSON)(value['config']),
    };
}
