// @vectorize-io/vectorize-client/dist/models/SUPABASEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSUPABASEConfig = instanceOfSUPABASEConfig;
exports.SUPABASEConfigFromJSON = SUPABASEConfigFromJSON;
exports.SUPABASEConfigFromJSONTyped = SUPABASEConfigFromJSONTyped;
exports.SUPABASEConfigToJSON = SUPABASEConfigToJSON;
exports.SUPABASEConfigToJSONTyped = SUPABASEConfigToJSONTyped;
/**
 * Check if a given object implements the SUPABASEConfig interface.
 */
function instanceOfSUPABASEConfig(value) {
    if (!('table' in value) || value['table'] === undefined)
        return false;
    return true;
}
function SUPABASEConfigFromJSON(json) {
    return SUPABASEConfigFromJSONTyped(json, false);
}
function SUPABASEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'table': json['table'],
    };
}
function SUPABASEConfigToJSON(json) {
    return SUPABASEConfigToJSONTyped(json, false);
}
function SUPABASEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'table': value['table'],
    };
}
