// @vectorize-io/vectorize-client/dist/models/Supabase.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupabaseTypeEnum = void 0;
exports.instanceOfSupabase = instanceOfSupabase;
exports.SupabaseFromJSON = SupabaseFromJSON;
exports.SupabaseFromJSONTyped = SupabaseFromJSONTyped;
exports.SupabaseToJSON = SupabaseToJSON;
exports.SupabaseToJSONTyped = SupabaseToJSONTyped;
var SUPABASEAuthConfig_1 = require("./SUPABASEAuthConfig");
/**
 * @export
 */
exports.SupabaseTypeEnum = {
    Supabase: 'SUPABASE'
};
/**
 * Check if a given object implements the Supabase interface.
 */
function instanceOfSupabase(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function SupabaseFromJSON(json) {
    return SupabaseFromJSONTyped(json, false);
}
function SupabaseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, SUPABASEAuthConfig_1.SUPABASEAuthConfigFromJSON)(json['config']),
    };
}
function SupabaseToJSON(json) {
    return SupabaseToJSONTyped(json, false);
}
function SupabaseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, SUPABASEAuthConfig_1.SUPABASEAuthConfigToJSON)(value['config']),
    };
}
