// @vectorize-io/vectorize-client/dist/models/Turbopuffer1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTurbopuffer1 = instanceOfTurbopuffer1;
exports.Turbopuffer1FromJSON = Turbopuffer1FromJSON;
exports.Turbopuffer1FromJSONTyped = Turbopuffer1FromJSONTyped;
exports.Turbopuffer1ToJSON = Turbopuffer1ToJSON;
exports.Turbopuffer1ToJSONTyped = Turbopuffer1ToJSONTyped;
var TURBOPUFFERAuthConfig_1 = require("./TURBOPUFFERAuthConfig");
/**
 * Check if a given object implements the Turbopuffer1 interface.
 */
function instanceOfTurbopuffer1(value) {
    return true;
}
function Turbopuffer1FromJSON(json) {
    return Turbopuffer1FromJSONTyped(json, false);
}
function Turbopuffer1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, TURBOPUFFERAuthConfig_1.TURBOPUFFERAuthConfigFromJSON)(json['config']),
    };
}
function Turbopuffer1ToJSON(json) {
    return Turbopuffer1ToJSONTyped(json, false);
}
function Turbopuffer1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, TURBOPUFFERAuthConfig_1.TURBOPUFFERAuthConfigToJSON)(value['config']),
    };
}
