// @vectorize-io/vectorize-client/dist/models/Turbopuffer.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TurbopufferTypeEnum = void 0;
exports.instanceOfTurbopuffer = instanceOfTurbopuffer;
exports.TurbopufferFromJSON = TurbopufferFromJSON;
exports.TurbopufferFromJSONTyped = TurbopufferFromJSONTyped;
exports.TurbopufferToJSON = TurbopufferToJSON;
exports.TurbopufferToJSONTyped = TurbopufferToJSONTyped;
var TURBOPUFFERAuthConfig_1 = require("./TURBOPUFFERAuthConfig");
/**
 * @export
 */
exports.TurbopufferTypeEnum = {
    Turbopuffer: 'TURBOPUFFER'
};
/**
 * Check if a given object implements the Turbopuffer interface.
 */
function instanceOfTurbopuffer(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function TurbopufferFromJSON(json) {
    return TurbopufferFromJSONTyped(json, false);
}
function TurbopufferFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, TURBOPUFFERAuthConfig_1.TURBOPUFFERAuthConfigFromJSON)(json['config']),
    };
}
function TurbopufferToJSON(json) {
    return TurbopufferToJSONTyped(json, false);
}
function TurbopufferToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, TURBOPUFFERAuthConfig_1.TURBOPUFFERAuthConfigToJSON)(value['config']),
    };
}
