// @vectorize-io/vectorize-client/dist/models/TURBOPUFFERConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTURBOPUFFERConfig = instanceOfTURBOPUFFERConfig;
exports.TURBOPUFFERConfigFromJSON = TURBOPUFFERConfigFromJSON;
exports.TURBOPUFFERConfigFromJSONTyped = TURBOPUFFERConfigFromJSONTyped;
exports.TURBOPUFFERConfigToJSON = TURBOPUFFERConfigToJSON;
exports.TURBOPUFFERConfigToJSONTyped = TURBOPUFFERConfigToJSONTyped;
/**
 * Check if a given object implements the TURBOPUFFERConfig interface.
 */
function instanceOfTURBOPUFFERConfig(value) {
    if (!('namespace' in value) || value['namespace'] === undefined)
        return false;
    return true;
}
function TURBOPUFFERConfigFromJSON(json) {
    return TURBOPUFFERConfigFromJSONTyped(json, false);
}
function TURBOPUFFERConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'namespace': json['namespace'],
    };
}
function TURBOPUFFERConfigToJSON(json) {
    return TURBOPUFFERConfigToJSONTyped(json, false);
}
function TURBOPUFFERConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'namespace': value['namespace'],
    };
}
