// @vectorize-io/vectorize-client/dist/models/TURBOPUFFERAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTURBOPUFFERAuthConfig = instanceOfTURBOPUFFERAuthConfig;
exports.TURBOPUFFERAuthConfigFromJSON = TURBOPUFFERAuthConfigFromJSON;
exports.TURBOPUFFERAuthConfigFromJSONTyped = TURBOPUFFERAuthConfigFromJSONTyped;
exports.TURBOPUFFERAuthConfigToJSON = TURBOPUFFERAuthConfigToJSON;
exports.TURBOPUFFERAuthConfigToJSONTyped = TURBOPUFFERAuthConfigToJSONTyped;
/**
 * Check if a given object implements the TURBOPUFFERAuthConfig interface.
 */
function instanceOfTURBOPUFFERAuthConfig(value) {
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function TURBOPUFFERAuthConfigFromJSON(json) {
    return TURBOPUFFERAuthConfigFromJSONTyped(json, false);
}
function TURBOPUFFERAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'apiKey': json['api-key'],
    };
}
function TURBOPUFFERAuthConfigToJSON(json) {
    return TURBOPUFFERAuthConfigToJSONTyped(json, false);
}
function TURBOPUFFERAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'api-key': value['apiKey'],
    };
}
