// @vectorize-io/vectorize-client/dist/models/UpdateSourceConnectorResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateSourceConnectorResponseData = instanceOfUpdateSourceConnectorResponseData;
exports.UpdateSourceConnectorResponseDataFromJSON = UpdateSourceConnectorResponseDataFromJSON;
exports.UpdateSourceConnectorResponseDataFromJSONTyped = UpdateSourceConnectorResponseDataFromJSONTyped;
exports.UpdateSourceConnectorResponseDataToJSON = UpdateSourceConnectorResponseDataToJSON;
exports.UpdateSourceConnectorResponseDataToJSONTyped = UpdateSourceConnectorResponseDataToJSONTyped;
var SourceConnector_1 = require("./SourceConnector");
/**
 * Check if a given object implements the UpdateSourceConnectorResponseData interface.
 */
function instanceOfUpdateSourceConnectorResponseData(value) {
    if (!('updatedConnector' in value) || value['updatedConnector'] === undefined)
        return false;
    return true;
}
function UpdateSourceConnectorResponseDataFromJSON(json) {
    return UpdateSourceConnectorResponseDataFromJSONTyped(json, false);
}
function UpdateSourceConnectorResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'updatedConnector': (0, SourceConnector_1.SourceConnectorFromJSON)(json['updatedConnector']),
        'pipelineIds': json['pipelineIds'] == null ? undefined : json['pipelineIds'],
    };
}
function UpdateSourceConnectorResponseDataToJSON(json) {
    return UpdateSourceConnectorResponseDataToJSONTyped(json, false);
}
function UpdateSourceConnectorResponseDataToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'updatedConnector': (0, SourceConnector_1.SourceConnectorToJSON)(value['updatedConnector']),
        'pipelineIds': value['pipelineIds'],
    };
}
