// @vectorize-io/vectorize-client/dist/models/UpdateSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateSourceConnectorResponse = instanceOfUpdateSourceConnectorResponse;
exports.UpdateSourceConnectorResponseFromJSON = UpdateSourceConnectorResponseFromJSON;
exports.UpdateSourceConnectorResponseFromJSONTyped = UpdateSourceConnectorResponseFromJSONTyped;
exports.UpdateSourceConnectorResponseToJSON = UpdateSourceConnectorResponseToJSON;
exports.UpdateSourceConnectorResponseToJSONTyped = UpdateSourceConnectorResponseToJSONTyped;
var UpdateSourceConnectorResponseData_1 = require("./UpdateSourceConnectorResponseData");
/**
 * Check if a given object implements the UpdateSourceConnectorResponse interface.
 */
function instanceOfUpdateSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function UpdateSourceConnectorResponseFromJSON(json) {
    return UpdateSourceConnectorResponseFromJSONTyped(json, false);
}
function UpdateSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (0, UpdateSourceConnectorResponseData_1.UpdateSourceConnectorResponseDataFromJSON)(json['data']),
    };
}
function UpdateSourceConnectorResponseToJSON(json) {
    return UpdateSourceConnectorResponseToJSONTyped(json, false);
}
function UpdateSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (0, UpdateSourceConnectorResponseData_1.UpdateSourceConnectorResponseDataToJSON)(value['data']),
    };
}
