// @vectorize-io/vectorize-client/dist/models/UpdateDestinationConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDestinationConnectorRequestFromJSON = UpdateDestinationConnectorRequestFromJSON;
exports.UpdateDestinationConnectorRequestFromJSONTyped = UpdateDestinationConnectorRequestFromJSONTyped;
exports.UpdateDestinationConnectorRequestToJSON = UpdateDestinationConnectorRequestToJSON;
exports.UpdateDestinationConnectorRequestToJSONTyped = UpdateDestinationConnectorRequestToJSONTyped;
var Azureaisearch1_1 = require("./Azureaisearch1");
var Capella1_1 = require("./Capella1");
var Datastax1_1 = require("./Datastax1");
var Elastic1_1 = require("./Elastic1");
var Milvus1_1 = require("./Milvus1");
var Pinecone1_1 = require("./Pinecone1");
var Postgresql1_1 = require("./Postgresql1");
var Qdrant1_1 = require("./Qdrant1");
var Singlestore1_1 = require("./Singlestore1");
var Supabase1_1 = require("./Supabase1");
var Turbopuffer1_1 = require("./Turbopuffer1");
var Weaviate1_1 = require("./Weaviate1");
function UpdateDestinationConnectorRequestFromJSON(json) {
    return UpdateDestinationConnectorRequestFromJSONTyped(json, false);
}
function UpdateDestinationConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    if ((0, Azureaisearch1_1.instanceOfAzureaisearch1)(json)) {
        return (0, Azureaisearch1_1.Azureaisearch1FromJSONTyped)(json, true);
    }
    if ((0, Capella1_1.instanceOfCapella1)(json)) {
        return (0, Capella1_1.Capella1FromJSONTyped)(json, true);
    }
    if ((0, Datastax1_1.instanceOfDatastax1)(json)) {
        return (0, Datastax1_1.Datastax1FromJSONTyped)(json, true);
    }
    if ((0, Elastic1_1.instanceOfElastic1)(json)) {
        return (0, Elastic1_1.Elastic1FromJSONTyped)(json, true);
    }
    if ((0, Milvus1_1.instanceOfMilvus1)(json)) {
        return (0, Milvus1_1.Milvus1FromJSONTyped)(json, true);
    }
    if ((0, Pinecone1_1.instanceOfPinecone1)(json)) {
        return (0, Pinecone1_1.Pinecone1FromJSONTyped)(json, true);
    }
    if ((0, Postgresql1_1.instanceOfPostgresql1)(json)) {
        return (0, Postgresql1_1.Postgresql1FromJSONTyped)(json, true);
    }
    if ((0, Qdrant1_1.instanceOfQdrant1)(json)) {
        return (0, Qdrant1_1.Qdrant1FromJSONTyped)(json, true);
    }
    if ((0, Singlestore1_1.instanceOfSinglestore1)(json)) {
        return (0, Singlestore1_1.Singlestore1FromJSONTyped)(json, true);
    }
    if ((0, Supabase1_1.instanceOfSupabase1)(json)) {
        return (0, Supabase1_1.Supabase1FromJSONTyped)(json, true);
    }
    if ((0, Turbopuffer1_1.instanceOfTurbopuffer1)(json)) {
        return (0, Turbopuffer1_1.Turbopuffer1FromJSONTyped)(json, true);
    }
    if ((0, Weaviate1_1.instanceOfWeaviate1)(json)) {
        return (0, Weaviate1_1.Weaviate1FromJSONTyped)(json, true);
    }
    return {};
}
function UpdateDestinationConnectorRequestToJSON(json) {
    return UpdateDestinationConnectorRequestToJSONTyped(json, false);
}
function UpdateDestinationConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    if ((0, Azureaisearch1_1.instanceOfAzureaisearch1)(value)) {
        return (0, Azureaisearch1_1.Azureaisearch1ToJSON)(value);
    }
    if ((0, Capella1_1.instanceOfCapella1)(value)) {
        return (0, Capella1_1.Capella1ToJSON)(value);
    }
    if ((0, Datastax1_1.instanceOfDatastax1)(value)) {
        return (0, Datastax1_1.Datastax1ToJSON)(value);
    }
    if ((0, Elastic1_1.instanceOfElastic1)(value)) {
        return (0, Elastic1_1.Elastic1ToJSON)(value);
    }
    if ((0, Milvus1_1.instanceOfMilvus1)(value)) {
        return (0, Milvus1_1.Milvus1ToJSON)(value);
    }
    if ((0, Pinecone1_1.instanceOfPinecone1)(value)) {
        return (0, Pinecone1_1.Pinecone1ToJSON)(value);
    }
    if ((0, Postgresql1_1.instanceOfPostgresql1)(value)) {
        return (0, Postgresql1_1.Postgresql1ToJSON)(value);
    }
    if ((0, Qdrant1_1.instanceOfQdrant1)(value)) {
        return (0, Qdrant1_1.Qdrant1ToJSON)(value);
    }
    if ((0, Singlestore1_1.instanceOfSinglestore1)(value)) {
        return (0, Singlestore1_1.Singlestore1ToJSON)(value);
    }
    if ((0, Supabase1_1.instanceOfSupabase1)(value)) {
        return (0, Supabase1_1.Supabase1ToJSON)(value);
    }
    if ((0, Turbopuffer1_1.instanceOfTurbopuffer1)(value)) {
        return (0, Turbopuffer1_1.Turbopuffer1ToJSON)(value);
    }
    if ((0, Weaviate1_1.instanceOfWeaviate1)(value)) {
        return (0, Weaviate1_1.Weaviate1ToJSON)(value);
    }
    return {};
}
