// @vectorize-io/vectorize-client/dist/models/UpdatedAIPlatformConnectorData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdatedAIPlatformConnectorData = instanceOfUpdatedAIPlatformConnectorData;
exports.UpdatedAIPlatformConnectorDataFromJSON = UpdatedAIPlatformConnectorDataFromJSON;
exports.UpdatedAIPlatformConnectorDataFromJSONTyped = UpdatedAIPlatformConnectorDataFromJSONTyped;
exports.UpdatedAIPlatformConnectorDataToJSON = UpdatedAIPlatformConnectorDataToJSON;
exports.UpdatedAIPlatformConnectorDataToJSONTyped = UpdatedAIPlatformConnectorDataToJSONTyped;
var AIPlatformConnector_1 = require("./AIPlatformConnector");
/**
 * Check if a given object implements the UpdatedAIPlatformConnectorData interface.
 */
function instanceOfUpdatedAIPlatformConnectorData(value) {
    if (!('updatedConnector' in value) || value['updatedConnector'] === undefined)
        return false;
    return true;
}
function UpdatedAIPlatformConnectorDataFromJSON(json) {
    return UpdatedAIPlatformConnectorDataFromJSONTyped(json, false);
}
function UpdatedAIPlatformConnectorDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'updatedConnector': (0, AIPlatformConnector_1.AIPlatformConnectorFromJSON)(json['updatedConnector']),
        'pipelineIds': json['pipelineIds'] == null ? undefined : json['pipelineIds'],
    };
}
function UpdatedAIPlatformConnectorDataToJSON(json) {
    return UpdatedAIPlatformConnectorDataToJSONTyped(json, false);
}
function UpdatedAIPlatformConnectorDataToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'updatedConnector': (0, AIPlatformConnector_1.AIPlatformConnectorToJSON)(value['updatedConnector']),
        'pipelineIds': value['pipelineIds'],
    };
}
