// @vectorize-io/vectorize-client/dist/models/UpdateUserInSourceConnectorRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateUserInSourceConnectorRequest = instanceOfUpdateUserInSourceConnectorRequest;
exports.UpdateUserInSourceConnectorRequestFromJSON = UpdateUserInSourceConnectorRequestFromJSON;
exports.UpdateUserInSourceConnectorRequestFromJSONTyped = UpdateUserInSourceConnectorRequestFromJSONTyped;
exports.UpdateUserInSourceConnectorRequestToJSON = UpdateUserInSourceConnectorRequestToJSON;
exports.UpdateUserInSourceConnectorRequestToJSONTyped = UpdateUserInSourceConnectorRequestToJSONTyped;
var AddUserToSourceConnectorRequestSelectedFiles_1 = require("./AddUserToSourceConnectorRequestSelectedFiles");
/**
 * Check if a given object implements the UpdateUserInSourceConnectorRequest interface.
 */
function instanceOfUpdateUserInSourceConnectorRequest(value) {
    if (!('userId' in value) || value['userId'] === undefined)
        return false;
    return true;
}
function UpdateUserInSourceConnectorRequestFromJSON(json) {
    return UpdateUserInSourceConnectorRequestFromJSONTyped(json, false);
}
function UpdateUserInSourceConnectorRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'userId': json['userId'],
        'selectedFiles': json['selectedFiles'] == null ? undefined : (0, AddUserToSourceConnectorRequestSelectedFiles_1.AddUserToSourceConnectorRequestSelectedFilesFromJSON)(json['selectedFiles']),
        'refreshToken': json['refreshToken'] == null ? undefined : json['refreshToken'],
        'accessToken': json['accessToken'] == null ? undefined : json['accessToken'],
    };
}
function UpdateUserInSourceConnectorRequestToJSON(json) {
    return UpdateUserInSourceConnectorRequestToJSONTyped(json, false);
}
function UpdateUserInSourceConnectorRequestToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'userId': value['userId'],
        'selectedFiles': (0, AddUserToSourceConnectorRequestSelectedFiles_1.AddUserToSourceConnectorRequestSelectedFilesToJSON)(value['selectedFiles']),
        'refreshToken': value['refreshToken'],
        'accessToken': value['accessToken'],
    };
}
