// @vectorize-io/vectorize-client/dist/models/UpdatedDestinationConnectorData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdatedDestinationConnectorData = instanceOfUpdatedDestinationConnectorData;
exports.UpdatedDestinationConnectorDataFromJSON = UpdatedDestinationConnectorDataFromJSON;
exports.UpdatedDestinationConnectorDataFromJSONTyped = UpdatedDestinationConnectorDataFromJSONTyped;
exports.UpdatedDestinationConnectorDataToJSON = UpdatedDestinationConnectorDataToJSON;
exports.UpdatedDestinationConnectorDataToJSONTyped = UpdatedDestinationConnectorDataToJSONTyped;
var DestinationConnector_1 = require("./DestinationConnector");
/**
 * Check if a given object implements the UpdatedDestinationConnectorData interface.
 */
function instanceOfUpdatedDestinationConnectorData(value) {
    if (!('updatedConnector' in value) || value['updatedConnector'] === undefined)
        return false;
    return true;
}
function UpdatedDestinationConnectorDataFromJSON(json) {
    return UpdatedDestinationConnectorDataFromJSONTyped(json, false);
}
function UpdatedDestinationConnectorDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'updatedConnector': (0, DestinationConnector_1.DestinationConnectorFromJSON)(json['updatedConnector']),
        'pipelineIds': json['pipelineIds'] == null ? undefined : json['pipelineIds'],
    };
}
function UpdatedDestinationConnectorDataToJSON(json) {
    return UpdatedDestinationConnectorDataToJSONTyped(json, false);
}
function UpdatedDestinationConnectorDataToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'updatedConnector': (0, DestinationConnector_1.DestinationConnectorToJSON)(value['updatedConnector']),
        'pipelineIds': value['pipelineIds'],
    };
}
