// @vectorize-io/vectorize-client/dist/models/UpdateAIPlatformConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateAIPlatformConnectorResponse = instanceOfUpdateAIPlatformConnectorResponse;
exports.UpdateAIPlatformConnectorResponseFromJSON = UpdateAIPlatformConnectorResponseFromJSON;
exports.UpdateAIPlatformConnectorResponseFromJSONTyped = UpdateAIPlatformConnectorResponseFromJSONTyped;
exports.UpdateAIPlatformConnectorResponseToJSON = UpdateAIPlatformConnectorResponseToJSON;
exports.UpdateAIPlatformConnectorResponseToJSONTyped = UpdateAIPlatformConnectorResponseToJSONTyped;
var UpdatedAIPlatformConnectorData_1 = require("./UpdatedAIPlatformConnectorData");
/**
 * Check if a given object implements the UpdateAIPlatformConnectorResponse interface.
 */
function instanceOfUpdateAIPlatformConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function UpdateAIPlatformConnectorResponseFromJSON(json) {
    return UpdateAIPlatformConnectorResponseFromJSONTyped(json, false);
}
function UpdateAIPlatformConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (0, UpdatedAIPlatformConnectorData_1.UpdatedAIPlatformConnectorDataFromJSON)(json['data']),
    };
}
function UpdateAIPlatformConnectorResponseToJSON(json) {
    return UpdateAIPlatformConnectorResponseToJSONTyped(json, false);
}
function UpdateAIPlatformConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (0, UpdatedAIPlatformConnectorData_1.UpdatedAIPlatformConnectorDataToJSON)(value['data']),
    };
}
