// @vectorize-io/vectorize-client/dist/models/UpdateDestinationConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateDestinationConnectorResponse = instanceOfUpdateDestinationConnectorResponse;
exports.UpdateDestinationConnectorResponseFromJSON = UpdateDestinationConnectorResponseFromJSON;
exports.UpdateDestinationConnectorResponseFromJSONTyped = UpdateDestinationConnectorResponseFromJSONTyped;
exports.UpdateDestinationConnectorResponseToJSON = UpdateDestinationConnectorResponseToJSON;
exports.UpdateDestinationConnectorResponseToJSONTyped = UpdateDestinationConnectorResponseToJSONTyped;
var UpdatedDestinationConnectorData_1 = require("./UpdatedDestinationConnectorData");
/**
 * Check if a given object implements the UpdateDestinationConnectorResponse interface.
 */
function instanceOfUpdateDestinationConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    if (!('data' in value) || value['data'] === undefined)
        return false;
    return true;
}
function UpdateDestinationConnectorResponseFromJSON(json) {
    return UpdateDestinationConnectorResponseFromJSONTyped(json, false);
}
function UpdateDestinationConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
        'data': (0, UpdatedDestinationConnectorData_1.UpdatedDestinationConnectorDataFromJSON)(json['data']),
    };
}
function UpdateDestinationConnectorResponseToJSON(json) {
    return UpdateDestinationConnectorResponseToJSONTyped(json, false);
}
function UpdateDestinationConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
        'data': (0, UpdatedDestinationConnectorData_1.UpdatedDestinationConnectorDataToJSON)(value['data']),
    };
}
