// @vectorize-io/vectorize-client/dist/models/UpdateUserInSourceConnectorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateUserInSourceConnectorResponse = instanceOfUpdateUserInSourceConnectorResponse;
exports.UpdateUserInSourceConnectorResponseFromJSON = UpdateUserInSourceConnectorResponseFromJSON;
exports.UpdateUserInSourceConnectorResponseFromJSONTyped = UpdateUserInSourceConnectorResponseFromJSONTyped;
exports.UpdateUserInSourceConnectorResponseToJSON = UpdateUserInSourceConnectorResponseToJSON;
exports.UpdateUserInSourceConnectorResponseToJSONTyped = UpdateUserInSourceConnectorResponseToJSONTyped;
/**
 * Check if a given object implements the UpdateUserInSourceConnectorResponse interface.
 */
function instanceOfUpdateUserInSourceConnectorResponse(value) {
    if (!('message' in value) || value['message'] === undefined)
        return false;
    return true;
}
function UpdateUserInSourceConnectorResponseFromJSON(json) {
    return UpdateUserInSourceConnectorResponseFromJSONTyped(json, false);
}
function UpdateUserInSourceConnectorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
function UpdateUserInSourceConnectorResponseToJSON(json) {
    return UpdateUserInSourceConnectorResponseToJSONTyped(json, false);
}
function UpdateUserInSourceConnectorResponseToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'message': value['message'],
    };
}
