// @vectorize-io/vectorize-client/dist/models/UploadFile.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUploadFile = instanceOfUploadFile;
exports.UploadFileFromJSON = UploadFileFromJSON;
exports.UploadFileFromJSONTyped = UploadFileFromJSONTyped;
exports.UploadFileToJSON = UploadFileToJSON;
exports.UploadFileToJSONTyped = UploadFileToJSONTyped;
/**
 * Check if a given object implements the UploadFile interface.
 */
function instanceOfUploadFile(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('size' in value) || value['size'] === undefined)
        return false;
    if (!('lastModified' in value) || value['lastModified'] === undefined)
        return false;
    if (!('metadata' in value) || value['metadata'] === undefined)
        return false;
    return true;
}
function UploadFileFromJSON(json) {
    return UploadFileFromJSONTyped(json, false);
}
function UploadFileFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
        'name': json['name'],
        'size': json['size'],
        'extension': json['extension'] == null ? undefined : json['extension'],
        'lastModified': json['lastModified'],
        'metadata': json['metadata'],
    };
}
function UploadFileToJSON(json) {
    return UploadFileToJSONTyped(json, false);
}
function UploadFileToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
        'name': value['name'],
        'size': value['size'],
        'extension': value['extension'],
        'lastModified': value['lastModified'],
        'metadata': value['metadata'],
    };
}
