// @vectorize-io/vectorize-client/dist/models/VERTEXAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfVERTEXAuthConfig = instanceOfVERTEXAuthConfig;
exports.VERTEXAuthConfigFromJSON = VERTEXAuthConfigFromJSON;
exports.VERTEXAuthConfigFromJSONTyped = VERTEXAuthConfigFromJSONTyped;
exports.VERTEXAuthConfigToJSON = VERTEXAuthConfigToJSON;
exports.VERTEXAuthConfigToJSONTyped = VERTEXAuthConfigToJSONTyped;
/**
 * Check if a given object implements the VERTEXAuthConfig interface.
 */
function instanceOfVERTEXAuthConfig(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    if (!('region' in value) || value['region'] === undefined)
        return false;
    return true;
}
function VERTEXAuthConfigFromJSON(json) {
    return VERTEXAuthConfigFromJSONTyped(json, false);
}
function VERTEXAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
        'region': json['region'],
    };
}
function VERTEXAuthConfigToJSON(json) {
    return VERTEXAuthConfigToJSONTyped(json, false);
}
function VERTEXAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
        'region': value['region'],
    };
}
