// @vectorize-io/vectorize-client/dist/models/Vertex.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VertexTypeEnum = void 0;
exports.instanceOfVertex = instanceOfVertex;
exports.VertexFromJSON = VertexFromJSON;
exports.VertexFromJSONTyped = VertexFromJSONTyped;
exports.VertexToJSON = VertexToJSON;
exports.VertexToJSONTyped = VertexToJSONTyped;
var VERTEXAuthConfig_1 = require("./VERTEXAuthConfig");
/**
 * @export
 */
exports.VertexTypeEnum = {
    Vertex: 'VERTEX'
};
/**
 * Check if a given object implements the Vertex interface.
 */
function instanceOfVertex(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function VertexFromJSON(json) {
    return VertexFromJSONTyped(json, false);
}
function VertexFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, VERTEXAuthConfig_1.VERTEXAuthConfigFromJSON)(json['config']),
    };
}
function VertexToJSON(json) {
    return VertexToJSONTyped(json, false);
}
function VertexToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, VERTEXAuthConfig_1.VERTEXAuthConfigToJSON)(value['config']),
    };
}
