// @vectorize-io/vectorize-client/dist/models/Vertex1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfVertex1 = instanceOfVertex1;
exports.Vertex1FromJSON = Vertex1FromJSON;
exports.Vertex1FromJSONTyped = Vertex1FromJSONTyped;
exports.Vertex1ToJSON = Vertex1ToJSON;
exports.Vertex1ToJSONTyped = Vertex1ToJSONTyped;
/**
 * Check if a given object implements the Vertex1 interface.
 */
function instanceOfVertex1(value) {
    return true;
}
function Vertex1FromJSON(json) {
    return Vertex1FromJSONTyped(json, false);
}
function Vertex1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Vertex1ToJSON(json) {
    return Vertex1ToJSONTyped(json, false);
}
function Vertex1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
