// @vectorize-io/vectorize-client/dist/models/VOYAGEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfVOYAGEAuthConfig = instanceOfVOYAGEAuthConfig;
exports.VOYAGEAuthConfigFromJSON = VOYAGEAuthConfigFromJSON;
exports.VOYAGEAuthConfigFromJSONTyped = VOYAGEAuthConfigFromJSONTyped;
exports.VOYAGEAuthConfigToJSON = VOYAGEAuthConfigToJSON;
exports.VOYAGEAuthConfigToJSONTyped = VOYAGEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the VOYAGEAuthConfig interface.
 */
function instanceOfVOYAGEAuthConfig(value) {
    if (!('key' in value) || value['key'] === undefined)
        return false;
    return true;
}
function VOYAGEAuthConfigFromJSON(json) {
    return VOYAGEAuthConfigFromJSONTyped(json, false);
}
function VOYAGEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
function VOYAGEAuthConfigToJSON(json) {
    return VOYAGEAuthConfigToJSONTyped(json, false);
}
function VOYAGEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
    };
}
