// @vectorize-io/vectorize-client/dist/models/Voyage1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfVoyage1 = instanceOfVoyage1;
exports.Voyage1FromJSON = Voyage1FromJSON;
exports.Voyage1FromJSONTyped = Voyage1FromJSONTyped;
exports.Voyage1ToJSON = Voyage1ToJSON;
exports.Voyage1ToJSONTyped = Voyage1ToJSONTyped;
/**
 * Check if a given object implements the Voyage1 interface.
 */
function instanceOfVoyage1(value) {
    return true;
}
function Voyage1FromJSON(json) {
    return Voyage1FromJSONTyped(json, false);
}
function Voyage1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : json['config'],
    };
}
function Voyage1ToJSON(json) {
    return Voyage1ToJSONTyped(json, false);
}
function Voyage1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': value['config'],
    };
}
