// @vectorize-io/vectorize-client/dist/models/Voyage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoyageTypeEnum = void 0;
exports.instanceOfVoyage = instanceOfVoyage;
exports.VoyageFromJSON = VoyageFromJSON;
exports.VoyageFromJSONTyped = VoyageFromJSONTyped;
exports.VoyageToJSON = VoyageToJSON;
exports.VoyageToJSONTyped = VoyageToJSONTyped;
var VOYAGEAuthConfig_1 = require("./VOYAGEAuthConfig");
/**
 * @export
 */
exports.VoyageTypeEnum = {
    Voyage: 'VOYAGE'
};
/**
 * Check if a given object implements the Voyage interface.
 */
function instanceOfVoyage(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function VoyageFromJSON(json) {
    return VoyageFromJSONTyped(json, false);
}
function VoyageFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, VOYAGEAuthConfig_1.VOYAGEAuthConfigFromJSON)(json['config']),
    };
}
function VoyageToJSON(json) {
    return VoyageToJSONTyped(json, false);
}
function VoyageToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, VOYAGEAuthConfig_1.VOYAGEAuthConfigToJSON)(value['config']),
    };
}
