// @vectorize-io/vectorize-client/dist/models/Weaviate1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWeaviate1 = instanceOfWeaviate1;
exports.Weaviate1FromJSON = Weaviate1FromJSON;
exports.Weaviate1FromJSONTyped = Weaviate1FromJSONTyped;
exports.Weaviate1ToJSON = Weaviate1ToJSON;
exports.Weaviate1ToJSONTyped = Weaviate1ToJSONTyped;
var WEAVIATEAuthConfig_1 = require("./WEAVIATEAuthConfig");
/**
 * Check if a given object implements the Weaviate1 interface.
 */
function instanceOfWeaviate1(value) {
    return true;
}
function Weaviate1FromJSON(json) {
    return Weaviate1FromJSONTyped(json, false);
}
function Weaviate1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, WEAVIATEAuthConfig_1.WEAVIATEAuthConfigFromJSON)(json['config']),
    };
}
function Weaviate1ToJSON(json) {
    return Weaviate1ToJSONTyped(json, false);
}
function Weaviate1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, WEAVIATEAuthConfig_1.WEAVIATEAuthConfigToJSON)(value['config']),
    };
}
