// @vectorize-io/vectorize-client/dist/models/WEAVIATEConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWEAVIATEConfig = instanceOfWEAVIATEConfig;
exports.WEAVIATEConfigFromJSON = WEAVIATEConfigFromJSON;
exports.WEAVIATEConfigFromJSONTyped = WEAVIATEConfigFromJSONTyped;
exports.WEAVIATEConfigToJSON = WEAVIATEConfigToJSON;
exports.WEAVIATEConfigToJSONTyped = WEAVIATEConfigToJSONTyped;
/**
 * Check if a given object implements the WEAVIATEConfig interface.
 */
function instanceOfWEAVIATEConfig(value) {
    if (!('collection' in value) || value['collection'] === undefined)
        return false;
    return true;
}
function WEAVIATEConfigFromJSON(json) {
    return WEAVIATEConfigFromJSONTyped(json, false);
}
function WEAVIATEConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'collection': json['collection'],
    };
}
function WEAVIATEConfigToJSON(json) {
    return WEAVIATEConfigToJSONTyped(json, false);
}
function WEAVIATEConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'collection': value['collection'],
    };
}
