// @vectorize-io/vectorize-client/dist/models/Weaviate.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeaviateTypeEnum = void 0;
exports.instanceOfWeaviate = instanceOfWeaviate;
exports.WeaviateFromJSON = WeaviateFromJSON;
exports.WeaviateFromJSONTyped = WeaviateFromJSONTyped;
exports.WeaviateToJSON = WeaviateToJSON;
exports.WeaviateToJSONTyped = WeaviateToJSONTyped;
var WEAVIATEAuthConfig_1 = require("./WEAVIATEAuthConfig");
/**
 * @export
 */
exports.WeaviateTypeEnum = {
    Weaviate: 'WEAVIATE'
};
/**
 * Check if a given object implements the Weaviate interface.
 */
function instanceOfWeaviate(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function WeaviateFromJSON(json) {
    return WeaviateFromJSONTyped(json, false);
}
function WeaviateFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, WEAVIATEAuthConfig_1.WEAVIATEAuthConfigFromJSON)(json['config']),
    };
}
function WeaviateToJSON(json) {
    return WeaviateToJSONTyped(json, false);
}
function WeaviateToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, WEAVIATEAuthConfig_1.WEAVIATEAuthConfigToJSON)(value['config']),
    };
}
