// @vectorize-io/vectorize-client/dist/models/WEAVIATEAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWEAVIATEAuthConfig = instanceOfWEAVIATEAuthConfig;
exports.WEAVIATEAuthConfigFromJSON = WEAVIATEAuthConfigFromJSON;
exports.WEAVIATEAuthConfigFromJSONTyped = WEAVIATEAuthConfigFromJSONTyped;
exports.WEAVIATEAuthConfigToJSON = WEAVIATEAuthConfigToJSON;
exports.WEAVIATEAuthConfigToJSONTyped = WEAVIATEAuthConfigToJSONTyped;
/**
 * Check if a given object implements the WEAVIATEAuthConfig interface.
 */
function instanceOfWEAVIATEAuthConfig(value) {
    if (!('host' in value) || value['host'] === undefined)
        return false;
    if (!('apiKey' in value) || value['apiKey'] === undefined)
        return false;
    return true;
}
function WEAVIATEAuthConfigFromJSON(json) {
    return WEAVIATEAuthConfigFromJSONTyped(json, false);
}
function WEAVIATEAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'host': json['host'],
        'apiKey': json['api-key'],
    };
}
function WEAVIATEAuthConfigToJSON(json) {
    return WEAVIATEAuthConfigToJSONTyped(json, false);
}
function WEAVIATEAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'host': value['host'],
        'api-key': value['apiKey'],
    };
}
