// @vectorize-io/vectorize-client/dist/models/WEBCRAWLERConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWEBCRAWLERConfig = instanceOfWEBCRAWLERConfig;
exports.WEBCRAWLERConfigFromJSON = WEBCRAWLERConfigFromJSON;
exports.WEBCRAWLERConfigFromJSONTyped = WEBCRAWLERConfigFromJSONTyped;
exports.WEBCRAWLERConfigToJSON = WEBCRAWLERConfigToJSON;
exports.WEBCRAWLERConfigToJSONTyped = WEBCRAWLERConfigToJSONTyped;
/**
 * Check if a given object implements the WEBCRAWLERConfig interface.
 */
function instanceOfWEBCRAWLERConfig(value) {
    return true;
}
function WEBCRAWLERConfigFromJSON(json) {
    return WEBCRAWLERConfigFromJSONTyped(json, false);
}
function WEBCRAWLERConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'allowedDomainsOpt': json['allowed-domains-opt'] == null ? undefined : json['allowed-domains-opt'],
        'forbiddenPaths': json['forbidden-paths'] == null ? undefined : json['forbidden-paths'],
        'minTimeBetweenRequests': json['min-time-between-requests'] == null ? undefined : json['min-time-between-requests'],
        'maxErrorCount': json['max-error-count'] == null ? undefined : json['max-error-count'],
        'maxUrls': json['max-urls'] == null ? undefined : json['max-urls'],
        'maxDepth': json['max-depth'] == null ? undefined : json['max-depth'],
        'reindexIntervalSeconds': json['reindex-interval-seconds'] == null ? undefined : json['reindex-interval-seconds'],
    };
}
function WEBCRAWLERConfigToJSON(json) {
    return WEBCRAWLERConfigToJSONTyped(json, false);
}
function WEBCRAWLERConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'allowed-domains-opt': value['allowedDomainsOpt'],
        'forbidden-paths': value['forbiddenPaths'],
        'min-time-between-requests': value['minTimeBetweenRequests'],
        'max-error-count': value['maxErrorCount'],
        'max-urls': value['maxUrls'],
        'max-depth': value['maxDepth'],
        'reindex-interval-seconds': value['reindexIntervalSeconds'],
    };
}
