// @vectorize-io/vectorize-client/dist/models/WEBCRAWLERAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWEBCRAWLERAuthConfig = instanceOfWEBCRAWLERAuthConfig;
exports.WEBCRAWLERAuthConfigFromJSON = WEBCRAWLERAuthConfigFromJSON;
exports.WEBCRAWLERAuthConfigFromJSONTyped = WEBCRAWLERAuthConfigFromJSONTyped;
exports.WEBCRAWLERAuthConfigToJSON = WEBCRAWLERAuthConfigToJSON;
exports.WEBCRAWLERAuthConfigToJSONTyped = WEBCRAWLERAuthConfigToJSONTyped;
/**
 * Check if a given object implements the WEBCRAWLERAuthConfig interface.
 */
function instanceOfWEBCRAWLERAuthConfig(value) {
    if (!('seedUrls' in value) || value['seedUrls'] === undefined)
        return false;
    return true;
}
function WEBCRAWLERAuthConfigFromJSON(json) {
    return WEBCRAWLERAuthConfigFromJSONTyped(json, false);
}
function WEBCRAWLERAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'seedUrls': json['seed-urls'],
    };
}
function WEBCRAWLERAuthConfigToJSON(json) {
    return WEBCRAWLERAuthConfigToJSONTyped(json, false);
}
function WEBCRAWLERAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'seed-urls': value['seedUrls'],
    };
}
