// @vectorize-io/vectorize-client/dist/models/WebCrawler.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebCrawlerTypeEnum = void 0;
exports.instanceOfWebCrawler = instanceOfWebCrawler;
exports.WebCrawlerFromJSON = WebCrawlerFromJSON;
exports.WebCrawlerFromJSONTyped = WebCrawlerFromJSONTyped;
exports.WebCrawlerToJSON = WebCrawlerToJSON;
exports.WebCrawlerToJSONTyped = WebCrawlerToJSONTyped;
var WEBCRAWLERAuthConfig_1 = require("./WEBCRAWLERAuthConfig");
/**
 * @export
 */
exports.WebCrawlerTypeEnum = {
    WebCrawler: 'WEB_CRAWLER'
};
/**
 * Check if a given object implements the WebCrawler interface.
 */
function instanceOfWebCrawler(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function WebCrawlerFromJSON(json) {
    return WebCrawlerFromJSONTyped(json, false);
}
function WebCrawlerFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, WEBCRAWLERAuthConfig_1.WEBCRAWLERAuthConfigFromJSON)(json['config']),
    };
}
function WebCrawlerToJSON(json) {
    return WebCrawlerToJSONTyped(json, false);
}
function WebCrawlerToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, WEBCRAWLERAuthConfig_1.WEBCRAWLERAuthConfigToJSON)(value['config']),
    };
}
