// @vectorize-io/vectorize-client/dist/models/WebCrawler1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWebCrawler1 = instanceOfWebCrawler1;
exports.WebCrawler1FromJSON = WebCrawler1FromJSON;
exports.WebCrawler1FromJSONTyped = WebCrawler1FromJSONTyped;
exports.WebCrawler1ToJSON = WebCrawler1ToJSON;
exports.WebCrawler1ToJSONTyped = WebCrawler1ToJSONTyped;
var WEBCRAWLERAuthConfig_1 = require("./WEBCRAWLERAuthConfig");
/**
 * Check if a given object implements the WebCrawler1 interface.
 */
function instanceOfWebCrawler1(value) {
    return true;
}
function WebCrawler1FromJSON(json) {
    return WebCrawler1FromJSONTyped(json, false);
}
function WebCrawler1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, WEBCRAWLERAuthConfig_1.WEBCRAWLERAuthConfigFromJSON)(json['config']),
    };
}
function WebCrawler1ToJSON(json) {
    return WebCrawler1ToJSONTyped(json, false);
}
function WebCrawler1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, WEBCRAWLERAuthConfig_1.WEBCRAWLERAuthConfigToJSON)(value['config']),
    };
}
