// @vectorize-io/vectorize-client/dist/models/Zoom1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZoom1 = instanceOfZoom1;
exports.Zoom1FromJSON = Zoom1FromJSON;
exports.Zoom1FromJSONTyped = Zoom1FromJSONTyped;
exports.Zoom1ToJSON = Zoom1ToJSON;
exports.Zoom1ToJSONTyped = Zoom1ToJSONTyped;
var ZOOMAuthConfig_1 = require("./ZOOMAuthConfig");
/**
 * Check if a given object implements the Zoom1 interface.
 */
function instanceOfZoom1(value) {
    return true;
}
function Zoom1FromJSON(json) {
    return Zoom1FromJSONTyped(json, false);
}
function Zoom1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, ZOOMAuthConfig_1.ZOOMAuthConfigFromJSON)(json['config']),
    };
}
function Zoom1ToJSON(json) {
    return Zoom1ToJSONTyped(json, false);
}
function Zoom1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, ZOOMAuthConfig_1.ZOOMAuthConfigToJSON)(value['config']),
    };
}
