// @vectorize-io/vectorize-client/dist/models/ZOOMAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZOOMAuthConfig = instanceOfZOOMAuthConfig;
exports.ZOOMAuthConfigFromJSON = ZOOMAuthConfigFromJSON;
exports.ZOOMAuthConfigFromJSONTyped = ZOOMAuthConfigFromJSONTyped;
exports.ZOOMAuthConfigToJSON = ZOOMAuthConfigToJSON;
exports.ZOOMAuthConfigToJSONTyped = ZOOMAuthConfigToJSONTyped;
/**
 * Check if a given object implements the ZOOMAuthConfig interface.
 */
function instanceOfZOOMAuthConfig(value) {
    if (!('refreshToken' in value) || value['refreshToken'] === undefined)
        return false;
    return true;
}
function ZOOMAuthConfigFromJSON(json) {
    return ZOOMAuthConfigFromJSONTyped(json, false);
}
function ZOOMAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'refreshToken': json['refresh-token'],
    };
}
function ZOOMAuthConfigToJSON(json) {
    return ZOOMAuthConfigToJSONTyped(json, false);
}
function ZOOMAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'refresh-token': value['refreshToken'],
    };
}
