// @vectorize-io/vectorize-client/dist/models/ZOOMADMINAuthConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZOOMADMINAuthConfig = instanceOfZOOMADMINAuthConfig;
exports.ZOOMADMINAuthConfigFromJSON = ZOOMADMINAuthConfigFromJSON;
exports.ZOOMADMINAuthConfigFromJSONTyped = ZOOMADMINAuthConfigFromJSONTyped;
exports.ZOOMADMINAuthConfigToJSON = ZOOMADMINAuthConfigToJSON;
exports.ZOOMADMINAuthConfigToJSONTyped = ZOOMADMINAuthConfigToJSONTyped;
/**
 * Check if a given object implements the ZOOMADMINAuthConfig interface.
 */
function instanceOfZOOMADMINAuthConfig(value) {
    if (!('refreshToken' in value) || value['refreshToken'] === undefined)
        return false;
    return true;
}
function ZOOMADMINAuthConfigFromJSON(json) {
    return ZOOMADMINAuthConfigFromJSONTyped(json, false);
}
function ZOOMADMINAuthConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'refreshToken': json['refresh-token'],
    };
}
function ZOOMADMINAuthConfigToJSON(json) {
    return ZOOMADMINAuthConfigToJSONTyped(json, false);
}
function ZOOMADMINAuthConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'refresh-token': value['refreshToken'],
    };
}
