// @vectorize-io/vectorize-client/dist/models/ZOOMConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZOOMConfig = instanceOfZOOMConfig;
exports.ZOOMConfigFromJSON = ZOOMConfigFromJSON;
exports.ZOOMConfigFromJSONTyped = ZOOMConfigFromJSONTyped;
exports.ZOOMConfigToJSON = ZOOMConfigToJSON;
exports.ZOOMConfigToJSONTyped = ZOOMConfigToJSONTyped;
/**
 * Check if a given object implements the ZOOMConfig interface.
 */
function instanceOfZOOMConfig(value) {
    if (!('startDate' in value) || value['startDate'] === undefined)
        return false;
    return true;
}
function ZOOMConfigFromJSON(json) {
    return ZOOMConfigFromJSONTyped(json, false);
}
function ZOOMConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'startDate': (new Date(json['start-date'])),
        'endDate': json['end-date'] == null ? undefined : (new Date(json['end-date'])),
        'titleFilter': json['title-filter'] == null ? undefined : json['title-filter'],
        'maxMeetings': json['max-meetings'] == null ? undefined : json['max-meetings'],
    };
}
function ZOOMConfigToJSON(json) {
    return ZOOMConfigToJSONTyped(json, false);
}
function ZOOMConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'start-date': ((value['startDate']).toISOString().substring(0, 10)),
        'end-date': value['endDate'] == null ? undefined : ((value['endDate']).toISOString().substring(0, 10)),
        'title-filter': value['titleFilter'],
        'max-meetings': value['maxMeetings'],
    };
}
