// @vectorize-io/vectorize-client/dist/models/Zoom.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoomTypeEnum = void 0;
exports.instanceOfZoom = instanceOfZoom;
exports.ZoomFromJSON = ZoomFromJSON;
exports.ZoomFromJSONTyped = ZoomFromJSONTyped;
exports.ZoomToJSON = ZoomToJSON;
exports.ZoomToJSONTyped = ZoomToJSONTyped;
var ZOOMAuthConfig_1 = require("./ZOOMAuthConfig");
/**
 * @export
 */
exports.ZoomTypeEnum = {
    Zoom: 'ZOOM'
};
/**
 * Check if a given object implements the Zoom interface.
 */
function instanceOfZoom(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function ZoomFromJSON(json) {
    return ZoomFromJSONTyped(json, false);
}
function ZoomFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, ZOOMAuthConfig_1.ZOOMAuthConfigFromJSON)(json['config']),
    };
}
function ZoomToJSON(json) {
    return ZoomToJSONTyped(json, false);
}
function ZoomToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, ZOOMAuthConfig_1.ZOOMAuthConfigToJSON)(value['config']),
    };
}
