// @vectorize-io/vectorize-client/dist/models/ZoomAdmin.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoomAdminTypeEnum = void 0;
exports.instanceOfZoomAdmin = instanceOfZoomAdmin;
exports.ZoomAdminFromJSON = ZoomAdminFromJSON;
exports.ZoomAdminFromJSONTyped = ZoomAdminFromJSONTyped;
exports.ZoomAdminToJSON = ZoomAdminToJSON;
exports.ZoomAdminToJSONTyped = ZoomAdminToJSONTyped;
var ZOOMADMINAuthConfig_1 = require("./ZOOMADMINAuthConfig");
/**
 * @export
 */
exports.ZoomAdminTypeEnum = {
    ZoomAdmin: 'ZOOM_ADMIN'
};
/**
 * Check if a given object implements the ZoomAdmin interface.
 */
function instanceOfZoomAdmin(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('type' in value) || value['type'] === undefined)
        return false;
    if (!('config' in value) || value['config'] === undefined)
        return false;
    return true;
}
function ZoomAdminFromJSON(json) {
    return ZoomAdminFromJSONTyped(json, false);
}
function ZoomAdminFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'type': json['type'],
        'config': (0, ZOOMADMINAuthConfig_1.ZOOMADMINAuthConfigFromJSON)(json['config']),
    };
}
function ZoomAdminToJSON(json) {
    return ZoomAdminToJSONTyped(json, false);
}
function ZoomAdminToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'type': value['type'],
        'config': (0, ZOOMADMINAuthConfig_1.ZOOMADMINAuthConfigToJSON)(value['config']),
    };
}
