// @vectorize-io/vectorize-client/dist/models/ZoomAdmin1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZoomAdmin1 = instanceOfZoomAdmin1;
exports.ZoomAdmin1FromJSON = ZoomAdmin1FromJSON;
exports.ZoomAdmin1FromJSONTyped = ZoomAdmin1FromJSONTyped;
exports.ZoomAdmin1ToJSON = ZoomAdmin1ToJSON;
exports.ZoomAdmin1ToJSONTyped = ZoomAdmin1ToJSONTyped;
var ZOOMADMINAuthConfig_1 = require("./ZOOMADMINAuthConfig");
/**
 * Check if a given object implements the ZoomAdmin1 interface.
 */
function instanceOfZoomAdmin1(value) {
    return true;
}
function ZoomAdmin1FromJSON(json) {
    return ZoomAdmin1FromJSONTyped(json, false);
}
function ZoomAdmin1FromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'config': json['config'] == null ? undefined : (0, ZOOMADMINAuthConfig_1.ZOOMADMINAuthConfigFromJSON)(json['config']),
    };
}
function ZoomAdmin1ToJSON(json) {
    return ZoomAdmin1ToJSONTyped(json, false);
}
function ZoomAdmin1ToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'config': (0, ZOOMADMINAuthConfig_1.ZOOMADMINAuthConfigToJSON)(value['config']),
    };
}
