// @vectorize-io/vectorize-client/dist/models/ZOOMADMINConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Vectorize API
 * API for Vectorize services (Beta)
 *
 * The version of the OpenAPI document: 0.1.2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfZOOMADMINConfig = instanceOfZOOMADMINConfig;
exports.ZOOMADMINConfigFromJSON = ZOOMADMINConfigFromJSON;
exports.ZOOMADMINConfigFromJSONTyped = ZOOMADMINConfigFromJSONTyped;
exports.ZOOMADMINConfigToJSON = ZOOMADMINConfigToJSON;
exports.ZOOMADMINConfigToJSONTyped = ZOOMADMINConfigToJSONTyped;
/**
 * Check if a given object implements the ZOOMADMINConfig interface.
 */
function instanceOfZOOMADMINConfig(value) {
    if (!('startDate' in value) || value['startDate'] === undefined)
        return false;
    if (!('participantFilterType' in value) || value['participantFilterType'] === undefined)
        return false;
    return true;
}
function ZOOMADMINConfigFromJSON(json) {
    return ZOOMADMINConfigFromJSONTyped(json, false);
}
function ZOOMADMINConfigFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'startDate': (new Date(json['start-date'])),
        'endDate': json['end-date'] == null ? undefined : (new Date(json['end-date'])),
        'userEmails': json['user-emails'] == null ? undefined : json['user-emails'],
        'titleFilter': json['title-filter'] == null ? undefined : json['title-filter'],
        'maxMeetings': json['max-meetings'] == null ? undefined : json['max-meetings'],
        'participantFilterType': json['participant-filter-type'],
        'participantFilter': json['participant-filter'] == null ? undefined : json['participant-filter'],
    };
}
function ZOOMADMINConfigToJSON(json) {
    return ZOOMADMINConfigToJSONTyped(json, false);
}
function ZOOMADMINConfigToJSONTyped(value, ignoreDiscriminator) {
    if (ignoreDiscriminator === void 0) { ignoreDiscriminator = false; }
    if (value == null) {
        return value;
    }
    return {
        'start-date': ((value['startDate']).toISOString().substring(0, 10)),
        'end-date': value['endDate'] == null ? undefined : ((value['endDate']).toISOString().substring(0, 10)),
        'user-emails': value['userEmails'],
        'title-filter': value['titleFilter'],
        'max-meetings': value['maxMeetings'],
        'participant-filter-type': value['participantFilterType'],
        'participant-filter': value['participantFilter'],
    };
}
